<!DOCTYPE html>

<html lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Identify Pathways &middot; Pathway Guide
    
  </title>

  <!-- Icons -->
  <link rel="shortcut icon" href="/guide/media/favicons/favicon.ico">
  <link rel="apple-touch-icon" sizes="180x180" href="/guide/media/favicons/apple-touch-icon.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/guide/media/favicons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/guide/media/favicons/favicon-16x16.png">
  <link rel="manifest" href="/guide/media/favicons/manifest.json">
  <meta name="theme-color" content="#ffffff">

  <!-- CSS -->
  
  <!-- Third-party CSS -->
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

  <!-- Custom CSS -->
  <link rel="stylesheet" href="/guide/public/css/main.css">

  <!-- Javascript -->
  <!-- Third-party javascript -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
  <script type="text/javascript" src="/guide/public/js/deps.js"></script>
  <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
    MathJax.Hub.Config({ TeX: { equationNumbers: { autoNumber: "AMS" } } });
  </script>
  <!-- Custom javascript -->
  <script type="text/javascript" src="/guide/public/js/babel-compiled.js" defer></script>
  
</head>


  <body class="theme-base-01">
    <a name="top"></a>
    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
         content to avoid any CSS collisions with our real content. -->
    <div class="wrap">
    <div class="masthead">
  <div class="container">
    <h1 class="masthead-title">
      <a href="/guide/" title="Home">Pathway Guide</a>
    </h1>
    <a class="masthead-author" href="http://www.pathwaycommons.org/">
        Pathway Commons
    </a>
  </div>
</div>


      <div class="container content">
        <div id='ajax-spinner'><img src="/guide/media/spinner.gif"/></div>
        <div class="page">
    

  
    
    
    
    <ol class="breadcrumb">
    
      
      

      <!-- 1st part is the root  -->
        <li><a href="/guide/">Guide</a></li>
      
    
      
      

      <!-- If its a collection it is an index page  -->
        
          <li><a href="/guide/workflows/archive/">Workflows</a></li>
        
      
    
      
      

      
        <li><a href="/guide/workflows/rna_seq_to_enrichment_map/">Rna seq to enrichment map</a></li>
      
    
      
      

      <!-- If its the second last path  -->
        <li class="active">Identify Pathways</li>
        
    </ol>
  


  

  <h1 class="page-title">Identify Pathways</h1>
  <hr />

<ul>
  <li class="list-unstyled">Table of Contents
    <ul>
      <li class="list-unstyled"><a href="#goals">I. Goals</a></li>
      <li class="list-unstyled"><a href="#background">II. Background</a></li>
      <li class="list-unstyled"><a href="#workflow_step">III. Workflow Step</a></li>
    </ul>
  </li>
</ul>

<hr />

<h2 id="i-goals"><a href="#goals" name="goals">I. Goals</a></h2>
<p>Previously, we introduced a study by Best <em>et al.</em> (Best 2015) that examined the feasibility of using platelet transcriptomes to distinguish individuals diagnosed with breast cancer (BrCa) from healthy donors (HD). The workflow step took in RNA-Seq counts and metadata describing the samples (i.e. HD or  BrCa) and analyzed the data for gene-wise differential expression (DE). One of the outputs from that step is a list of each RNA species and a respective rank, calculated from the DE test. In brief, the magnitude of rank is proportional to the ‘rareness’ of a  difference in RNA counts at least as large as that observed, <em>assuming no association between sample class assignment and RNA count</em>.</p>

<blockquote>
  <p>Note: For the sake of simplicity, we will use the terms ‘pathway’ and ‘gene set’ interchangeably. Typically, a ‘pathway’ refers to gene set components that enact a change or product (e.g. signal transduction, metabolism).</p>
</blockquote>

<p>In this section we discuss the use of <a href="http://software.broadinstitute.org/gsea/index.jsp" target="_blank">Gene Set Enrichment Analysis (GSEA)</a> to sift out pathways from the underlying alterations in gene expression (Figure 1). The pathways that emerge from this analysis will be passed on to the next workflow step where we attempt to simplify their interpretation using the Enrichment Map visualization tool.</p>

<p>By then end of this discussion you should:</p>

<ol>
  <li>Be able to install and set up the GSEA desktop software</li>
  <li>Be able to use GSEA to identify enriched gene sets using as inputs a rank file and a gene set database. The gene set database is pre-prepared.</li>
  <li>Obtain a pair of reports describing gene sets enrichment in BrCa and HD that are dependencies for subsequent workflow steps</li>
</ol>

<p><br /></p>

<p><img src="figure_idpathways_overview.jpg" alt="image" class="img-responsive slim" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 1. Workflow step goals.</strong> Gene Set Enrichment Analysis (GSEA) attempts to answer the question: Can we summarize changes in gene expression as a list of altered pathways? Here we provide GSEA with a list where each gene is assigned a rank based upon differential expression testing along with a database of candidate gene sets to test. The goal of this workflow step will be to obtain two reports, each describing the gene sets up-regulated in either of the input sample classes.
</div>

<h2 id="ii-background"><a href="#background" name="background">II. Background</a></h2>

<p>A detailed description of GSEA is beyond the scope of this discussion. Recall that, in addition to a rank file, GSEA requires a set of candidate gene sets in the form of a gene set database file (Figure 1). Below we describe a typical gene set database and provide an extremely brief overview for how GSEA operates.</p>

<h3 id="is-there-a-difference-between-gene-sets-and-pathways">Is there a difference between ‘gene sets’ and ‘pathways’?</h3>

<p>GSEA searches through candidate gene sets to identify those that are enriched within our ranked gene list. A gene set is a list of genes that share some attribute. For example, a gene set could consist of loci assigned to a particular chromosomal band. Alternatively, pathways - those genes whose coordinated activity lead to change or produce a product - can also be represented as a gene set. For example, a gene set for a pathway could be represented by genes involved in DNA damage cell cycle checkpoint.</p>

<p>Concretely, let us consider the ‘IL-5 Signaling Pathway’ curated by as part of the NetPath database (Figure 2).</p>

<p><img src="figure_idpathways_il5_netpath_diagram.png" alt="image" class="img-responsive short" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 2. Diagram of the IL5 Signaling Pathway.</strong> IL-5 is a glycoprotein which belongs to the cytokine superfamily and plays an important role in the proliferation and differentiation of eosinophils. IL-5 is produced by eosinophils, mast cells, Th2 cells, Tc2 cells and gamma delta T cells. The signaling pathway has been implicated in the pathogenesis of asthma, hypereosinophilic syndromes and eosinophil-dependent inflammatory diseases. <em>Adapted from resources at NetPath (http://www.netpath.org/)</em>.
</div>

<p>The pathway depicted in Figure 2 has a corresponding entry in NetPath that contains a listing of the genes and a brief description (Figure 3).</p>

<p><img src="figure_idpathways_il5_netpath_list.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 3. NetPath entry corresponding to IL-5 Signaling Pathway.</strong> <em>NetPath Accession number: NetPath_17</em>.
</div>

<h3 id="what-is-a-gene-set-database">What is a ‘gene set database’?</h3>

<p>The group of candidate gene sets under consideration for a given GSEA run can be individually selected to reflect the interest and focus of the study at hand. More often, GSEA is used in an exploratory fashion and so candidate gene sets curated by others are often employed, which typically cover a wide swath.</p>

<p>To make life easier, GSEA has standardized the acceptable formats for defining gene set candidates and these are collectively known as <a href="http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#Gene_Set_Database_Formats" target="_blank">gene set databases</a>. These are simple tab-delimited files with the extensions that include <code class="highlighter-rouge">.gmx</code>, <code class="highlighter-rouge">gmt</code> and <code class="highlighter-rouge">.grp</code>. Concretely, let us consider a sample gene set database file developed by <a href="http://www.baderlab.org/" target="_blank">Gary Bader’s laboratory</a> that defines human gene sets gathered from several sources including <a href="http://software.broadinstitute.org/gsea/msigdb/index.jsp" target="_blank">MSigDB</a>; <a href="http://geneontology.org/" target="_blank">Gene Ontology</a>; <a href="http://www.reactome.org/" target="_blank">Reactome</a>; <a href="http://www.pantherdb.org/" target="_blank">Panther</a>; <a href="http://www.netpath.org/" target="_blank">NetPath</a>; <a href="http://www.ndexbio.org/#/user/nci-pid" target="_blank">NCI</a>; and <a href="http://humancyc.org/" target="_blank">HumanCyc</a>. This database file defines a table, one gene set per row: The first column holds the gene set name, the second is a description and the subsequent fields are the names of genes in the set. Figure 4 shows an exceprt of the database, which has an entry for the NetPath IL-5 pathway.</p>

<p><img src="figure_idpathways_gmt.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 4. Gene set database excerpt</strong>. Gene sets are defined in rows where the first column is the name, the second is a description and gene names within the set occupy the remaining row entries. Highlighted in blue is the NetPath IL-5 signal transduction pathway that includes 59 gene names.
  <div class="text-left">
    <a type="button" class="btn btn-info" href="figure_idpathways_gmt.png" target="_blank">Open in new window</a>
  </div>
</div>

<h3 id="finding-enriched-gene-sets">Finding enriched gene sets</h3>

<p>Once we have gathered our candidate gene set database and the ranked gene list, we are ready to proceed. The details of GSEA are rather technical; Instead, we provide an overview of the algorithm (Box 1).</p>

<div class="box">
  <div class="box-title">Box 1. GSEA steps</div>

  <dl class="box-terms">

    <ol>
      <li>
        Select one candidate gene set from the database
      </li>
      <li>
        Calculate an enrichment score (ES) for selected candidate gene set
        <blockquote>
          An ES effectively tells us how many of the candidate set's genes are 'bunched up' at the top (positive rank value) or the bottom (negative rank value) of the ranked gene list. More 'bunching' at the top means that the candidate's genes are more likely to be truly upregulated and <em>vice versa</em>.
        </blockquote>
      </li>
      <li>
        Calculate an ES for many (1 000) gene sets composed of randomly chosen genes of the same size as the selected gene set
      </li>
      <li>
        Calculate a p-value
        <blockquote>
          This is the fraction of the random gene set ESs that exceed the candidate ES. A smaller p-value corresponds to a increasingly rare observation.
        </blockquote>
      </li>
      <li>
        Repeat for each remaining candidate gene set
      </li>
      <li>
        Correct for <a href="/guide/primers/statistics/multiple_testing/"> multiple testing</a>
        <blockquote>
          The chance that a gene set will be significant increases with the number of gene sets tested. Thus, corrections must be made to take this into account.
        </blockquote>
      </li>
    </ol>
  </dl>

</div>

<h2 id="iii-workflow-step"><a href="#workflow_step" name="workflow_step">III. Workflow Step</a></h2>

<p>Let us review what we have learned in our lead up to this step: GSEA can be used to identify altered pathways from underlying changes in gene expression; The two inputs to GSEA are a rank gene list generated from our differential expression analysis along with a gene set database composed of candidate gene sets that GSEA will filter.</p>

<p>In this workflow step, we will use GSEA to generate two files that report enriched gene sets in each of the two conditions tested. An  <a href="http://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm#_Detailed_Enrichment_Results" target="_blank">enrichment report</a> is a table that lists the statistics describing the extent of enrichment (Table 1).</p>

<p><strong>Table 1. Sample enrichment report</strong></p>

<table class="table table-hover table-condensed table-responsive">
  <thead>
    <tr>
      <th>NAME</th>
      <th>…</th>
      <th>SIZE</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>…</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>EUKARYOTIC TRANSLATION ELONGATION</td>
      <td>…</td>
      <td>82</td>
      <td>-0.9144994</td>
      <td>-3.0786333</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <td>…</td>
      <td>…</td>
      <td>…</td>
      <td>…</td>
      <td>…</td>
      <td>…</td>
      <td>…</td>
      <td>…</td>
    </tr>
  </tbody>
</table>

<p>Previously, we compared RNA counts in BrCa relative to HD. Accordingly, GSEA will provide one report for each class, or in GSEA terminology, each ‘phenotype’:</p>

<ol>
  <li>Report for BrCa enrichment - Pathways that are enriched in BrCa relative to HD</li>
  <li>Report for HD enrichment - Pathways that are enriched in HD relative to BrCa</li>
</ol>

<p>These reports are the dependencies for the next workflow step where we use software to view and interact with these enriched pathways in a manner that is far more interpretable than a list of output.</p>

<h4 id="software-requirements">Software requirements</h4>

<p>We are opting to use the desktop version of GSEA. There are some hoops to jump through in order to get it:</p>

<ol>
  <li>Install <a href="http://www.oracle.com/technetwork/java/javase/downloads/jre8-downloads-2133155.html" target="_blank">Java version 8</a></li>
  <li><a href="http://software.broadinstitute.org/gsea/register.jsp" target="_blank">Register</a> to use the GSEA software</li>
  <li><a href="http://software.broadinstitute.org/gsea/login.jsp" target="_blank">Login</a></li>
  <li><a href="http://software.broadinstitute.org/gsea/downloads.jsp" target="_blank">Download</a></li>
</ol>

<p>Launch the GSEA application. You will see the GSEA logo splash then the application itself (Figure 5).</p>

<p><img src="figure_idpathways_gsea_start.png" alt="image" class="img-responsive" /></p>

<div class="figure-legend well well-lg text-justify">
  <strong>Figure 5. The GSEA interface.</strong> The control panel on the left (e.g. <code>Steps in GSEA analysis</code>) provides quick access to the most common actions. The main window displays the <code>Home</code> tab by default. Each control panel action typically opens a new tab in the main window.
  <div class="text-left">
    <a type="button" class="btn btn-info" href="figure_idpathways_gsea_start.png" target="_blank">Open in new window</a>
  </div>
</div>

<h3 id="input">Input</h3>

<h4 id="ranked-gene-list">Ranked gene list</h4>

<p>This is the output from the previous step.</p>

<p><a href="/guide/workflows/rna_seq_to_enrichment_map/process_data/brca_hd_tep_ranks.rnk" type="button" class="btn btn-info btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Ranks (.rnk)</a></p>

<h4 id="gene-set-database">Gene set database</h4>

<p>This is the human gene set database culled from various sources, generated by the Bader lab (<code class="highlighter-rouge">Human_GOBP_AllPathways_no_GO_iea_February_01_2017_symbol.gmt</code>).</p>

<p><a href="Human_GOBP_AllPathways_no_GO_iea_February_01_2017_symbol.gmt.zip" type="button" class="btn btn-info btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Gene Set Database (.gmt)</a></p>

<hr />

<p>Let us assume you have installed the GSEA software and placed the input files in an <code class="highlighter-rouge">input</code> directory. Then you should now have a directory structure similar to the following.</p>

<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>...
|--- input
|    |
|    |--- brca_hd_tep_ranks.rnk
|    |--- Human_GOBP_AllPathways_no_GO_iea_February_01_2017_symbol.gmt
...
</code></pre></div></div>

<h3 id="analysis">Analysis</h3>

<p>Here we will step through four tasks that we will need to perform within the GSEA software that will generate our enrichment reports.</p>

<h4 id="1-load-data-files">1. Load data files</h4>

<p>In this part we get our inputs into GSEA. In the <code class="highlighter-rouge">Steps in GSEA analysis</code> panel (Figure 5, left) click the <code class="highlighter-rouge">Load data</code> button which will bring up a panel in the main window (Figure 6).</p>

<ol>
  <li><code class="highlighter-rouge">Browse for files...</code> then <code class="highlighter-rouge">Choose</code> the ranked gene list</li>
  <li><code class="highlighter-rouge">Browse for files...</code> then <code class="highlighter-rouge">Choose</code> the gene set database file</li>
</ol>

<p><img src="figure_idpathways_load.png" alt="image" class="img-responsive" /></p>

<div class="figure-legend well well-lg text-justify">
  <strong>Figure 6. Load data tab.</strong> To bring up this tab, click the <code>Load data</code> button in the control panel <code>Steps in GSEA analysis</code>.
</div>

<blockquote>
  <p>Wait a few seconds for the files to load into memory. You will receive a pop-up dialog if the file was successfully loaded. You will also see the files in the <code>Object cache</code> panel of the <code>Load data</code> tab.</p>
</blockquote>

<h4 id="2--settings">2.  Settings</h4>

<p>We now tell GSEA what these files actually represent and tailor the GSEA run accordingly. Bring up the GSEA pre-ranked tab by selecting <code class="highlighter-rouge">GseaPreranked</code> from the <code class="highlighter-rouge">Tools</code> panel and fill in the details for the <code class="highlighter-rouge">Required</code> and <code class="highlighter-rouge">Basic</code> fields (Figure 7).</p>

<ul>
  <li><code class="highlighter-rouge">Required fields</code>
    <ul>
      <li><code class="highlighter-rouge">Gene sets database</code>: Click the ellipsis and wait a few moments for a dialog to pop up. Navigate to <code class="highlighter-rouge">Gene matrix (local gmx/gmt)</code> (click arrow along top). Select the gene set database that you obtained above (<code class="highlighter-rouge">.gmt</code> file)</li>
      <li><code class="highlighter-rouge">Collapse dataset to gene symbols</code>: False
        <ul>
          <li>This option is a vestige of microarray technology where an additional step was necessary to match array probes/spots to a single gene identifier.</li>
        </ul>
      </li>
    </ul>
  </li>
  <li><code class="highlighter-rouge">Basic fields</code>
    <ul>
      <li><code class="highlighter-rouge">Analysis name</code>: Choose a name for this particular run of GSEA</li>
      <li><code class="highlighter-rouge">Save results in this folder</code>: Choose one</li>
    </ul>
  </li>
</ul>

<p><img src="figure_idpathways_settings.png" alt="image" class="img-responsive" /></p>

<div class="figure-legend well well-lg text-justify">
  <strong>Figure 7. The GSEA Pre-ranked tab.</strong> Select <code>Tools</code> in the menu dropdown, then <code>GseaPreranked</code> to bring up the <code>Run GSEA on a Pre-Ranked gene list</code> tab.
</div>

<h4 id="3-run-gsea">3. Run GSEA</h4>

<p>Click <code class="highlighter-rouge">Run</code> in the <code class="highlighter-rouge">Run Gsea on a Pre-Ranked gene list</code> tab. The <code class="highlighter-rouge">GSEA reports</code> panel (Figure 5, bottom left) will show the <code class="highlighter-rouge">Name</code> of this run and the <code class="highlighter-rouge">Status</code> as <code class="highlighter-rouge">Running</code> while in progress.</p>

<ul>
  <li class="aside">
    <h4 id="typical-run-times">Typical run times</h4>

    <ul>
      <li>Mac: MacBook Air (Early 2015), 2.2 GHz Intel Core i7, 8 GB 1600 MHz DDR3, OS X El Capitan (10.11.6) ~ 10 minutes</li>
    </ul>
  </li>
</ul>

<h4 id="4-post-gsea">4. Post-GSEA</h4>

<p>Take a look at the directory you set for <code class="highlighter-rouge">Save results in this folder</code>. The default location is <code class="highlighter-rouge">gsea_home</code> in your user space. You should see something like the following:</p>

<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>...
|--- input
|    |
|    |--- brca_hd_tep_ranks.rnk
|    |--- Human_GOBP_AllPathways_no_GO_iea_February_01_2017_symbol.gmt
|
|--- gsea_home
    |
    |--- output
        |
        |--- monthDay
            |
            |--- tep_brca_vs_hd.GseaPreranked.XXXXXXXXXXXXX
                |
                |--- index.html
                |--- pos_snapshot.html
                |--- neg_snapshot.html
                |--- gsea_report_for_na_pos_XXXXXXXXXXXXX.xls
                |--- gsea_report_for_na_neg_XXXXXXXXXXXXX.xls
                |--- my_analysis.GseaPreranked.XXXXXXXXXXXXX.rpt
                ...
...
</code></pre></div></div>

<h5 id="gsea-report">GSEA report</h5>

<p>When the GSEA software has completed its analysis, the <code class="highlighter-rouge">Status</code> inside the  <code class="highlighter-rouge">GSEA reports</code> panel will update to <code class="highlighter-rouge">Success ...</code> (Figure 5, bottom left). You may click this link to view the HTML report inside a browser (Figure 8). Alternatively, open the <code class="highlighter-rouge">index.html</code> file located in the GSEA results directory in a browser.</p>

<p><img src="figure_idpathways_index.png" alt="image" class="img-responsive" /></p>

<div class="figure-legend well well-lg text-justify">
  <strong>Figure 8. The GSEA Report.</strong> Example of a GSEA report opened inside a browser. The links inside this report reference local files declared in the GSEA results folder. Note the two sections, one for each class: In our case the <code>na_pos</code> phenotype corresponds to the BrCa class and <code>na_neg</code> refers to the HD class.
</div>

<blockquote>
  <p>You can read the complete guide to <a href="http://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm#_Interpreting_GSEA_Results" target="_blank">Interpreting GSEA Results</a> that includes a description of the GSEA report.</p>
</blockquote>

<p>Below we will briefly highlight a few aspects to take special note of.</p>

<h5 id="enrichment-in-phenotype">Enrichment in phenotype</h5>

<p>There are two sections by this name which refers to those gene sets with positive and negative enrichment scores, respectively (Figure 8). Recall that we assessed gene expression in the BrCa class relative to the HD class. Consequently, in our case, the <code class="highlighter-rouge">na_pos</code> phenotype corresponds to the BrCa class and <code class="highlighter-rouge">na_neg</code> refers to the HD class. We briefly discuss the meaning of two of the entries in this section below.</p>

<ul>
  <li>
    <p><strong>Snapshot of the enrichment results.</strong> Recall from Box 1, Step 2 that during GSEA, an enrichment score (ES) is calculated for each candidate gene set. The snapshots display the raw data that GSEA uses in its calculations: The ES is the peak vertical axis value of this plot. Roughly speaking, the magnitude of the ES is indicative of the chances that it will be deemed significant. If you look inside your GSEA home folder you will see <code class="highlighter-rouge">pos_snapshot.html</code> and <code class="highlighter-rouge">neg_snapshot.html</code>. For the <code class="highlighter-rouge">na_pos</code> case, click on the link for <code class="highlighter-rouge">Snapshot of enrichment results</code> to bring up a panel of enrichment score plots; There should be an entry for our old friend the IL-5 signal transduction pathway, which you can click to bring up the full report showing the summary, genes and the enrichment plot (Figure 9).</p>

    <p><img src="figure_idpathways_nesplot_il5netpath.png" alt="image" class="img-responsive ht-400" /></p>

    <div class="figure-legend well well-lg text-justify">
  <strong>Figure 9. Enrichment plot for NetPath IL-5 signalling pathway.</strong> Shown is a the running sum plot for the IL-5 signal transduction pathway described earlier.
  <div class="text-left">
    <a type="button" class="btn btn-info" href="figure_idpathways_nesplot_il5netpath.png" target="_blank">Open in new window</a>
  </div>
</div>
  </li>
  <li>
    <p><strong>Detailed enrichment results.</strong> These are the end-goal for our workflow step! These files provide a summary report of gene sets enriched in this phenotype. You can find these files inside your GSEA results folder named <code class="highlighter-rouge">gsea_report_for_na_pos_XXXXXXXXXXXXX.xls</code> and <code class="highlighter-rouge">gsea_report_for_na_neg_XXXXXXXXXXXXX.xls</code>. The rows of the report include information for each enriched gene set (Table 2).</p>

    <p><strong>Table 2. Exceprt of enrichment report for BrCa class</strong></p>

    <table class="table table-hover table-condensed table-responsive">
      <thead>
        <tr>
          <th style="text-align: left">NAME</th>
          <th style="text-align: center">SIZE</th>
          <th style="text-align: center">ES</th>
          <th style="text-align: center">NES</th>
          <th style="text-align: center">NOM p-val</th>
          <th style="text-align: center">FDR q-val</th>
          <th style="text-align: center">FWER p-val</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td style="text-align: left">PID_FAK_PATHWAY</td>
          <td style="text-align: center">48</td>
          <td style="text-align: center">0.8144367</td>
          <td style="text-align: center">2.2033582</td>
          <td style="text-align: center">0.0</td>
          <td style="text-align: center">0.0</td>
          <td style="text-align: center">0.0</td>
        </tr>
        <tr>
          <td style="text-align: left">THROMBOXANE A2 RECEPTOR SIGNALING</td>
          <td style="text-align: center">39</td>
          <td style="text-align: center">0.84532094</td>
          <td style="text-align: center">2.175933</td>
          <td style="text-align: center">0.0</td>
          <td style="text-align: center">0.0</td>
          <td style="text-align: center">0.0</td>
        </tr>
        <tr>
          <td style="text-align: left">SIGNALING EVENTS MEDIATED BY FOCAL ADHESION KINASE</td>
          <td style="text-align: center">48</td>
          <td style="text-align: center">0.8144367</td>
          <td style="text-align: center">2.16101</td>
          <td style="text-align: center">0.0</td>
          <td style="text-align: center">0.0</td>
          <td style="text-align: center">0.0</td>
        </tr>
        <tr>
          <td style="text-align: left">…</td>
          <td style="text-align: center">…</td>
          <td style="text-align: center">…</td>
          <td style="text-align: center">…</td>
          <td style="text-align: center">…</td>
          <td style="text-align: center">…</td>
          <td style="text-align: center">…</td>
        </tr>
        <tr>
          <td style="text-align: left">IL5%NETPATH%IL5	IL5</td>
          <td style="text-align: center">49</td>
          <td style="text-align: center">0.7718686</td>
          <td style="text-align: center">2.0658088</td>
          <td style="text-align: center">0.0</td>
          <td style="text-align: center">0.0</td>
          <td style="text-align: center">0.0</td>
        </tr>
        <tr>
          <td style="text-align: left">…</td>
          <td style="text-align: center">…</td>
          <td style="text-align: center">…</td>
          <td style="text-align: center">…</td>
          <td style="text-align: center">…</td>
          <td style="text-align: center">…</td>
          <td style="text-align: center">…</td>
        </tr>
      </tbody>
    </table>
  </li>
</ul>

<h3 id="output"><a href="#output" name="output">Output</a></h3>

<p>Listed below are the outputs of this step that will be required as input dependencies for the next steps of the workflow.</p>

<h4 id="enrichment-map-dependencies">Enrichment Map dependencies</h4>

<p>We have renamed the enrichment report files and provide them below.</p>

<ol>
  <li>
    <p>Enrichment report for BrCa
  <a href="gsea_report_for_brca.xls" type="button" class="btn btn-success btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> BrCa report (.xls)</a></p>
  </li>
  <li>
    <p>Enrichment report for HD
  <a href="gsea_report_for_hd.xls" type="button" class="btn btn-success btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> HD report (.xls)</a></p>
  </li>
</ol>

  
  <hr/>
  <h2>References</h2>
  
  <div class="reference_group" data-reflist="[26525104]"></div>


  
  <br/>
  <hr/>
  <div id="disqus_thread"></div>
  <script>
    /**
    *  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
    *  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
    var disqus_config = function () {
    this.page.url = "http://pathwaycommons.github.io/guide/workflows/rna_seq_to_enrichment_map/identify_pathways/index";  // Replace PAGE_URL with your page's canonical URL variable
    this.page.identifier = "/workflows/rna_seq_to_enrichment_map/identify_pathways/index"; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
    };

    (function() { // DON'T EDIT BELOW THIS LINE
    var d = document, s = d.createElement('script');
    s.src = '//pathway-commons.disqus.com/embed.js';
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
    })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>


</div>

      </div>
    </div>

  </body>
</html>
