<!DOCTYPE html>

<html lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Process Data &middot; Guide
    
  </title>

  <!-- CSS -->
  <!-- Third-party CSS -->
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css" />
  <!-- Custom CSS -->
  <link rel="stylesheet" href="/guide/public/css/main.css">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/guide/media/apple-touch-icon-precomposed.png">
  <link rel="shortcut icon" href="/guide/media/favicon.ico">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">

  <!-- Javascript -->
  <!-- Third-party javascript -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/react/15.0.1/react.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/react/15.0.1/react-dom.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.2/jquery.min.js"></script>
  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js" integrity="sha384-0mSbJDEHialfmuBBQP6A4Qrprq5OVfW37PRR3j5ELqxss1yVqOtnepnHVP9aJ7xS" crossorigin="anonymous"></script>
  <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
    MathJax.Hub.Config({ TeX: { equationNumbers: { autoNumber: "AMS" } } });
  </script>

  <!-- Custom javascript -->
  <script type="text/javascript" src="/guide/public/js/babel-compiled.js" defer></script>

  
</head>


  <body class="theme-base-01">
    <div id="embedded">
    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
           content to avoid any CSS collisions with our real content. -->
      <div class="wrap">
        <div class="container content embedded">
          <div id='ajax-spinner'><img src="/guide/media/spinner.gif"/></div>

            <h1 class="page-title">Process Data</h1>
            <div class="subtitle">Derive a list of differentially expressed genes from TCGA-OV RNA sequencing data</div>
            <hr/>
            <div class="linkout clearfix">
              <ul class="list-inline pull-right">
              <li>
  <a href="http://www.facebook.com/sharer/sharer.php?u=/workflows/pathway_enrichment/process_data/&title=Process Data"
  data-toggle="tooltip"
  title="Share this page on Facebook"
  target="_blank" >
    <i class="fa fa-facebook-official fa-2x"></i>
  </a>
</li>
<li>
  <a href="http://twitter.com/intent/tweet?text=Derive a list of differentially expressed genes from TCGA-OV RNA sequencing data+/workflows/pathway_enrichment/process_data/&via=pathwaycommons"
  data-toggle="tooltip"
  title="Tweet this page"
  target="_blank" >
    <i class="fa fa-twitter-square fa-2x"></i>
  </a>
</li>
<li>
  <a href="https://plus.google.com/share?url=/workflows/pathway_enrichment/process_data/"
  data-toggle="tooltip"
  title="Share on Google+"
  target="_blank" >
    <i class="fa fa-google-plus fa-2x"></i>
  </a>
</li>

              </ul>
            </div>
            <hr/>
            <!-- Global options -->

<ul>
  <li class="list-unstyled">Table of Contents
    <ul>
      <li class="list-unstyled"><a href="#goals">I. Goals</a></li>
      <li class="list-unstyled"><a href="#background">II. Background</a></li>
      <li class="list-unstyled"><a href="#practical">III. Practical</a></li>
      <li class="list-unstyled"><a href="#data">IV. Data</a></li>
      <li class="list-unstyled"><a href="#references">V. References</a></li>
    </ul>
  </li>
</ul>

<hr />

<div class="alert alert-warning text-justify" role="alert">
  The list of differentially expressed genes in TCGA-OV for mesenchymal relative to immunoreative subtypes ranked by p-value is in <a href="#data">IV. Data</a>.
</div>

<h2 id="a-hrefgoals-namegoalsi-goalsa"><a href="#goals" name="goals">I. Goals</a></h2>
<p>This section is a follow-up to the section describing how to get <a href="http://cancergenome.nih.gov/abouttcga/overview" target="_blank">The Cancer Genome Atlas</a> (TCGA) RNA sequencing (RNA-seq) data from high-grade serous ovarian cancer (HGS-OvCa) (Cancer Genome Atlas Research Network 2011).</p>

<p>In this section we will assess differential expression of RNA species between subtypes. Our over-overarching goal is to generate a list of those expressed genes ranked according to a probability value (p-value) suitable for downstream analysis (Figure 1).</p>

<p>By then end of this discussion you should:</p>

<ol>
  <li>Be able to use the <a href="https://www.r-project.org/" target="_blank">R</a> package <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html" target="_blank">edgeR</a> to test for differential expression</li>
  <li>Obtain a list of genes for TCGA HGS-OvCa ranked by a function of the p-value for differential expression</li>
</ol>

<p><br />
<img src="/guide/media/workflows/pathway_enrichment/process_data/figure_processdata_overview.jpg" alt="image" class="img-responsive slim" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 1. Goals.</strong> Steps required to process TCGA-OV RNA-seq data for differential expression between 'immunoreactive' and 'mesenchymal' subtypes. Text to the right of arrows are functions provided as part of the R/Bioconductor packages. In step 1, the data is filtered for genes with a minimum number of counts. The filtered data is then normalized and fit to a statistical model in steps 2-3.  In step 4, p-values for differential expression between subtypes is assessed and adjusted for multiple testing in step 5. Finally, the genes are ordered based upon a function of the p-values.
</div>

<h2 id="a-hrefbackground-namebackgroundii-backgrounda"><a href="#background" name="background">II. Background</a></h2>

<p>We refer the reader to our primer on <a href="/guide/primers/functional_analysis/rna_sequencing_analysis/" target="_blank">RNA sequencing analysis</a> for a detailed description of the theory underlying the processing steps described here.</p>

<h2 id="a-hrefpractical-namepracticaliii-practicala"><a href="#practical" name="practical">III. Practical</a></h2>

<h3 id="software-requirements">Software requirements</h3>

<p><strong>Run inside Docker</strong> (<em>Recommended</em>). To ease the burden of loading the correct software and dependencies, we have generated a <a href="https://github.com/jvwong/docker_enrichment_workflow_gdc/tree/e4735bc9d94b44259c432f11985cf0131629d53a" target="_blank">Github repository</a> containing the neccessary code to run a <a href="https://www.docker.com/" target="_blank">Docker</a> version of <a href="https://www.rstudio.com/" target="_blank">RStudio</a> linked to the necessary workflow files.</p>

<p><strong>Run code on your computer</strong>. You can run the R code on your own computer and will need the following files and software</p>

<ul>
  <li>RNA-seq assay and category information (from ‘Get Data’)
    <ul>
      <li><a href="/guide/workflows/pathway_enrichment/get_data/#data" target="_blank">TCGAOV_data.rda</a></li>
    </ul>
  </li>
  <li><a href="https://www.r-project.org/" target="_blank">R</a>: &gt;version 3.3.1
    <ul>
      <li><a href="https://bioconductor.org" target="_blank">Bioconductor</a>: &gt;version 3.3
        <ul>
          <li>edgeR</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<p>We will assume files are located in a local directory <code class="highlighter-rouge">/home/TCGA/</code>.</p>

<div class="language-shell highlighter-rouge"><pre class="highlight"><code>home
|
|--- TCGA
|   |
|   |--- scripts
|   |    |
|   |    |--- get_data.R
|   |    |--- process_data.R
|   |
|   |--- data
|   |    |
|   |    |--- Verhaak_JCI_2013_tableS1.txt
|   |
|   |--- output
|   |    |
|   |    |--- TCGAOV_data.rda
|   |    |--- TCGAOV_se_datahtseq_counts.rda
...
</code></pre>
</div>

<h3 id="data-processing">Data processing</h3>

<p><strong>Step 0: Installation and setup</strong></p>

<p>Install and load the required packages from R/Bioconductor. Load the TCGA HGS-OvCa RNA-seq data and subtype assignments in the DGEList variable <code class="highlighter-rouge">TCGAOV_data</code> from the previous section.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r">  
<span class="c1">### ============ Install packages from Bioconductor ========
</span><span class="n">library</span><span class="p">(</span><span class="s2">"edgeR"</span><span class="p">)</span>


<span class="c1">### ============ Declare directories =========
</span><span class="n">BASE_DIR</span> <span class="o">=</span> <span class="s2">"/home/TCGA"</span> <span class="c1"># same as volumes in docker-compose.yml
</span><span class="n">TCGAOV_DATA_DIR</span> <span class="o">=</span> <span class="n">file.path</span><span class="p">(</span><span class="n">BASE_DIR</span><span class="p">,</span> <span class="s2">"data"</span><span class="p">)</span>
<span class="n">TCGAOV_OUTPUT_DIR</span> <span class="o">=</span> <span class="n">file.path</span><span class="p">(</span><span class="n">BASE_DIR</span><span class="p">,</span> <span class="s2">"output"</span><span class="p">)</span>

<span class="n">TCGAOV_DGE_OUTPUT_FILE</span> <span class="o">=</span> <span class="n">file.path</span><span class="p">(</span><span class="n">TCGAOV_OUTPUT_DIR</span><span class="p">,</span>
                                  <span class="s2">"TCGAOV_data.rda"</span><span class="p">)</span>
<span class="n">TCGAOV_RANKS_OUTPUT_FILE</span> <span class="o">=</span> <span class="n">file.path</span><span class="p">(</span><span class="n">TCGAOV_OUTPUT_DIR</span><span class="p">,</span>
                 <span class="s2">"MesenchymalvsImmunoreactive_edger_ranks.rnk"</span><span class="p">)</span>

<span class="c1">### ============ Declare samples =========
</span><span class="n">CATEGORY_BASELINE</span> <span class="o">=</span> <span class="s2">"Immunoreactive"</span>
<span class="n">CATEGORY_TEST</span> <span class="o">=</span> <span class="s2">"Mesenchymal"</span>

<span class="n">load</span><span class="p">(</span><span class="n">TCGAOV_DGE_OUTPUT_FILE</span><span class="p">)</span></code></pre></figure>

<p>Set the <code class="highlighter-rouge">CATEGORY_TEST</code> to measure expression relative to <code class="highlighter-rouge">CATEGORY_BASELINE</code>.</p>

<p>The DGEList contains an attribute <code class="highlighter-rouge">counts</code> which is a table identical to our input. The attribute <code class="highlighter-rouge">samples</code> is a table created with a column <code class="highlighter-rouge">lib.size</code> that states the total counts for the case.</p>

<table class="table table-hover table-condensed table-responsive">
  <thead>
    <tr>
      <th style="text-align: center"> </th>
      <th style="text-align: center">group</th>
      <th style="text-align: center">lib.size</th>
      <th style="text-align: center">norm.factors</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><strong>TCGA-24-2024-01A-02R-1568-13</strong></td>
      <td style="text-align: center">Differentiated</td>
      <td style="text-align: center">88674430</td>
      <td style="text-align: center">1</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>TCGA-23-1026-01B-01R-1569-13</strong></td>
      <td style="text-align: center">Differentiated</td>
      <td style="text-align: center">48116062</td>
      <td style="text-align: center">1</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>TCGA-04-1357-01A-01R-1565-13</strong></td>
      <td style="text-align: center">Immunoreactive</td>
      <td style="text-align: center">38896502</td>
      <td style="text-align: center">1</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>TCGA-61-2000-01A-01R-1568-13</strong></td>
      <td style="text-align: center">Immunoreactive</td>
      <td style="text-align: center">58593539</td>
      <td style="text-align: center">1</td>
    </tr>
  </tbody>
</table>

<p><strong>Step 1: Filter</strong></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r">  
<span class="c1">### ============ 1. Filter ===============
</span><span class="n">index_BASELINE</span> <span class="o">=</span> <span class="n">TCGAOV_data</span><span class="o">$</span><span class="n">samples</span><span class="o">$</span><span class="n">group</span> <span class="o">==</span> <span class="n">CATEGORY_BASELINE</span>
<span class="n">index_TEST</span> <span class="o">=</span> <span class="n">TCGAOV_data</span><span class="o">$</span><span class="n">samples</span><span class="o">$</span><span class="n">group</span> <span class="o">==</span> <span class="n">CATEGORY_TEST</span>
<span class="n">index_BASELINE_TEST</span> <span class="o">=</span> <span class="n">which</span><span class="p">(</span><span class="n">index_BASELINE</span> <span class="o">|</span> <span class="n">index_TEST</span><span class="p">)</span>
<span class="n">N_BASELINE</span> <span class="o">=</span> <span class="n">sum</span><span class="p">(</span><span class="n">index_BASELINE</span><span class="p">)</span>
<span class="n">N_TEST</span> <span class="o">=</span> <span class="n">sum</span><span class="p">(</span><span class="n">index_TEST</span><span class="p">)</span>
<span class="n">TCGAOV_data_BT</span> <span class="o">=</span> <span class="n">TCGAOV_data</span><span class="p">[,</span><span class="n">index_BASELINE_TEST</span><span class="p">]</span>

<span class="n">row_with_mincount</span> <span class="o">=</span> <span class="n">rowSums</span><span class="p">(</span><span class="n">cpm</span><span class="p">(</span><span class="n">TCGAOV_data_BT</span><span class="p">)</span> <span class="o">&gt;</span> <span class="m">10</span><span class="p">)</span> <span class="o">&gt;=</span> <span class="n">min</span><span class="p">(</span><span class="n">N_BASELINE</span><span class="p">,</span>
                                                             <span class="n">N_TEST</span><span class="p">)</span>
<span class="n">TCGAOV_filtered</span> <span class="o">=</span> <span class="n">TCGAOV_data_BT</span><span class="p">[</span><span class="n">row_with_mincount</span><span class="p">,</span> <span class="p">,</span> <span class="n">keep.lib.sizes</span><span class="o">=</span><span class="n">FALSE</span><span class="p">]</span></code></pre></figure>

<p>The variable <code class="highlighter-rouge">row_with_mincount</code> stores genes with more than a minimum number of counts (10) per million mapped reads in n cases, where n is the smallest of the two subtypes. This step is intended to genes with low expression.</p>

<p><strong>Step 2: Normalize</strong></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r">  
<span class="c1">### ============ 2. Normalize ===============
</span><span class="n">TCGAOV_normalized_TMM</span> <span class="o">=</span> <span class="n">calcNormFactors</span><span class="p">(</span><span class="n">TCGAOV_filtered</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s2">"TMM"</span><span class="p">)</span></code></pre></figure>

<p>The function <code class="highlighter-rouge">calcNormFactors</code> is a <a href="/guide/primers/functional_analysis/rna_sequencing_analysis/#normalization" target="_blank">normalization procedure</a> using the trimmed mean of M-values (TMM) approach. The reference sample can be specified as the parameter <code class="highlighter-rouge">refColumn</code> otherwise the library whose upper quartile is closest to the mean upper quartile is used.</p>

<table class="table table-hover table-condensed table-responsive">
  <thead>
    <tr>
      <th style="text-align: center"> </th>
      <th style="text-align: center">group</th>
      <th style="text-align: center">lib.size</th>
      <th style="text-align: center">norm.factors</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><strong>TCGA-04-1357-01A-01R-1565-13</strong></td>
      <td style="text-align: center">Immunoreactive</td>
      <td style="text-align: center">37476326</td>
      <td style="text-align: center">0.983</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>TCGA-61-2000-01A-01R-1568-13</strong></td>
      <td style="text-align: center">Immunoreactive</td>
      <td style="text-align: center">56796486</td>
      <td style="text-align: center">0.9858</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>TCGA-31-1953-01A-01R-1568-13</strong></td>
      <td style="text-align: center">Immunoreactive</td>
      <td style="text-align: center">66315291</td>
      <td style="text-align: center">1.106</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>TCGA-25-1628-01A-01R-1566-13</strong></td>
      <td style="text-align: center">Mesenchymal</td>
      <td style="text-align: center">73974514</td>
      <td style="text-align: center">1.153</td>
    </tr>
  </tbody>
</table>

<p>The column <code class="highlighter-rouge">norm.factors</code> is simply our global correction factor for each library <script type="math/tex">k</script> relative to the reference <script type="math/tex">r</script>.</p>

<script type="math/tex; mode=display">% <![CDATA[
\begin{equation}
  \begin{split}
    f_k^r &= \frac{S_k}{S_{r}}
  \end{split}
\end{equation} %]]></script>

<p>Recall from our discussion on normalization that <script type="math/tex">S_k</script> represents our total RNA output whose increase will lead to under-sampling genes via RNA-seq and <em>vice versa</em>. The ‘effective library size’ replaces <code class="highlighter-rouge">lib.size</code> for each case and is computed by simply multiplying by <code class="highlighter-rouge">norm.factors</code>. This normalized result is used in downstream, differential expression testing and is a ‘model-based’ normalization in contrast to those that directly transform the raw data.</p>

<p><strong>Step 3: Fit</strong></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r">  
<span class="c1">### ============ 3. Fit ===============
</span><span class="n">TCGAOV_fitted_commondisp</span> <span class="o">=</span> <span class="n">estimateCommonDisp</span><span class="p">(</span><span class="n">TCGAOV_normalized_TMM</span><span class="p">)</span>
<span class="n">TCGAOV_fitted_tagwise</span> <span class="o">=</span> <span class="n">estimateTagwiseDisp</span><span class="p">(</span><span class="n">TCGAOV_fitted_commondisp</span><span class="p">)</span></code></pre></figure>

<p>Here we’re attempting to derive a squared biological coefficient of variation (<script type="math/tex">\phi</script>) from the data in order to parametrize our negative binomial model which we’ll use in DE testing. The function <code class="highlighter-rouge">estimateCommonDisp</code> estimates the dispersion across all genes and adds the value as <code class="highlighter-rouge">common.dispersion</code> in DGEList.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">names</span><span class="p">(</span><span class="n">TCGAOV_fitted_commondisp</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## [1] "counts"            "samples"           "genes"            
## [4] "common.dispersion" "pseudo.counts"     "pseudo.lib.size"  
## [7] "AveLogCPM"</code></pre></figure>

<p><code class="highlighter-rouge">estimateTagwiseDisp</code> estimates the dispersion for each gene and adds the list <code class="highlighter-rouge">tagwise.dispersion</code> to DGEList.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">names</span><span class="p">(</span><span class="n">TCGAOV_fitted_tagwise</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## [1] "counts"             "samples"            "genes"             
## [4] "common.dispersion"  "pseudo.counts"      "pseudo.lib.size"   
## [7] "AveLogCPM"          "prior.n"            "tagwise.dispersion"</code></pre></figure>

<p>Let us take a look at the data we’ve generated. Below we plot the common dispersion (red) and per-gene dispersions estimates. Next up are the variances compared to those expected with a Poisson model (line) demonstrating the inflation due to biological sources.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1">### ============ Plotting ===============
</span><span class="n">plotBCV</span><span class="p">(</span><span class="n">TCGAOV_fitted_tagwise</span><span class="p">)</span></code></pre></figure>

<p><img src="/guide/media/workflows/pathway_enrichment/process_data/unnamed-chunk-6-1.png" title="plot of chunk unnamed-chunk-6" alt="plot of chunk unnamed-chunk-6" width="400" style="display: block; margin: auto;" /></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">plotMeanVar</span><span class="p">(</span><span class="n">TCGAOV_fitted_tagwise</span><span class="p">,</span> <span class="n">show.tagwise.vars</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span> <span class="n">NBline</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span></code></pre></figure>

<p><img src="/guide/media/workflows/pathway_enrichment/process_data/unnamed-chunk-6-2.png" title="plot of chunk unnamed-chunk-6" alt="plot of chunk unnamed-chunk-6" width="400" style="display: block; margin: auto;" /></p>

<p>A negative binomial model can be fit from our data and dispersion estimated. From this, we calculate p-values <script type="math/tex">P</script> for each gene. As described in our discussion of <a href="/guide/primers/functional_analysis/rna_sequencing_analysis/#differentialExpression" target="_blank">differential expression testing</a>, <script type="math/tex">P</script> represents the sum of all probabilities less than or equal to the probability under the null hypothesis for the observed count.</p>

<p><strong>Step 4: Test</strong></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r">  <span class="c1">### ============ 4. Test ===============
</span><span class="n">TCGAOV_DE</span> <span class="o">=</span> <span class="n">exactTest</span><span class="p">(</span><span class="n">TCGAOV_fitted_tagwise</span><span class="p">,</span>
                      <span class="n">pair</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="n">CATEGORY_BASELINE</span><span class="p">,</span> <span class="n">CATEGORY_TEST</span><span class="p">))</span></code></pre></figure>

<ul>
  <li class="aside">
    <h4 id="note-on-edgerexacttestobject-pair">Note on <code class="highlighter-rouge">edgeR::exactTest(object, pair=...)</code></h4>

    <p><strong>pair:</strong> Vector of length two, either numeric or character, providing the pair of groups to be compared; if a character vector, then should be the names of two groups (e.g. two levels of object$samples$group); if numeric, then groups to be compared are chosen by finding the levels of object$samples$group corresponding to those numeric values and using those levels as the groups to be compared; if NULL, then first two levels of object$samples$group (a factor) are used. Note that the first group listed in the pair is the baseline for the comparison—so if the pair is c(“A”,”B”) then the comparison is B - A, so genes with positive log-fold change are up-regulated in group B compared with group A (and vice versa for genes with negative log-fold change).</p>
  </li>
</ul>

<p>The result of the function <code class="highlighter-rouge">exactTest</code> is a data structure with a <code class="highlighter-rouge">table</code> attribute that stores the p-values for each gene.</p>

<table class="table table-hover table-condensed table-responsive">
  <thead>
    <tr>
      <th style="text-align: center"> </th>
      <th style="text-align: center">logFC</th>
      <th style="text-align: center">logCPM</th>
      <th style="text-align: center">PValue</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><strong>ENSG00000000003</strong></td>
      <td style="text-align: center">-0.02072</td>
      <td style="text-align: center">6.285</td>
      <td style="text-align: center">0.8293</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>ENSG00000000419</strong></td>
      <td style="text-align: center">-0.08461</td>
      <td style="text-align: center">5.877</td>
      <td style="text-align: center">0.3485</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>ENSG00000000457</strong></td>
      <td style="text-align: center">-0.07762</td>
      <td style="text-align: center">3.88</td>
      <td style="text-align: center">0.2453</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>ENSG00000000460</strong></td>
      <td style="text-align: center">-0.2578</td>
      <td style="text-align: center">3.483</td>
      <td style="text-align: center">0.004774</td>
    </tr>
  </tbody>
</table>

<p><strong>Step 5: Adjust</strong></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r">  
<span class="c1">### ============ 5. Adjust ===============
</span><span class="n">TCGAOV_TT</span> <span class="o">=</span> <span class="n">topTags</span><span class="p">(</span><span class="n">TCGAOV_DE</span><span class="p">,</span>
                    <span class="n">n</span><span class="o">=</span><span class="n">nrow</span><span class="p">(</span><span class="n">TCGAOV_filtered</span><span class="p">),</span>
                    <span class="n">adjust.method</span><span class="o">=</span><span class="s2">"BH"</span><span class="p">,</span>
                    <span class="n">sort.by</span><span class="o">=</span><span class="s2">"PValue"</span><span class="p">)</span></code></pre></figure>

<p>The function <code class="highlighter-rouge">topTags</code> takes the output from <code class="highlighter-rouge">exactTest</code> and uses the <a href="/guide/primers/functional_analysis/multiple_testing/#controllingFDR" target="_blank">Bejamini-Hochberg (BH) procedure</a> to adjust the p-values yielding the a ‘BH-adjusted p-value’ also known as ‘q-value’ (Yekutieli and Benjamini, J. Stat. Plan. Inf. v82, pp.171-196, 1999). In terms of the BH procedure, the BH-adjusted p-value is the smallest value of <script type="math/tex">q^∗</script> for which the hypothesis corresponding to the p-value is still rejected. In practical terms, it means that values smaller than or equal to the given p-value have a false discovery rate equal to the BH-adjusted p-value. <code class="highlighter-rouge">topTags</code> returns the top differentially expressed genes. The output is similar to that of <code class="highlighter-rouge">exactTest</code> but with a column of adjusted p-values and sorted by increasing p-value.</p>

<table class="table table-hover table-condensed table-responsive">
  <thead>
    <tr>
      <th style="text-align: center"> </th>
      <th style="text-align: center">logFC</th>
      <th style="text-align: center">logCPM</th>
      <th style="text-align: center">PValue</th>
      <th style="text-align: center">FDR</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><strong>ENSG00000182492</strong></td>
      <td style="text-align: center">1.865</td>
      <td style="text-align: center">9.44</td>
      <td style="text-align: center">1.739e-52</td>
      <td style="text-align: center">1.841e-48</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>ENSG00000106624</strong></td>
      <td style="text-align: center">2.148</td>
      <td style="text-align: center">9.224</td>
      <td style="text-align: center">1.308e-47</td>
      <td style="text-align: center">6.927e-44</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>ENSG00000038427</strong></td>
      <td style="text-align: center">2.04</td>
      <td style="text-align: center">7.567</td>
      <td style="text-align: center">1.887e-43</td>
      <td style="text-align: center">6.66e-40</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>ENSG00000084636</strong></td>
      <td style="text-align: center">1.779</td>
      <td style="text-align: center">5.911</td>
      <td style="text-align: center">3.297e-43</td>
      <td style="text-align: center">8.728e-40</td>
    </tr>
  </tbody>
</table>

<p>We can now plot our differentially expressed genes (red) over our full data.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="c1">### ============ Plotting ===============
</span><span class="n">rn</span> <span class="o">=</span> <span class="n">rownames</span><span class="p">(</span><span class="n">TCGAOV_TT</span><span class="o">$</span><span class="n">table</span><span class="p">)</span>
<span class="n">deg</span> <span class="o">=</span><span class="n">rn</span><span class="p">[</span><span class="n">TCGAOV_TT</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">FDR</span><span class="o">&lt;</span><span class="m">0.05</span><span class="p">]</span>
<span class="n">plotSmear</span><span class="p">(</span><span class="n">TCGAOV_filtered</span><span class="p">,</span> <span class="n">pair</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="n">CATEGORY_BASELINE</span><span class="p">,</span> <span class="n">CATEGORY_TEST</span><span class="p">),</span> <span class="n">de.tags</span><span class="o">=</span><span class="n">deg</span><span class="p">)</span></code></pre></figure>

<p><img src="/guide/media/workflows/pathway_enrichment/process_data/unnamed-chunk-9-1.png" title="plot of chunk unnamed-chunk-9" alt="plot of chunk unnamed-chunk-9" width="500" style="display: block; margin: auto;" /></p>

<p><strong>Step 6: Rank</strong></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r">  
<span class="c1">### ============ 6. Rank ===============
</span><span class="n">TCGAOV_ranks</span> <span class="o">&lt;-</span> <span class="n">sign</span><span class="p">(</span><span class="n">TCGAOV_TT</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">logFC</span><span class="p">)</span> <span class="o">*</span> <span class="p">(</span><span class="m">-1</span><span class="p">)</span> <span class="o">*</span> <span class="n">log10</span><span class="p">(</span><span class="n">TCGAOV_TT</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">PValue</span><span class="p">)</span>
<span class="n">genenames</span> <span class="o">&lt;-</span> <span class="n">TCGAOV_TT</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">external_gene_name</span>
<span class="n">TCGAOV_ranks</span> <span class="o">&lt;-</span> <span class="n">data.frame</span><span class="p">(</span><span class="n">gene</span><span class="o">=</span><span class="n">genenames</span><span class="p">,</span> <span class="n">rank</span><span class="o">=</span><span class="n">TCGAOV_ranks</span><span class="p">)</span>
<span class="n">TCGAOV_ordered_ranks</span> <span class="o">&lt;-</span> <span class="n">TCGAOV_ranks</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">TCGAOV_ranks</span><span class="p">[,</span><span class="m">2</span><span class="p">],</span> <span class="n">decreasing</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),]</span></code></pre></figure>

<p>The rank of each gene is inversely proportional to the log of the <script type="math/tex">P</script> as smaller values are less likely under the null hypothesis.
Set the gene name from the Ensembl gene ID to the <code class="highlighter-rouge">external_gene_name</code> which is compatible with the HGNC namespace.</p>

<p>The resulting data frame is saved as a tab-delimited file <code class="highlighter-rouge">MesenchymalvsImmunoreactive_edger_ranks.rnk</code> for use in downstream differential expression analysis.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r">  
<span class="c1">## Save to file as tab-delimited text
</span><span class="n">write.table</span><span class="p">(</span><span class="n">TCGAOV_ordered_ranks</span><span class="p">,</span>
            <span class="n">TCGAOV_RANKS_OUTPUT_FILE</span><span class="p">,</span>
            <span class="n">col.name</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span>
            <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">,</span>
            <span class="n">row.names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span>
            <span class="n">quote</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span></code></pre></figure>

<p>Your directory should now contain the rank file.</p>

<div class="language-shell highlighter-rouge"><pre class="highlight"><code>Documents
|
|--- data
    |
    |--- TCGA
        |
        |--- MesenchymalvsImmunoreactive_edger_ranks.rnk
        |
        |--- TCGAOV_data.rda
        |
        |--- Verhaak_JCI_2013_tableS1.txt
        |
        |--- TCGA-OV
            |--- ...
...
</code></pre>
</div>

<hr />

<p>The preceding R code is presented in its entirety and available at Github
<a href="https://github.com/jvwong/docker_enrichment_workflow_gdc" target="_blank">
  <i class="fa fa-github fa-2x"></i>
</a></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r">  <span class="n">rm</span><span class="p">(</span><span class="n">list</span><span class="o">=</span><span class="n">ls</span><span class="p">(</span><span class="n">all</span><span class="o">=</span><span class="n">TRUE</span><span class="p">))</span>

<span class="c1">### ============ Install packages from Bioconductor ========
</span><span class="n">library</span><span class="p">(</span><span class="s2">"edgeR"</span><span class="p">)</span>


<span class="c1">### ============ Declare directories =========
</span><span class="n">BASE_DIR</span> <span class="o">=</span> <span class="s2">"/home/TCGA"</span> <span class="c1"># same as volumes in docker-compose.yml
</span><span class="n">TCGAOV_DATA_DIR</span> <span class="o">=</span> <span class="n">file.path</span><span class="p">(</span><span class="n">BASE_DIR</span><span class="p">,</span> <span class="s2">"data"</span><span class="p">)</span>
<span class="n">TCGAOV_OUTPUT_DIR</span> <span class="o">=</span> <span class="n">file.path</span><span class="p">(</span><span class="n">BASE_DIR</span><span class="p">,</span> <span class="s2">"output"</span><span class="p">)</span>

<span class="n">TCGAOV_DGE_OUTPUT_FILE</span> <span class="o">=</span> <span class="n">file.path</span><span class="p">(</span><span class="n">TCGAOV_OUTPUT_DIR</span><span class="p">,</span>
                                  <span class="s2">"TCGAOV_data.rda"</span><span class="p">)</span>
<span class="n">TCGAOV_RANKS_OUTPUT_FILE</span> <span class="o">=</span> <span class="n">file.path</span><span class="p">(</span><span class="n">TCGAOV_OUTPUT_DIR</span><span class="p">,</span>
                 <span class="s2">"MesenchymalvsImmunoreactive_edger_ranks.rnk"</span><span class="p">)</span>

<span class="c1">### ============ Declare samples =========
</span><span class="n">CATEGORY_BASELINE</span> <span class="o">=</span> <span class="s2">"Immunoreactive"</span>
<span class="n">CATEGORY_TEST</span> <span class="o">=</span> <span class="s2">"Mesenchymal"</span>

<span class="n">load</span><span class="p">(</span><span class="n">TCGAOV_DGE_OUTPUT_FILE</span><span class="p">)</span>


<span class="c1">### ============ 1. Filter ===============
</span><span class="n">index_BASELINE</span> <span class="o">=</span> <span class="n">TCGAOV_data</span><span class="o">$</span><span class="n">samples</span><span class="o">$</span><span class="n">group</span> <span class="o">==</span> <span class="n">CATEGORY_BASELINE</span>
<span class="n">index_TEST</span> <span class="o">=</span> <span class="n">TCGAOV_data</span><span class="o">$</span><span class="n">samples</span><span class="o">$</span><span class="n">group</span> <span class="o">==</span> <span class="n">CATEGORY_TEST</span>
<span class="n">index_BASELINE_TEST</span> <span class="o">=</span> <span class="n">which</span><span class="p">(</span><span class="n">index_BASELINE</span> <span class="o">|</span> <span class="n">index_TEST</span><span class="p">)</span>
<span class="n">N_BASELINE</span> <span class="o">=</span> <span class="n">sum</span><span class="p">(</span><span class="n">index_BASELINE</span><span class="p">)</span>
<span class="n">N_TEST</span> <span class="o">=</span> <span class="n">sum</span><span class="p">(</span><span class="n">index_TEST</span><span class="p">)</span>
<span class="n">TCGAOV_data_BT</span> <span class="o">=</span> <span class="n">TCGAOV_data</span><span class="p">[,</span><span class="n">index_BASELINE_TEST</span><span class="p">]</span>

<span class="n">row_with_mincount</span> <span class="o">=</span> <span class="n">rowSums</span><span class="p">(</span><span class="n">cpm</span><span class="p">(</span><span class="n">TCGAOV_data_BT</span><span class="p">)</span> <span class="o">&gt;</span> <span class="m">10</span><span class="p">)</span> <span class="o">&gt;=</span> <span class="n">min</span><span class="p">(</span><span class="n">N_BASELINE</span><span class="p">,</span>
                                                             <span class="n">N_TEST</span><span class="p">)</span>
<span class="n">TCGAOV_filtered</span> <span class="o">=</span> <span class="n">TCGAOV_data_BT</span><span class="p">[</span><span class="n">row_with_mincount</span><span class="p">,</span> <span class="p">,</span> <span class="n">keep.lib.sizes</span><span class="o">=</span><span class="n">FALSE</span><span class="p">]</span>


<span class="c1">### ============ 2. Normalize ===============
</span><span class="n">TCGAOV_normalized_TMM</span> <span class="o">=</span> <span class="n">calcNormFactors</span><span class="p">(</span><span class="n">TCGAOV_filtered</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s2">"TMM"</span><span class="p">)</span>


<span class="c1">### ============ 3. Fit ===============
</span><span class="n">TCGAOV_fitted_commondisp</span> <span class="o">=</span> <span class="n">estimateCommonDisp</span><span class="p">(</span><span class="n">TCGAOV_normalized_TMM</span><span class="p">)</span>
<span class="n">TCGAOV_fitted_tagwise</span> <span class="o">=</span> <span class="n">estimateTagwiseDisp</span><span class="p">(</span><span class="n">TCGAOV_fitted_commondisp</span><span class="p">)</span>

<span class="c1">###  Plotting BCV
</span><span class="n">plotBCV</span><span class="p">(</span><span class="n">TCGAOV_fitted_tagwise</span><span class="p">)</span>
<span class="n">plotMeanVar</span><span class="p">(</span><span class="n">TCGAOV_fitted_tagwise</span><span class="p">,</span> <span class="n">show.tagwise.vars</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span> <span class="n">NBline</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span>

<span class="c1">### ============ 4. Test ===============
</span><span class="n">TCGAOV_DE</span> <span class="o">=</span> <span class="n">exactTest</span><span class="p">(</span><span class="n">TCGAOV_fitted_tagwise</span><span class="p">,</span>
                      <span class="n">pair</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="n">CATEGORY_BASELINE</span><span class="p">,</span> <span class="n">CATEGORY_TEST</span><span class="p">))</span>


<span class="c1">### ============ 5. Adjust ===============
</span><span class="n">TCGAOV_TT</span> <span class="o">=</span> <span class="n">topTags</span><span class="p">(</span><span class="n">TCGAOV_DE</span><span class="p">,</span>
                    <span class="n">n</span><span class="o">=</span><span class="n">nrow</span><span class="p">(</span><span class="n">TCGAOV_filtered</span><span class="p">),</span>
                    <span class="n">adjust.method</span><span class="o">=</span><span class="s2">"BH"</span><span class="p">,</span>
                    <span class="n">sort.by</span><span class="o">=</span><span class="s2">"PValue"</span><span class="p">)</span>

<span class="c1">###  Plotting
</span><span class="n">rn</span> <span class="o">=</span> <span class="n">rownames</span><span class="p">(</span><span class="n">TCGAOV_TT</span><span class="o">$</span><span class="n">table</span><span class="p">)</span>
<span class="n">deg</span> <span class="o">=</span><span class="n">rn</span><span class="p">[</span><span class="n">TCGAOV_TT</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">FDR</span><span class="o">&lt;</span><span class="m">0.05</span><span class="p">]</span>
<span class="n">plotSmear</span><span class="p">(</span><span class="n">TCGAOV_filtered</span><span class="p">,</span> <span class="n">pair</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="n">CATEGORY_BASELINE</span><span class="p">,</span> <span class="n">CATEGORY_TEST</span><span class="p">),</span> <span class="n">de.tags</span><span class="o">=</span><span class="n">deg</span><span class="p">)</span>


<span class="c1">### ============ 6. Rank ===============
</span><span class="n">TCGAOV_ranks</span> <span class="o">&lt;-</span> <span class="n">sign</span><span class="p">(</span><span class="n">TCGAOV_TT</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">logFC</span><span class="p">)</span> <span class="o">*</span> <span class="p">(</span><span class="m">-1</span><span class="p">)</span> <span class="o">*</span> <span class="n">log10</span><span class="p">(</span><span class="n">TCGAOV_TT</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">PValue</span><span class="p">)</span>
<span class="n">genenames</span> <span class="o">&lt;-</span> <span class="n">TCGAOV_TT</span><span class="o">$</span><span class="n">table</span><span class="o">$</span><span class="n">external_gene_name</span>
<span class="n">TCGAOV_ranks</span> <span class="o">&lt;-</span> <span class="n">data.frame</span><span class="p">(</span><span class="n">gene</span><span class="o">=</span><span class="n">genenames</span><span class="p">,</span> <span class="n">rank</span><span class="o">=</span><span class="n">TCGAOV_ranks</span><span class="p">)</span>
<span class="n">TCGAOV_ordered_ranks</span> <span class="o">&lt;-</span> <span class="n">TCGAOV_ranks</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">TCGAOV_ranks</span><span class="p">[,</span><span class="m">2</span><span class="p">],</span> <span class="n">decreasing</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),]</span>


<span class="c1">## Save to file as tab-delimited text
</span><span class="n">write.table</span><span class="p">(</span><span class="n">TCGAOV_ordered_ranks</span><span class="p">,</span>
            <span class="n">TCGAOV_RANKS_OUTPUT_FILE</span><span class="p">,</span>
            <span class="n">col.name</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span>
            <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">,</span>
            <span class="n">row.names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span>
            <span class="n">quote</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span></code></pre></figure>

<h2 id="a-hrefdata-namedataiv-dataa"><a href="#data" name="data">IV. Data</a></h2>

<p>Gene list ranked by differential gene expression between ‘Mesenchymal’ vs ‘Immunoreactive’ TCGA HGS-OvCa subtypes. This data is saved in tab-delimited format to a file <code class="highlighter-rouge">MesenchymalvsImmunoreactive_edger_ranks.rnk</code>.
<a href="/guide/media/workflows/pathway_enrichment/process_data/MesenchymalvsImmunoreactive_edger_ranks.rnk" type="button" class="btn btn-success btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Ranked list (.rnk)</a></p>

<table class="table table-hover table-condensed table-responsive">
  <thead>
    <tr>
      <th style="text-align: center"> </th>
      <th style="text-align: center">gene</th>
      <th style="text-align: center">rank</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center">1</td>
      <td style="text-align: center">BGN</td>
      <td style="text-align: center">51.7598126672467</td>
    </tr>
    <tr>
      <td style="text-align: center">2</td>
      <td style="text-align: center">AEBP1</td>
      <td style="text-align: center">46.8833012028318</td>
    </tr>
    <tr>
      <td style="text-align: center">3</td>
      <td style="text-align: center">VCAN</td>
      <td style="text-align: center">42.7242756718847</td>
    </tr>
    <tr>
      <td style="text-align: center">4</td>
      <td style="text-align: center">COL16A1</td>
      <td style="text-align: center">42.4819173450002</td>
    </tr>
    <tr>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
    </tr>
    <tr>
      <td style="text-align: center">10589</td>
      <td style="text-align: center">PSMB9</td>
      <td style="text-align: center">-22.3106957916029</td>
    </tr>
    <tr>
      <td style="text-align: center">10590</td>
      <td style="text-align: center">PSMB8-AS1</td>
      <td style="text-align: center">-23.244789795428</td>
    </tr>
    <tr>
      <td style="text-align: center">10591</td>
      <td style="text-align: center">TAP1</td>
      <td style="text-align: center">-26.92687699671</td>
    </tr>
  </tbody>
</table>

<p>The R code is available at Github <a href="https://github.com/jvwong/docker_enrichment_workflow_gdc" target="_blank">
  <i class="fa fa-github fa-2x"></i>
</a></p>

<hr />

<h2 id="a-hrefreferences-namereferencesv-referencesa"><a href="#references" name="references">V. References</a></h2>
<div class="panel_group" data-inline="23975260,21720365,23359318"></div>
<!-- - Anders S et al. Count-based differential expression analysis of RNA sequencing data using R and Bioconductor. Nat Protoc vol. 8 (2013)
- Cancer Genome Atlas Research Network. Integrated genomic analyses of ovarian carcinoma. Nature vol. 474 (2011)
- Hout M et al. Multidimensional scaling. Wiley Interdiscip Rev Cogn Sci vol. 4 (2013) -->


        </div>
      </div>
    </div>

  </body>
</html>
