<!DOCTYPE html>

<html lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Visualize &middot; Guide
    
  </title>

  <!-- Icons -->
  <link rel="shortcut icon" href="/guide/media/favicon.ico">

  <!-- CSS -->
  
  <!-- Third-party CSS -->
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/bootstrap/dist/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/progress-tracker/app/styles/progress-tracker.css" />
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/font-awesome/css/font-awesome.min.css" />

  <!-- Custom CSS -->
  <link rel="stylesheet" href="/guide/public/css/main.css">

  <!-- Javascript -->
  <!-- Third-party javascript -->
  <script type="text/javascript" src="/guide/public/bower_components/jquery/dist/jquery.min.js"></script>
  <script type="text/javascript" src="/guide/public/bower_components/bootstrap/dist/js/bootstrap.min.js"></script>
  <script type="text/javascript" src="/guide/public/bower_components/cytoscape/dist/cytoscape.min.js"></script>

  <script type="text/javascript" src="/guide/public/js/deps.js"></script>
  <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
    MathJax.Hub.Config({ TeX: { equationNumbers: { autoNumber: "AMS" } } });
  </script>
  <!-- Custom javascript -->
  <script type="text/javascript" src="/guide/public/js/babel-compiled.js" defer></script>
  
</head>


  <body class="theme-base-01">
    <div id="embedded">
    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
           content to avoid any CSS collisions with our real content. -->
      <div class="wrap">
        <div class="container content embedded">
          <div id='ajax-spinner'><img src="/guide/media/spinner.gif"/></div>

            <h1 class="page-title">Visualize</h1>
            <div class="subtitle">Create an Enrichment Map displaying the landscape of pathways</div>
            <hr/>
            <div class="linkout clearfix">
              <ul class="list-inline pull-right">
              <li>
  <a href="http://www.facebook.com/sharer/sharer.php?u=/workflows/pathway_enrichment_ii/visualize/&title=Visualize"
  data-toggle="tooltip"
  title="Share this page on Facebook"
  target="_blank" >
    <i class="fa fa-facebook-official fa-2x"></i>
  </a>
</li>
<li>
  <a href="http://twitter.com/intent/tweet?text=Create an Enrichment Map displaying the landscape of pathways+/workflows/pathway_enrichment_ii/visualize/&via=pathwaycommons"
  data-toggle="tooltip"
  title="Tweet this page"
  target="_blank" >
    <i class="fa fa-twitter-square fa-2x"></i>
  </a>
</li>
<li>
  <a href="https://plus.google.com/share?url=/workflows/pathway_enrichment_ii/visualize/"
  data-toggle="tooltip"
  title="Share on Google+"
  target="_blank" >
    <i class="fa fa-google-plus fa-2x"></i>
  </a>
</li>

              </ul>
            </div>
            <hr/>
            <ul>
  <li class="list-unstyled">Table of Contents
    <ul>
      <li class="list-unstyled"><a href="#goals">I. Goals</a></li>
      <li class="list-unstyled"><a href="#background">II. Background</a></li>
      <li class="list-unstyled"><a href="#workflow_step">III. Workflow Step</a></li>
      <li class="list-unstyled"><a href="#interpretation">IV. Interpretation</a></li>
    </ul>
  </li>
</ul>

<hr />

<h2 id="a-hrefgoals-namegoalsi-goalsa"><a href="#goals" name="goals">I. Goals</a></h2>

<p><a href="/guide/workflows/pathway_enrichment_ii/identify_pathways/">Previously</a>, we passed a rank list of differentially expressed genes between platelets of healthy donors and breast cancer patients along with a gene set database to the Gene Set Enrichment Analysis (GSEA) software in hopes of identifying significantly altered pathways. We could have simply ended there and examined our GSEA reports for those pathways in the list that peaked our scientific interest. In many instances, however, the results of GSEA are voluminous in part because of a high degree of redundancy amongst pathways. Can we reduce this redundancy and provide a more meaningful, interactive means to reason over the results?</p>

<p>In this workflow step, we describe Enrichment Map (EM), a visual tool that simplifies the display of a list of gene sets with a large degree of redundancy. By then end of this discussion you should:</p>

<ol>
  <li>Understand how an Enrichment Map can simplify enrichment results</li>
  <li>Learn how to create and tune an Enrichment Map</li>
  <li>Be able to generate a figure describing the landscape of altered pathways</li>
</ol>

<h2 id="a-hrefbackground-namebackgroundii-backgrounda"><a href="#background" name="background">II. Background</a></h2>

<p>EM was originally described by Merico <em>et al.</em> (Merico 2010) as an aid in the interpretation of gene sets emerging from enrichment analyses. The motivation for this tool is the growing number of gene sets and gene annotation detail being made available, which leads to large numbers of gene sets emerging from enrichment analyses. Even with stringent criteria, it is not uncommon for analyses to generate hundreds or thousands of pathways, making results difficult to interpret on whole. To illustrate the value of EM, let us return to our analysis of alterations in pathways of platelets by Best <em>et al.</em>.</p>

<h3 id="a-long-list-of-pathways">A long list of pathways</h3>

<p><a href="/guide/workflows/pathway_enrichment_ii/identify_pathways/">Previously</a>, we used GSEA to distill gene sets enriched in the underlying differences in RNA abundance between platelets from healthy donors (HD) and individuals diagnosed with breast cancer (BrCa). Our rationale for applying enrichment analyis was that we would much rather reason at the pathway-level than the gene-level. Figure 1 shows a section of the GSEA report and Table 1 shows an excerpt of the actual enrichment report for BrCa.</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_gsea_report.png" alt="image" class="img-responsive slim" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 1. GSEA report for BrCa and HD phenotypes.</strong> An excerpt of the report taken from the results of <a href="/guide/workflows/pathway_enrichment_ii/identify_pathways/">Identify Pathways</a>.
</div>

<p><strong>Table 1. Enrichment report for BrCa (na_pos phenotype)</strong></p>

<table class="table table-hover table-condensed table-responsive">
  <thead>
    <tr>
      <th style="text-align: left">NAME</th>
      <th style="text-align: center">SIZE</th>
      <th style="text-align: center">ES</th>
      <th style="text-align: center">NES</th>
      <th style="text-align: center">NOM p-val</th>
      <th style="text-align: center">FDR q-val</th>
      <th style="text-align: center">FWER p-val</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">PID_FAK_PATHWAY</td>
      <td style="text-align: center">48</td>
      <td style="text-align: center">0.8144367</td>
      <td style="text-align: center">2.2033582</td>
      <td style="text-align: center">0.0</td>
      <td style="text-align: center">0.0</td>
      <td style="text-align: center">0.0</td>
    </tr>
    <tr>
      <td style="text-align: left">THROMBOXANE A2 RECEPTOR SIGNALING</td>
      <td style="text-align: center">39</td>
      <td style="text-align: center">0.84532094</td>
      <td style="text-align: center">2.175933</td>
      <td style="text-align: center">0.0</td>
      <td style="text-align: center">0.0</td>
      <td style="text-align: center">0.0</td>
    </tr>
    <tr>
      <td style="text-align: left">SIGNALING EVENTS MEDIATED BY FOCAL ADHESION KINASE</td>
      <td style="text-align: center">48</td>
      <td style="text-align: center">0.8144367</td>
      <td style="text-align: center">2.16101</td>
      <td style="text-align: center">0.0</td>
      <td style="text-align: center">0.0</td>
      <td style="text-align: center">0.0</td>
    </tr>
    <tr>
      <td style="text-align: left">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
    </tr>
    <tr>
      <td style="text-align: left">IL5%NETPATH%IL5	IL5</td>
      <td style="text-align: center">49</td>
      <td style="text-align: center">0.7718686</td>
      <td style="text-align: center">2.0658088</td>
      <td style="text-align: center">0.0</td>
      <td style="text-align: center">0.0</td>
      <td style="text-align: center">0.0</td>
    </tr>
    <tr>
      <td style="text-align: left">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
    </tr>
  </tbody>
</table>

<p>Notice that even under fairly stringent criteria (i.e. p-value &lt; 1%), there are a still 578 gene sets that are deemed significantly enriched in BrCa platelets and 394 in HD. Furthermore, our GSEA report declares that 8 191 genes were used as input in our ranked list. Have we simply kicked the can down the road and traded a long list of ~8 200 genes for a long list of ~1 000 gene sets?</p>

<h3 id="data-reduction">Data reduction</h3>

<p>Ideally, we want a simple way to reduce the size of the data from an enrichment analysis while retaining, to the best of our ability, the amount of meaningful <em>information</em>. To this end, many approaches have set their sights on reducing the amount of <em>redundancy</em> amongst gene sets.</p>

<h4 id="hierachical">Hierachical</h4>

<p>Many enrichment anlyses use the <a href="http://geneontology.org/" target="_blank">Gene Ontology</a> (Ashburner 2000) which is hierarchically organized. In this case, redundancy can be reduced by merging gene sets that are children of a parent cluster with broader scope. These and other similar approaches rely on variations of the same statistical test for enrichment. Importantly, these approaches do not apply to gene sets without a clearly defined hierarchy, for example, transcriptional regulator target genes. What we desire is a versatile method able to manage non-hierarchical gene sets that emerge from analysis methods, regarldless of nature of the underlying enrichment test used (Khatri 2012).</p>

<h3 id="enrichment-map">Enrichment Map</h3>

<p>EM is a visualization analysis tool that organizes gene sets into an information-rich similarity network. The true power of Enrichment Map is that it is a visual display method that reduces complexity by grouping similar gene sets as defined by the number of overlapping genes. These clusters can then be annotated with a representative label gleaned from the characteristics of the individual gene sets (Figure 2).</p>

<p>Gene sets are represented as nodes whose radius is proportional to the number of genes. Edges indicate nodes with shared genes, where the thickness of the line is proportional to the degree of overlap. Finally, EM can use node color to represent other dimensions of the data, for example, gene sets enriched in different classes.</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_rationale.jpg" alt="image" class="img-responsive slim" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 2. Rationale of the Enrichment Map.</strong> Genes within gene sets or pathways can overlap and often describe nearly identical biological concepts. An Enrichment Map reduces the redundancy in gene sets by creating a graph where nodes are gene sets, node size correlates with gene set size, and edge widths indicate the number of shared genes. Note that gene set 'd' does not share any genes with the others. Highly similar gene sets can be labelled with a single, over-arching label.
</div>

<p>The approach is modular in that it is compatible with any statistical test or gene set source. Enrichment Map shines when dealing with a collection of gene sets with a large degree of redundancy.</p>

<h3 id="preview-of-em-software">Preview of EM software</h3>

<p>Figure 3 provides a tour of the <a href="http://apps.cytoscape.org/apps/enrichmentmap" target="_blank">EM app</a> in the <a href="http://www.cytoscape.org/" target="_blank">Cytoscape</a> software.</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_preview.jpg" alt="img" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 3. Tour of the Enrichment Map app.</strong> The Enrichment Map app is accessed through Cytoscape. (A) The main window displays the gene sets (circles) and their similarity (edges). Gene set size is proportional to node radius and overlap is indicated by edge thickness. Gene sets associated with BrCa and HD classes are shaded red and blue, respectively. (B) A <code>Control Panel</code> is where settings for Cytoscape and each app appear. (C) The <code>Table Panel</code> displays data for each node and edge along with expression heatmaps. (D) The <code>Results Panel</code> is where settings delcared in the <code>Control Panel</code> can be fine-tuned. Note that the <a href="http://www.netpath.org/pathways?path_id=NetPath_17" target="_blank">NetPath IL-5 signalling pathway</a> has been highlighted in the main window (yellow node, top right) and the corresponding gene expression is displayed in the <code>Table panel</code>.
  <div class="text-left">
    <a type="button" class="btn btn-default" href="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_preview.jpg" target="_blank">Open in new window</a>
  </div>
</div>

<p>It should be immediately apparent from Figure 3 that the Enrichment Map app provides several helpful features including:</p>

<ul>
  <li class="list-unstyled"><strong>View underlying gene expression</strong>
    <ul>
      <li>This data originates from the <a href="/guide/workflows/pathway_enrichment_ii/process_data/#output" target="_blank">expression file</a> provided in workflow step 1: Process Data</li>
      <li>The ‘leading edge’ genes are those that played a prominant role in the GSEA signal and can be viewed alongside the heatmap when the <a href="/guide/workflows/pathway_enrichment_ii/process_data/#output" target="_blank">rank file</a> is available. We will load this file but won’t discuss it further.</li>
    </ul>
  </li>
  <li class="list-unstyled"><strong>Group similar gene sets</strong>
    <ul>
      <li>This data originates from the <a href="/guide/workflows/pathway_enrichment_ii/identify_pathways/#output" target="_blank">enrichment reports</a> generated in workflow step 2: Identify Pathways</li>
    </ul>
  </li>
  <li class="list-unstyled"><strong>Color classes/phenotypes</strong>
    <ul>
      <li>These assignments originate from the <a href="/guide/workflows/pathway_enrichment_ii/process_data/#output" target="_blank">phenotype file</a> provided in workflow step 1: Process Data</li>
    </ul>
  </li>
</ul>

<div class="alert alert-info text-justify" role="alert">
  Take a moment to reflect on how the previous workflow step outputs are dependencies for the Enrichment Map. In particular, review your understanding of what the rank file, phenotype file, enrichment reports and rank file contain and how they were obtained. This will ease the process of creating and understanding what the Enrichment Map results mean.
</div>

<h2 id="a-hrefworkflowstep-nameworkflowstepiii-workflow-stepa"><a href="#workflow_step" name="workflow_step">III. Workflow Step</a></h2>

<p>Below, we decribe the process of loading our file dependencies from previous workflow steps (i.e. enrichment reports, expression, phenotypes, ranks) into the <a href="http://www.cytoscape.org/" target="_blank">Cytoscape</a> desktop software towards genereating an Enrichment Map similar to Figure 3. We will then provide some tips on how to navigate, tune and interpret the EM in an effort to facilitate new scientific insight.</p>

<h4 id="software-requirements">Software requirements</h4>

<ul>
  <li>Install <a href="http://www.oracle.com/technetwork/java/javase/downloads/jre8-downloads-2133155.html" target="_blank">Java version 8</a></li>
  <li><a href="http://www.cytoscape.org/" target="_blank">Cytoscape</a>: version 3.4.0
    <ul>
      <li>Cytoscape is an indispensable tool for network visualization and analysis. The desktop software comes as a base installation and additional capabilities are added in the form of a large ecosystem of plugins or <a href="http://apps.cytoscape.org/" target="_blank">apps</a>. Checkout the <a href="http://manual.cytoscape.org/en/stable/index.html" target="_blank">Cytoscape User Manual</a> for full description of function and capabilities.</li>
    </ul>
  </li>
  <li>Cytoscape apps
    <ul>
      <li><a href="http://apps.cytoscape.org/apps/enrichmentmap" target="_blank">Enrichment Map</a>: version 2.2.0</li>
      <li><a href="http://apps.cytoscape.org/apps/clustermaker2" target="_blank">ClusterMaker2</a>: version 0.9.5</li>
      <li><a href="http://apps.cytoscape.org/apps/wordcloud" target="_blank">WordCloud</a>: version 3.1.0</li>
      <li><a href="http://apps.cytoscape.org/apps/autoannotate" target="_blank">AutoAnnotate</a>: version 1.1.0</li>
    </ul>
  </li>
</ul>

<h5 id="loading-apps">Loading apps</h5>

<p>The base Cytoscape installation will use a collection of apps to generate the Enrichment Map. Apps are loaded into Cytoscape using the built-in App Manager.</p>

<p>From the Cytoscape menu bar, select <code class="highlighter-rouge">'Apps' --&gt; 'App Manager'</code> to bring up an app search panel (Figure 4).</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_cytoscape_apps_em.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 4. Cytoscape App Manager screen.</strong> Search for the app of choice using the <code>Install Apps</code> tab. Beware that the App Manager search engine will respect spaces in entered text (e.g. 'EnrichmentMap' will be found but not 'Enrichment Map').
  <div class="text-left">
    <a type="button" class="btn btn-info" href="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_cytoscape_apps_em.png" target="_blank">Open in new window</a>
  </div>
</div>

<p>Search for the app, highlight the correct search result and click <code class="highlighter-rouge">Install</code>. If the installation was successful, you should be able to select it by name in the <code class="highlighter-rouge">Apps</code> menu bar drop-down. Also, if you bring up the <code class="highlighter-rouge">App Manager</code> you should see it as a listing under the <code class="highlighter-rouge">Currently Installed</code> tab.</p>

<hr />

<p>Launch the Enrichment Map app by selecting from the menu bar <code class="highlighter-rouge">Apps --&gt; EnrichmentMap --&gt; Create Enrichment Map</code> (Figure 5).</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_splash.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 5. Enrichment Map screen.</strong> The <code>Control Panel</code> on the left is where we will load our data files and adjust the Enrichment Map settings.
  <div class="text-left">
    <a type="button" class="btn btn-info" href="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_splash.png" target="_blank">Open in new window</a>
  </div>
</div>

<h3 id="input">Input</h3>

<p>Below we provide the dependencies that will be loaded into the Enrichment Map app through the Cytoscape <code class="highlighter-rouge">Control Panel</code> (Figure 5, left). All of these files were either provided of generated in previous workflow steps.</p>

<h4 id="gmt-file">GMT File</h4>

<p><a href="/guide/media/workflows/pathway_enrichment_ii/identify_pathways/Human_GOBP_AllPathways_no_GO_iea_February_01_2017_symbol.gmt.zip" type="button" class="btn btn-info btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Gene Set Database (.gmt)</a></p>

<h4 id="expression">Expression</h4>

<p><a href="/guide/media/workflows/pathway_enrichment_ii/process_data/brca_hd_tep_expression.txt" type="button" class="btn btn-info btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Expression (.txt)</a></p>

<h4 id="enrichments">Enrichments</h4>

<p><a href="/guide/media/workflows/pathway_enrichment_ii/identify_pathways/gsea_report_for_brca.xls" type="button" class="btn btn-info btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> BrCa report (.xls)</a></p>

<p><a href="/guide/media/workflows/pathway_enrichment_ii/identify_pathways/gsea_report_for_hd.xls" type="button" class="btn btn-info btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> HD report (.xls)</a></p>

<h4 id="ranks">Ranks</h4>

<p><a href="/guide/media/workflows/pathway_enrichment_ii/process_data/brca_hd_tep_ranks.rnk" type="button" class="btn btn-info btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Ranks (.rnk)</a></p>

<h4 id="classes">Classes</h4>

<p><a href="/guide/media/workflows/pathway_enrichment_ii/process_data/brca_hd_tep_phenotype.cls" type="button" class="btn btn-info btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Phenotype (.cls)</a></p>

<h3 id="analysis">Analysis</h3>

<h4 id="load-dependencies">1. Load dependencies</h4>

<p>Having obtained the six (6) file dependencies you may now load these into the Cytoscape Enrichment Map app via the <code class="highlighter-rouge">Control Panel</code> (Figure 6).</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_load.png" alt="image" class="img-responsive" /></p>

<div class="figure-legend well well-lg text-justify">
  <strong>Figure 6. Enrichment Map file load.</strong> Shown is the <code>Control Panel</code> tab for the <code>Enrichment Map Input</code>. Load in the file dependencies for <code>GMT File</code>, <code>Expression</code>, <code>Enrichments 1/2</code>, <code>Ranks</code> and <code>Classes</code>.
</div>

<h4 id="configure">2. Configure</h4>

<p>In the <code class="highlighter-rouge">Control Panel</code> tab for Enrichment Map (Figure 6), make the following adjustments to the following sections.</p>

<ul>
  <li class="list-unstyled"><strong>Advanced</strong>
    <ul>
      <li>Replace the <code class="highlighter-rouge">Phenotypes</code> field with <code class="highlighter-rouge">BrCa VS. HD</code> to reflect the comparisons we made. This also matches the order in which you loaded files into the <code class="highlighter-rouge">Enrichments 1</code> and <code class="highlighter-rouge">Enrichments 2</code> fields.</li>
    </ul>
  </li>
  <li class="list-unstyled"><strong>Parameters</strong>
    <ul>
      <li>Set <code class="highlighter-rouge">FDR Q-value Cutoff</code> to 1E-04 in <code class="highlighter-rouge">Scientific Notation</code>.
        <ul>
          <li>This number is inversely proportional to the stringency for displaying a gene set.</li>
          <li>Smaller number – fewer nodes.</li>
        </ul>
      </li>
      <li>For <code class="highlighter-rouge">Similarity Cutoff</code>, click the radio for <code class="highlighter-rouge">Jaccard+Overlap Combined</code> and keep the <code class="highlighter-rouge">Cutoff</code> to 0.375.
        <ul>
          <li>This number is the lower-bound for showing edges, which represent overlap between gene sets.</li>
          <li>Larger number – fewer edges</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<blockquote>
  <p>You will be able to adjust the cutoff parameters later on in the <code class="highlighter-rouge">Results Panel</code> after the Enrichment Map is built and displayed</p>
</blockquote>

<h4 id="build">3. Build</h4>

<p>Click <code class="highlighter-rouge">Build</code> to generate the EM displayd in Figure 7. Take some time to examine main window displaying the network. In particular, in the lower right region of the main window there is a bird’s eye view showing the region displayed in the main view. Let us recap what we are seeing in the main view:</p>

<ul>
  <li>Node
    <ul>
      <li>Each circle represents a gene set
        <ul>
          <li>Select a node with your mouse and examine the related data in the <code class="highlighter-rouge">Table Panel</code> tab <code class="highlighter-rouge">Node Table</code></li>
        </ul>
      </li>
      <li>Size: Number of genes in the gene set</li>
      <li>Colour: Phenotype (class) assigment and enrichment (p-value)</li>
    </ul>
  </li>
  <li>Edge
    <ul>
      <li>Indicate related gene sets as measured by shared genes
        <ul>
          <li>Select an edge with your mouse and examine the related data in the <code class="highlighter-rouge">Table Panel</code> tab <code class="highlighter-rouge">Edge Table</code></li>
        </ul>
      </li>
      <li>Thickness: Number of genes shared between gene sets</li>
    </ul>
  </li>
</ul>

<p>The <code class="highlighter-rouge">Results Panel</code> shows our currently selected parameters that were originally set in the <code class="highlighter-rouge">Control Panel</code>; <code class="highlighter-rouge">Table Panel</code> has the <code class="highlighter-rouge">Node Table</code> tab selected by default, listing our (selected) gene sets.</p>

<ul>
  <li class="list-unstyled"><strong>Results Panel</strong>
    <ul>
      <li>Look at the <code class="highlighter-rouge">Legend</code> tab. Increase or decrease the stringency for displaying nodes (<code class="highlighter-rouge">Q-value Cutoff</code>) and edges (<code class="highlighter-rouge">Similarity Cutoff</code>) using the sliders. For instance, one can restrict the displayed gene sets to those with very low chance of being a false positive.</li>
    </ul>
  </li>
  <li class="list-unstyled"><strong>Table Panel</strong>
    <ul>
      <li>If you activate the <code class="highlighter-rouge">Heat Map</code> tab as in Figure 7, you will see a depiction of the normalized gene expression values and the samples coloured by class. If you click on the <code class="highlighter-rouge">Node Table</code> (Figure 8) it should display the information summarized in the graph but displayed in tabular form. Note that if a node in the graph is selected or if there is a search match,only those results will be shown. Click anywhere on the map outside of a node to see the full results.</li>
    </ul>
  </li>
</ul>

<blockquote>
  <p>You can search the Enrichment Map by keyword using the input text box located at the top-right. For example, searching for ‘IL5%NETPATH%IL5’ higlights the corresponding node in yellow and selects it so you can immediately view data in the <code class="highlighter-rouge">Table Panel</code>.</p>
</blockquote>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_output.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 7. Enrichment Map for gene sets enriched in BrCa versus HD.</strong>  Nodes coloured red are gene sets enriched in BrCa whereas blue shaded are gene sets enriched in HD. The term 'IL5' was used as the search term (upper right) and detected a single node (yellow) in the main window corresponding to the IL-5 signal transduction pathway from NetPath (i.e. 'IL5%NETPATH%IL5'). The 'Control Panel' was hidden in this case.
  <div class="text-left">
    <a type="button" class="btn btn-info" href="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_output.png" target="_blank">Open in new window</a>
  </div>
</div>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_table_panel.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 8. Table Panel.</strong> This panel displays tabular information for gene sets. Highlighted is the IL-5 signal transduction pathway from NetPath.
  <div class="text-left">
    <a type="button" class="btn btn-info" href="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_table_panel.png" target="_blank">Open in new window</a>
  </div>
</div>

<h3 id="output">Output</h3>

<p>It is very easy to get overwhelmed by the amount of options, settings and visual information in an Enrichment Map (Figure 7) so now is a good time to pause to reflect back upon our goal: To reduce the complexity in our long list of enriched gene sets from GSEA while maintaining as much useful information. To this end, note the presence of gene set ‘clusters’ as indicated by a very dense group of edges connecting nodes (e.g. Figure 7, top-left in blue). These are gene sets with many shared genes that likely represent very similar processes. Our aim will be to display such clusters as a single node and annotate it with one label that reflects the overarching theme of its constituents (Figure 9).</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_collapse.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 9. Preview of Enrichment Map clustering and annotation.</strong> On the left, a schematic of an Enrichment Map shows gene sets enriched in two phenotypes (red and blue). Clusters of gene sets will be identified by the Cytoscape app 'Autoannotate' which in turn, will use the 'WordCloud' app to derive a descriptive label for the cluster ('Theme One' and 'Theme Two'). The labels are extracted from the individual metadata associated with the gene sets. Finally, the clusters will be collapsed into single nodes to simplify the view.
</div>

<h4 id="label-clusters-of-similar-gene-sets">Label clusters of similar gene sets</h4>

<p>Clusters within the Enrichment Map represent similar biological processes and themes. In order to better summarize the Enrichment map we want to be able to annotate each of these clusters with the main theme associated with it. To do this we use the <a href="http://apps.cytoscape.org/apps/autoannotate" target="_blank">AutoAnnotate</a> app to help us summarize the network and its themes. AutoAnnotate first clusters the network and then uses <a href="http://apps.cytoscape.org/apps/wordcloud" target="_blank">WordCloud</a> to calculate the most frequent words present in each cluster node labels in efforts to highlight commonalities between the nodes in the cluster.</p>

<p><strong>i. Separating gene sets in each class</strong></p>

<p>In the main window graph, let us spatially separate the gene sets enriched in each class so that when we go to add labels, they are readily distinguishable.</p>

<p>We will select the BrCa group in red and drag the entire group over to the right separate it from the HD group in blue. To do this, Highlight the class by creating a column filter for nodes with a positive NES:</p>

<ul>
  <li><code class="highlighter-rouge">Control Panel</code> tab for <code class="highlighter-rouge">Select</code>
    <ul>
      <li>Click <code class="highlighter-rouge">+</code> to add a filter
        <ul>
          <li>Select <code class="highlighter-rouge">Column Filter</code> from the drop-down</li>
        </ul>
      </li>
      <li><code class="highlighter-rouge">Choose column....</code> drop-down
        <ul>
          <li>Select <code class="highlighter-rouge">Node:EM1_NES_dataset1</code></li>
          <li>Enter range to include only positive values</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<p>Figure 10 shows the selected nodes in yellow representing the gene sets enriched in BrCa. Drag the entire group to the right of the unselected gene sets.</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_cluster_filter.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
    <strong>Figure 10. Separating gene sets enriched in each class.</strong> The enriched gene sets in the BrCa class was selected by creating a column filter in the <code>Control Panel</code> tab <code>Select</code> for NES above 0.
    <div class="text-left">
      <a type="button" class="btn btn-info" href="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_cluster_filter.png" target="_blank">Open in new window</a>
    </div>
  </div>

<p><strong>ii. AutoAnnotate groups</strong></p>

<p>AutoAnnotate first clusters the network and then uses WordCloud to calculate the most frequent words present in each cluster’s node labels (Figure 10).</p>

<ul>
  <li><code class="highlighter-rouge">Apps -&gt; AutoAnnotate -&gt; New Annotation Set...</code></li>
</ul>

<p>Click <code class="highlighter-rouge">Create Annotations</code> to start the annotation process. You should see clusters forming and being annotated in the main window (Figure 11).</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_cluster_autoannotate.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 11. AutoAnnotate options.</strong>  <strong>Cluster Options</strong> There are two clustering options. Either you can have Auto Annotate perform clustering using the Clustermaker App or you can run your own clustering algorithms. <strong>Use clusterMaker App.</strong> <em>Cluster algorithm:</em> Choose from the list of possible clustering algorithms supported by Auto Annotate including Affinity Propagation, Cluster fuzzifier, Community clustering, Connected Components Clustering, MCL, and SCPS. By default this is set to MCL. <em>Edge weight column:</em> Any numeric edge attribute column can be used as weights in the clustering algorithm. By default this is set to EM_similarity_coeffecient which is a measure of how many genes two nodes have in common. The more genes two nodes have in common the higher the value and therefore the more likely they are to be found in the same cluster. <strong>Label Options</strong> <em>Label column:</em> Select the column you would like to use to compute the labels for each cluster. By default this is set to the Enrichement Map gene set description column (EM_GS_DESCR) but any string or list of strings can be used to annotate the clusters.
  <div class="text-left">
    <a type="button" class="btn btn-info" href="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_cluster_autoannotate.png" target="_blank">Open in new window</a>
  </div>
</div>

<p><strong>iii. Collapse groups</strong></p>

<p>Figure 11 shows a pretty busy picture; It is natural to gravitate towards large clusters that appear in the enrichment results, however, in this case, size does not indicate importance or strength but rather the amount of database annotation there exist for a particular pathway or process. Single nodes represent processes that are less well known but no less important than the large clusters. In order to remove the bias introduced by redundant pathway annotations it is good to collapse the network, i.e. create a single group node for every cluster whose name is summary annotation calculated for it, in order to more easily see the overall themes present in the enrichment results (Figure 12).</p>

<ul>
  <li><code class="highlighter-rouge">Control Panel</code> select the <code class="highlighter-rouge">AutoAnnotate</code> tab
    <ul>
      <li>Click the menu drop-down (button with 3 lines)</li>
      <li>Select <code class="highlighter-rouge">Collapse All</code></li>
    </ul>
  </li>
</ul>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_cluster_collapsed.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 12. Results of collapsing AutoAnnotate.</strong> Note that we have changed the <a href="http://manual.cytoscape.org/en/stable/Styles.html" target="_blank">styling</a> to increase the clarity of the text labels.
  <div class="text-left">
    <a type="button" class="btn btn-info" href="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_cluster_collapsed.png" target="_blank">Open in new window</a>
  </div>
</div>

<p><strong>iv. Expand a group</strong></p>

<p>Let’s reverse the process selectively and drill down into a particular group. Recall our running interest in the IL-5 signal transduction pathway originally curated by NetPath? Well it is hidden inside the cluster labelled ‘pid angiopoietin receptor pathway’. We can recover the view for this gene set (Figure 13).</p>

<ul>
  <li><code class="highlighter-rouge">Control Panel</code> select the <code class="highlighter-rouge">AutoAnnotate</code> tab
    <ul>
      <li>Right click <code class="highlighter-rouge">pid angiopoietin receptor pathway</code></li>
      <li>Select <code class="highlighter-rouge">Expand</code></li>
    </ul>
  </li>
</ul>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_cluster_uncollapsed.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 13. Selectively expanding the network for a single cluster.</strong>
  <div class="text-left">
    <a type="button" class="btn btn-info" href="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_cluster_uncollapsed.png" target="_blank">Open in new window</a>
  </div>
</div>

<p>Our cluster ‘pid angiopoietin receptor pathway’ has now been expanded to show the constituent pathways that include our IL-5 gene set. The edges indicate shared genes, but to get a better idea of the overlap, let’s look at the overlap in all pairs of gene sets in table format (Figure 14).</p>

<ul>
  <li>Select the cluster
    <ul>
      <li>Hold Shift + drag mouse over nodes</li>
    </ul>
  </li>
  <li>Table Panel
    <ul>
      <li>Select <code class="highlighter-rouge">Edge Table</code> tab</li>
    </ul>
  </li>
</ul>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_cluster_uncollapsed_ilgroup.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 14. Showing gene overlap in gene set cluster .</strong> The <code>Table Panel</code> tab for <code>Edge Table</code> shows the pair-wise overlap in genes between different gene sets in the cluster  (EM1_Overlap_size, EM1_Overlap_genes). Highlighted in the <code>Table Panel</code> is the overlap between IL-3 and IL-5.
  <div class="text-left">
    <a type="button" class="btn btn-info" href="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_cluster_uncollapsed_ilgroup.png" target="_blank">Open in new window</a>
  </div>
</div>

<p>Take for instance the IL-3 and IL-5 signalling pathways: IL-3 consists of 75 genes and IL-5 contains 49 (<code class="highlighter-rouge">Node Table</code>); The gene sets share 33 genes between them (<code class="highlighter-rouge">Edge Table</code>).</p>

<hr />

<p>Cytoscape affords the user a great deal of control over <a href="http://manual.cytoscape.org/en/stable/Styles.html" target="_blank">styles</a> and <a href="http://manual.cytoscape.org/en/stable/Navigation_and_Layout.html?highlight=layout" target="_blank">layout</a>. There is only so much that can be automated, so it will be up to you to tweak the look of the Enrichment Map to suit your needs. Please refer to the <a href="http://manual.cytoscape.org/en/stable/index.html" target="_blank">user manual</a> for full description of capabilities.</p>

<h2 id="a-hrefinterpretation-nameinterpretationiv-interpretationa"><a href="#interpretation" name="interpretation">IV. Interpretation</a></h2>

<p>So what now? Below, we list some helpful tips to aid you in extracting value from an enrichment analysis.</p>

<blockquote>
  <p>Check out a <a href="/guide/case_studies/enrichment_analysis/2016-09-chinen/" target="_blank">case study</a> to see how an Enrichment Map workflow was used to provide the basis for experimentally verifiable, mechanistic insights into cell function</p>
</blockquote>

<h3 id="build-trust">1. Build trust</h3>

<p>No algorithm or statistic can replace the experience, expertise and critical eye of you the researcher. The best place to start with a completed EM is to examine it for genes, pathways and themes that would be expected <em>a priori</em>. These are unsurprising results that have either been previously reported or which one could have easily guessed before the analysis was even performed. Do the results pass the sanity test?</p>

<p>As an example, direct your attention to the cluster annonated with the label <em>‘blood coagulation platelet activation’</em> (Hint: Do a text search). This cluster consists of a hierarchy of three gene sets referencing increasingly specific GO terms (Figure 15):</p>

<ol>
  <li>Hemostasis <a href="http://www.ebi.ac.uk/QuickGO/GTerm?id=GO:0007599" target="_blank">GO:0007599</a></li>
  <li>Blood coagulation <a href="http://www.ebi.ac.uk/QuickGO/GTerm?id=GO:0007596" target="_blank">GO:0007596</a></li>
  <li>Platelet Activation <a href="http://www.ebi.ac.uk/QuickGO/GTerm?id=GO:0030168" target="_blank">GO:0030168</a></li>
</ol>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_coag.png" alt="image" class="img-responsive slim" /></p>

<div class="figure-legend well well-lg text-justify">
  <strong>Figure 15. Gene ontology relationship for gene set cluster 'blood coagulation platelet activation'.</strong> <em>Adapted from <a href="http://www.ebi.ac.uk/QuickGO/" target="_blank">QuickGO.</a></em>
</div>

<h3 id="identify-novelinteresting-groups">2. Identify novel/interesting groups</h3>

<p>This is where the rubber meets the road: The enrichment analysis and EM are evidence that helps you to bridge bodies of knowledge and make new, interesting connections. The discoveries that will nourish a steady-stream of publications and earn the envy of peers is largely up to you.</p>

<p>One simple way to get the creative juices flowing is to try and understand whether there are established connections between certain gene sets/groups and your specific biological context of interest. For example, one can ask whether there is an established connections between cytokine signaling pathways (‘IL-3’, ‘IL-5’, ‘IL-6’), platelets and malignancies.</p>

<h3 id="drill-down-to-gene-expression">3. Drill down to gene expression</h3>

<p>Perhaps we’ve noticed an interesting pathway not previously reported in our context. One way to build confidence in the result is to simply examine visually whether there is a contrast in expression between classes.</p>

<blockquote>
  <p>Recall that our p-values for differential gene expression do not provide any information about the magnitude of the expression differences.</p>
</blockquote>

<p>One approach to examine expression of a gene set is to focus on the ‘leading edge’ - the subset of genes that contribute to the enrichment analysis enrichment score (ES). This leading edge can be easily seen in the <code class="highlighter-rouge">Table Panel</code>, <code class="highlighter-rouge">Heat Map</code> tab by selecting <code class="highlighter-rouge">GSEA ranking</code> from the <code class="highlighter-rouge">Sorting</code> drop-down (Figure 16).</p>

<blockquote>
  <p>Recall that the leading edge data derives from our rank file.</p>
</blockquote>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_interpret_leadingedge_il5.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 16. Leading edge analysis for IL-5 signal transduction pathway.</strong> Select the <code>GSEA Ranking</code> option from the <code>Sorting</code> drop-down. Genes of the gene set that contribute most to the enrichment score - the 'leading edge' - are highlighted in yellow.
  <div class="text-left">
    <a type="button" class="btn btn-info" href="/guide/media/workflows/pathway_enrichment_ii/visualize/figure_visualize_em_interpret_leadingedge_il5.png" target="_blank">Open in new window</a>
  </div>
</div>

<h3 id="drill-down-to-the-pathway">4. Drill down to the pathway</h3>

<p>Coming soon to a web app near you:</p>

<ol>
  <li>View: View the pathway (topology)</li>
  <li>Paint: View gene expression data on a pathway</li>
  <li>Nexus: Find neighbouring genes outside the pathway</li>
</ol>

        </div>
        
  <hr/>
  <h2>References</h2>
  
  <div class="reference_group" data-reflist="[10802651 21085593 22383865]"></div>


        
  <br/>
  <hr/>
  <div id="disqus_thread"></div>
  <script>
    /**
    *  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
    *  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
    var disqus_config = function () {
    this.page.url = "http://pathwaycommons.github.io/guide/workflows/pathway_enrichment_ii/visualize";  // Replace PAGE_URL with your page's canonical URL variable
    this.page.identifier = "/workflows/pathway_enrichment_ii/visualize"; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
    };

    (function() { // DON'T EDIT BELOW THIS LINE
    var d = document, s = d.createElement('script');
    s.src = '//pathway-commons.disqus.com/embed.js';
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
    })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>


      </div>
    </div>

  <script id="dsq-count-scr" src="//pathway-commons.disqus.com/count.js" async></script>
  </body>
</html>
