---
title: Ovarian Cancer
subtitle: The Cancer Genome Atlas Research Network characterizes high-grade serous ovarian adenocarcinoma
pmid: 21720365
cover: cover.jpg
pdf: nihms-313090.pdf
date: 2014-02-27
layout: publication
category: RNA sequencing
figures:
  figure_1: figure_1.jpg
  figure_2: figure_2.jpg
  figure_3: figure_3.jpg
  figure_4: figure_4.jpg
---

-   [I. Data Description](#dataDescription)
    - Background
    - Study goals
    - Methods
-   [II. Data retrieval](#dataRetrieval)
    - GDC Data transfer
-   [III. Data processing](#dataProcessing)
-   [IV. References](#references)

<hr/>

## <a href="#dataDescription" name="dataDescription">I. Data Description</a>
[The Cancer Genome Atlas](http://cancergenome.nih.gov/) (TCGA) is a collaboration between the [National Cancer Institute](http://www.cancer.gov/) (NCI) and the [National Human Genome Research Institute](https://www.genome.gov/) (NHGRI) that has generated comprehensive, multi-dimensional maps of the key genomic changes in 33 types of cancer. The TCGA dataset, comprising more than two petabytes of genomic data, has been made publicly available, and this genomic information helps the cancer research community to improve the prevention, diagnosis, and treatment of cancer.

### Background
The majority of deaths (~70%) from ovarian cancer are attributed to high-grade serous ovarian cancer (HGS-OvCa) (Vaughan 2011). Therapy and survival rates for ovarian cancers in general have not appreciably changed over the last 40 years:  HGS-OvCa are treated with aggressive surgery and taxane-platinum therapy. Approximately a quarter of naive patients will relapse within a year with 80-90% showing resistance to therapy. The five-year survival rate remains only at 31%.

It is now appreciated that ovarian cancers arise in varying anatomical locations and possess modest similarity to one another with regards to epidemiology and molecular alterations. In particular, a substantial proportion of HGS-OvCa are believed to arise in the distal fallopian tubes (Figure 1).

<br/>
  ![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_1 }}){: .img-responsive }
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 1. Clinical and molecular features of HGS-OvCa.</strong> <strong>a |</strong> Different stages of HGS-OvCa development in the human fallopian tube marked by p53 staining and cellular morphology. A substantial proportion of HGS-OvCa arises from the fallopian tube, most likely PAX8-positive fallopian tube secretory epithelial cells (FTSECs). p53 staining marks clonal expansion of cells (signatures) in the absence of morphological transformation of the fallopian tube epithelium. Piling up of cells and loss of epithelial architecture occurs in early lesions (tubal intraepithelial carcinoma (TIC)), finally leading to invasive cancer. *Adapted from Bowtell et al. 2011*.
</div>
<br/>

Molecular characterization of HGS-OvCa tumor samples has found a near universal (~98%) inactivation of the p53 tumour suppressor pathway (Ahmed 2010) and half of these cancers are defective for the homologous recombination (HR) DNA repair pathway. However, the high prevalence of such alterations is the exception, as large-scale expression, copy number analyses, and mutational screens have failed to identify recurrent 'druggable' targets. This partly explains the finding that single-agent, molecularly targeted therapies have yielded but incremental benefits for HGS-OvCa in the clinic.

It is clear that one of the most pressing challenges for the management and treatment of HGS-OvCa is a much clearer understanding of the cellular and molecular events that underlie the genesis, progression, and resistance of this cancer to therapy.

### Study goals
From the TCGA report:

> The lack of successful treatment strategies led the Cancer Genome Atlas (TCGA) researchers to measure  comprehensively genomic and epigenomic abnormalities on clinically annotated HGS-OvCa samples to identify molecular abnormalities that influence pathophysiology, affect outcome and constitute therapeutic targets.

### Methods

#### Tissue samples
Stage II-IV, clinically annotated HGS-OvCa and matched normal samples were retrieved. Patients were selected to match the demographic and epidemiological attributes of the typical population diagnosed with disease. Patients had all been treated with platinum and most with taxane prior to surgery.

#### Data generation overview
On June 6, 2016, the TCGA [announced](http://cancergenome.nih.gov/newsevents/newsannouncements/genomic-data-commons-launch) that their data would be housed under the [Genomic Data Commons](https://gdc.nci.nih.gov/) in an attempt to centralize and harmonize access to large-scale biological data generation efforts. The GDC website provides extensive documentation on the [datasets](https://gdc.nci.nih.gov/about-gdc/contributed-genomic-data-cancer-research) they make available and ways that users can narrow their search and [access](https://gdc.nci.nih.gov/access-data) the data.

As of July 2016, the TCGA data at the Genomic Data Commons was not providing microarray data from the publication. Instead, the GDC is providing RNA-sequencing data for (376) patient samples which we focus on for the remainder of this discussion.

The pipeline from patient sample to data file consists of a constellation of partners, participants and analysis protocols. We refer the reader to the [TCGA site](https://wiki.nci.nih.gov/display/TCGA/Introduction+to+TCGA#IntroductiontoTCGA-TCGADataFlow) for a full description. Figure 2 presents the GDC data model which is the central method of organization of all data artifacts ingested by the GDC. It provides a high-level overview of the elements involved in the data generation process.

> *The GDC draws upon the expertise of collaborators in the development of pipelines supporting data harmonization including the standardization of biospecimen and clinical data, the re-alignment of DNA and RNA sequence data against a common reference genome build, and the generation of derived data. GDC pipelines are implemented using data processing software and algorithms selected in consultation with the expert genomics community.* - [NCI Genomic Data Commons](https://gdc.nci.nih.gov/about-data/data-harmonization-and-generation/introduction)

![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_2 }}){: .img-responsive.slim }
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 2. GDC data model.</strong> The data model is represented as a graph with nodes and edges, and this graph is the store of record for the GDC. It maintains the critical relationship between projects, cases, clinical data and molecular data and insures that this data is linked correctly to the actual data file objects themselves, by means of unique identifiers. <em>Adapted from the <a href="https://gdc.nci.nih.gov/submit-data/gdc-data-harmonization/">GDC website</a></em>.
</div>

The GDC data model administrative elements (blue nodes) represent the key criteria we will use in [section II](#dataRetrieval) to filter for HGS-OvCa data in the GDC data portal.

#### RNA-Sequencing
The GDC mRNA-Seq alignment workflow follows the [International Cancer Genome Consortium (ICGC)](https://icgc.org/) and [STAR](https://github.com/alexdobin/STAR) (Dobin 2013) alignment standard operating procedures (Figure 3). 
![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_3 }}){: .img-responsive.super-slim }
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 3. Overview of RNA-Seq alignment.</strong> FastQC and RNA-SeQC are used to collect alignment metrics. This figure was adapted from the GDC web page on <a href="https://gdc.nci.nih.gov/">Data Harmonization and Generation</a>.
</div>

The GDC generates gene level quantification in Fragments Per Kilobase of transcript per Million mapped reads (FPKM). To facilitate cross-sample comparison and differential expression analysis, the GDC also provides Upper Quartile normalized FPKM (UQ-FPKM) values and raw mapping count (Figure 4).

![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_4 }}){: .img-responsive.super-slim }
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 4. Overview of RNA-seq quantification.</strong> This figure was adapted from the GDC web page on <a href="https://gdc.nci.nih.gov/">Data Harmonization and Generation</a>.
</div>

## <a name="dataRetrieval">II. Data retrieval</a>

Below we provide a step-by-step instructions to retrieve the HGS-OvCa RNA-seq data files to your computer. To accomplish this, we will be using the GDC [Data Transfer Tool](https://gdc-docs.nci.nih.gov/Data_Transfer_Tool/Users_Guide/Getting_Started/) which is a command line tool for robust and efficient data download.

### Using the GDC transfer tool

#### 1. Download the GDC Data Transfer Tool
Download the [GDC Data Transfer Tool](https://gdc.nci.nih.gov/access-data/gdc-data-transfer-tool) which is available for Linux (Ubuntu 14.x or later), OS X (10.9 Mavericks or later) and Windows (7 or later). Save the file to a convenient location on your computer. For the purposes of this recipe, we will assume the `gdc-client` is unzipped and available at `/opt/gdc/gdc-client` on OSX v10.11.4.

Access the built-in help by using the `-h` option:

```
$ /opt/gdc/gdc-client -h

usage: gdc-client [-h] [--version] {download,upload,interactive} ...

The Genomic Data Commons Command Line Client

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit

commands:
  {download,upload,interactive}
                        for more information, specify -h after a command
    download            download data from the GDC
    upload              upload data to the GDC
    interactive         run in interactive mode

```

#### 2. Prepare for Data Download
The goal of this step is to obtain a 'manifest' file describing the RNA-Seq files that the gdc-client should download. There is a detailed [description](https://gdc-docs.nci.nih.gov/Data_Transfer_Tool/Users_Guide/Preparing_for_Data_Download_and_Upload/) of how to navigate the GDC Data Portal to obtain a manifest file for the data set of interest.

*Narrow down your data search*. Use the [faceted search](https://gdc-portal.nci.nih.gov/search/s) interface to view all the available data. Narrow down to our data of interest: From the 'Cases' tab select 'Primary Site' (Ovary) and 'Cancer Program' (TCGA); From the 'Files' tab select 'Workflow Type' (HT-Seq FKPM-UQ). You should see that we have narrowed our search down to 379 RNA-Seq data files for 376 cases.

> To go directly to a page filtered for the TCGA HGS-OvCa RNA-seq data, simply visit this [link](https://gdc-portal.nci.nih.gov/search/s?filters=%7B%22op%22:%22and%22,%22content%22:%5B%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22files.data_category%22,%22value%22:%5B%22Transcriptome%20Profiling%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22files.data_format%22,%22value%22:%5B%22TXT%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.project.program.name%22,%22value%22:%5B%22TCGA%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.project.project_id%22,%22value%22:%5B%22TCGA-OV%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.project.primary_site%22,%22value%22:%5B%22Ovary%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.project.disease_type%22,%22value%22:%5B%22Ovarian%20Serous%20Cystadenocarcinoma%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.demographic.gender%22,%22value%22:%5B%22female%22%5D%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22files.analysis.workflow_type%22,%22value%22:%5B%22HTSeq%20-%20FPKM-UQ%22%5D%7D%7D%5D%7D).  

![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/TCGAOv_search.gif)

*Add files to cart.* This step tells the GDC portal what files you wish to place in the manifest for eventual download. On the GDC Data Portal Search site Summary tab, you can click the 'Add all files to the Cart'; The 'Cases' and 'Files' tabs allows you to restrict your download to individual data sets. We will select 'Add all files to the Cart'.

*Download the manifest.* Download the manifest file named `gdc_manifest_XXX_XXX.txt` where `X` is some identifier for this manifest to your computer. For the purposes of this recipe, we assume the manifest is downloaded to the path `/Users/username/Downloads/TCGAOV_data/gdc_manifest_XXX_XXX.txt`. This is also where your raw data files will be placed by the gdc-client.

#### 3. Download Data
After obtaining the manifest for the desired files, simply hand over the work to the gdc-client. Navigate to directory where you downloaded your manifest

```
$ cd /Users/username/Downloads/TCGAOV_data/
$ /opt/gdc/gdc-client download -m gdc_manifest_XXX_XXX.txt
```

You should see some status information indicating what directories are being downloaded to your computer.

```shell
Downloading 065393c7-9928-4fdc-9dbf-68ba9c708561.htseq.counts.gz (UUID 44b78a11-dfad-46f5-a203-599dc26ff876):
100% [########################################################################################] Time: 0:00:00 374.29 kB/s
Validating checksum...
Downloading 0045a228-1dad-4ced-9bc6-5067aeee0518.htseq.counts.gz (UUID 7525f3b0-b416-49d5-b839-40eaee3a42e4):
100% [########################################################################################] Time: 0:00:00 409.22 kB/s
Validating checksum...
Downloading f6a67523-d202-4ce7-af2e-b2cc4d8db20a.FPKM-UQ.txt.gz (UUID 743d5d0f-7515-41d9-b78a-af15dee6311f):
100% [########################################################################################] Time: 0:00:00 599.99 kB/s
Validating checksum...

...

SUMMARY:
Successfully downloaded: 379
```

Be patient, depending on the number of files the download can take a few seconds to many minutes. If any errors occur, make a note of the UUID and download them using an analogous procedure as described above. 

<hr/>

## <a href="#dataProcessing" name="dataProcessing">III. Data processing</a>

If your download is successful, you will have a set of directories each named according to the case UUID. Within each case directory you will find a compressed file (`<UUID>.FPKM-UQ.txt.gz`) and a `logs` directory. 

```shell
TCGAOV_data
│
└─── 0eb57c2f-10bc-4bfa-a810-dc774ff72e40
│   │	84e6af70-0c66-410c-9fad-6188151ef356.FPKM-UQ.txt.gz
│   │
│   ├───logs
│   	84e6af70-0c66-410c-9fad-6188151ef356.FPKM-UQ.txt.gz.parcel
│    
│
├─── 0afefdcb-0c8e-4106-a803-7799a3badb8d   
│   │	0afefdcb-0c8e-4106-a803-7799a3badb8d.FPKM-UQ.txt.gz
...
```

Here's a list of the tasks we need to accomplish to get the data ready for our analyses.

1. Data extraction  
2. Map UUID to TCGA ID
3. Map gene IDs
4. Concatenate FKPM values 
5. Extract Immunoreactive and Mesenchymal 

### Data extraction  

What we desire is a single text file containing a data layout

| Tables   |      Are      |  Cool |
|----------|:-------------:|------:|
| col 1 is |  left-aligned | $1600 |
| col 2 is |    centered   |   $12 |
| col 3 is | right-aligned |    $1 |




<hr/>

## <a href="#references" name="references">III. References</a>
<div class="panel_group" data-inline="21720365,21941283,26493647,20229506,18698038,21436879,23104886"></div>
