<!DOCTYPE html>

<html lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Process Data &middot; Guide
    
  </title>

  <!-- Icons -->
  <link rel="shortcut icon" href="/guide/media/favicon.ico">

  <!-- CSS -->
  
  <!-- Third-party CSS -->
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/bootstrap/dist/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/progress-tracker/app/styles/progress-tracker.css" />
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/font-awesome/css/font-awesome.min.css" />

  <!-- Custom CSS -->
  <link rel="stylesheet" href="/guide/public/css/main.css">

  <!-- Javascript -->
  <!-- Third-party javascript -->
  <script type="text/javascript" src="/guide/public/bower_components/jquery/dist/jquery.min.js"></script>
  <script type="text/javascript" src="/guide/public/bower_components/bootstrap/dist/js/bootstrap.min.js"></script>
  <script type="text/javascript" src="/guide/public/bower_components/cytoscape/dist/cytoscape.min.js"></script>
  <script type="text/javascript" src="/guide/public/js/deps.js"></script>
  <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
    MathJax.Hub.Config({ TeX: { equationNumbers: { autoNumber: "AMS" } } });
  </script>
  <!-- Custom javascript -->
  <script type="text/javascript" src="/guide/public/js/babel-compiled.js" defer></script>
  
</head>


  <body class="theme-base-01">
    <div id="embedded">
    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
           content to avoid any CSS collisions with our real content. -->
      <div class="wrap">
        <div class="container content embedded">
          <div id='ajax-spinner'><img src="/guide/media/spinner.gif"/></div>

            <h1 class="page-title">Process Data</h1>
            <div class="subtitle">Derive a list of differentially expressed genes from RNA sequencing data</div>
            <hr/>
            <div class="linkout clearfix">
              <ul class="list-inline pull-right">
              <li>
  <a href="http://twitter.com/intent/tweet?text=Derive a list of differentially expressed genes from RNA sequencing data+http://pathwaycommons.github.io/guide/workflows/pathway_enrichment_ii/process_data/&via=pathwaycommons"
  data-toggle="tooltip"
  title="Tweet this page"
  target="_blank" >
    <i class="fa fa-twitter-square fa-2x"></i>
  </a>
</li>
<li>
  <a href="https://plus.google.com/share?url=http://pathwaycommons.github.io/guide/workflows/pathway_enrichment_ii/process_data/"
  data-toggle="tooltip"
  title="Share on Google+"
  target="_blank" >
    <i class="fa fa-google-plus fa-2x"></i>
  </a>
</li>

              </ul>
            </div>
            <hr/>
            <ul>
  <li class="list-unstyled">Table of Contents
    <ul>
      <li class="list-unstyled"><a href="#goals">I. Goals</a></li>
      <li class="list-unstyled"><a href="#background">II. Background</a></li>
      <li class="list-unstyled"><a href="#workflow_step">III. Workflow Step</a></li>
    </ul>
  </li>
</ul>

<hr />

<h2 id="a-hrefgoals-namegoalsi-goalsa"><a href="#goals" name="goals">I. Goals</a></h2>

<div class="alert alert-warning text-justify" role="alert">
  This workflow step provides files in the <a href="#output">Output</a> section that are dependencies for the remaining steps. The goals for this step then are to read through this section, download the files and make sure you are aware of their contents and relevance. No further action on your part is necessary.
</div>

<p>To make the concepts in our workflow concrete, we use expression data from a study by Myron G. Best and colleagues (Best 2015). Their aim was to differentiate blood platelets from healthy donors to those diagnosed with a malignancy towards a proof-of-principle for blood-based cancer diagnosis.</p>

<p>By then end of this discussion you should (Figure 1):</p>

<ol>
  <li>Be familiar with the efforts by Best <em>et al.</em> to find RNA signatures of cancers within platelets</li>
  <li>Be aware of procedures used to determine differential expression from RNA sequencing output</li>
  <li>Download a set of files that are dependencies for subsequent workflow steps</li>
</ol>

<p><br /></p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/process_data/figure_processdata_overview.jpg" alt="image" class="img-responsive short" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 1. Goals.</strong> To illustrate construction of an Enrichment Map, we follow the work done by Best et al. (Best 2015) comparing RNA-seq profiles of platelets from healthy donors (HD) to those with breast cancer (BrCa) with the goal of defining blood-based diagnostic signatures. We will provide a brief overview of the study and describe the methods used to generate RNA-seq profiles from platelet samples. In this workflow step, we use the RNA-seq data and sample class metadata as inputs and perform a differential expression analysis for each gene. The output of this step are files which are the dependencies for subsequent workflow steps. Besides downloading these files and understanding what they contain, there is no further action to be taken in the workflow.
</div>

<h2 id="a-hrefbackground-namebackgroundii-backgrounda"><a href="#background" name="background">II. Background</a></h2>

<p>This workflow is framed using sample data from a study by Best <em>et al.</em> who developed aimed to provide a proof-of-principle for platelet-based cancer diagnosis. We briefly review this and related material in this section.</p>

<h3 id="blood-based-cancer-biopsies">Blood-based cancer biopsies</h3>

<p>Cancer surveillance would be greatly aided by practical, low-cost alternatives to support early-detection, diagnosis, and treatment decisions. Ideally these tools would be non-invasive yet retain the sensitivity and accuracy needed to reliably differentiate between normal and pathological states. Blood-based biomarkers have been aggressively pursued as a means to diagnose malignancies. The components of blood that have been examined include both cell-free molecules (e.g. DNA, RNA, proteins) along with immune cells (monocytes, platelets) (Figure 2).</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/process_data/figure_processdata_joose_cancell_2015_figure_1.png" alt="image" class="img-responsive slim" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 2. Blood-based cancer biopsies.</strong> Blood can be separated into different fractions in order to enrich for tumor-associated biomarkers. From the mononuclear cell fraction, circulating tumor cells (CTCs) may provide genomic, transcriptomic, and proteomic information on the tumor. From plasma or serum, cell-free nucleotides and exosomes can be further used to interrogate cancer-secreted bioparticles. <em>Adapted from Joose et al. (Joose 2015), Figure 1</em>.
</div>

<h3 id="a-platelet-primer">A platelet primer</h3>

<p>Within the marrow, platelets originate as cytoplasmic fragments of megakaryocytes which ‘bud-off’ into the circulation via shear forces generated by circulating blood. Approximately 1 trillion platelets circulate an adult human at any one time making it the second most abundant cell type in blood. The primary physiological role of platelets is to sense and accumulate at the sites of damaged endothelial tissue and initiate a blood clot to mitigate and vessel leakage (Semple 2011). Disruption of the integrity of the endothelium exposes extracellular molecules that signal adhesion of platelets which, in turn, secrete a host of molecules and cytokines that recruit additional platelets to form the initial hemostatic plug.</p>

<p>Platelets are far from simple vestibules of biomolecules. An increasing body of research supports an active role for platelets in modulating innate and adaptive immune responses and a direct involvement in pathologies such as sepsis, atherosclerosis and rheumatoid arthritis (Semple 2011). In particular, the immune regulatory role of platelets arises from receptor-mediated interactions with pathogens, neutrophils and antigen-presenting cells.</p>

<p>Although anuclear, it has been shown that platelets are neither inert nor homogeneous. At the transcript-level, circulating platelets have been shown to possess functional splicing apparatus that is triggered in response to external activation (Denis 2005). At the protein-level, platelets possess a fully functional translation apparatus and its proteome has been described as a ‘fluid’ of components that rapidly alters depending on the conditions (Lindemann 2007). Thus, platelets are subject to many common aspects of gene regulation in order to sense and respond to their environment.</p>

<h3 id="platelet-education-in-cancers">Platelet ‘education’ in cancers</h3>

<p>The pathogenesis of cancer requires the cooperation of a host of stromal and immune cells termed the tumour microenvironment. Indeed, a variety of immune cells that normally suppress cancer are coopted by tumours to enable evasion of immune surveillance in a process termed ‘education’ (Quail and Joyce 2013). One mechanism of heterotypic signalling involves shedding of exomes by cancer cells containing pro-tumourigenic and pro-metastatic factors.</p>

<p>Platelets have been implicated in aiding and abetting the metastatic potential of cancer cells through a variety of routes (Gay 2011) (Figure 3). First, they aggregate in order to shield cancer cells that have entered the vasculature from immune cell recognition. Second, they facilitate their extravasation by enabling cancer cell to arrest and adhesion at points in the vasculature. Third, platelets secrete variety of molecules that support cancer cell survival and promote endothelial permeability, further promoting their extravasation.</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/process_data/figure_processdata_gay_figure_2.png" alt="image" class="img-responsive slim" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 3. Molecular coordination between platelets and tumour cells supports metastasis from the bloodstream.</strong> As platelets become activated, they undergo a shape change, increase their adhesiveness, release granules and microparticles, and perpetuate the cohesion of heteroaggregates containing tumour cells, platelets and leukocytes. Platelet granules contain growth factors, chemokines and proteases. During cohesion, circulating tumour cells can interact with activated platelets and leukocytes, and form heteroaggregates that support attachment to the endothelium and thereby contribute to metastasis. During adhesion, initial, transient adhesion is followed by firm attachment that is mediated by integrins and intercellular cell adhesion molecules. During immune evasion, multivalent plasma proteins form intercellular bridges, and activated platelets and fibrinogen protect tumour cells from natural killer cell lysis during haematogenous metastasis. GP, glycoprotein; icAM, intercellular adhesion molecule; PAr, protease-activated receptor; PsGL1, P-selectin- glycoprotein ligand 1; sLea, sialyl Lewis a antigen; sLex, sialyl Lewis x antigen; vcAM vascular cell adhesion molecule; vWF, von Willebrand factor. <em>Adapted from Gay et al. (Gay 2011), Figure 2</em>.
</div>

<p>The close contact between cancer cells and platelets results in their ‘education’. For example, tumour-associated RNA (e.g. EGFRvIII in brain and  PCA3 in prostate) could be detected in platelets (Nilsson 2011) and is consistent with observations that tumor exomes could be taken up by platelets.</p>

<h3 id="rna-seq-of-tumour-educated-platelets-tem">RNA-seq of tumour-educated platelets (TEM)</h3>

<p>The evidence indicating that platelets have intimate contact with cancer cells, take up exome-derived RNA, demonstrate differential splicing in response to their environment and their abundance in blood supports the notion that they may possess a large degree of heterogeneity. Such diversity could be clinically relevant if they enable discrimination between different stages of a malignancy. Best <em>et al.</em> set out to determine just how much diagnostic ‘information’ is contained in platelet transcriptomes.</p>

<h4 id="sample-collection">Sample collection</h4>

<p>Best <em>et al.</em> prospectively collected blood platelets from 55 healthy donors (HD) and from 189 treated and untreated patients with cancers at varying stages (Table 1). In particular, 39 of these were from breast cancers (BrCa) which will be the focus of our workflow.</p>

<p><strong>Table 1. Summary of Patient Characteristics</strong> <em><small>Adapted from Best et al.</small></em>
<img src="/guide/media/workflows/pathway_enrichment_ii/process_data/figure_processdata_best_table_1.png" alt="image" class="img-responsive" /></p>

<h4 id="rna-sequencing-and-analysis">RNA sequencing and analysis</h4>

<p>Figure 4 depicts the sample collection and processing scheme. For each patient, approximately 100-500 picograms of total platelet RNA  - the equivalent content in less than a drop of blood - was extracted for sequencing.</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/process_data/figure_processdata_best_figure_1.png" alt="image" class="img-responsive slim" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 4. Sample collection and RNA sequencing.</strong> (A) Schematic overview of tumor-educated platelets (TEPs) as biosource for liquid biopsies. (B) Number of platelet samples of healthy donors and patients with different types of cancer. (C) TEP mRNA sequencing (mRNA-seq) workflow, as starting from 6 ml EDTA-coated tubes, to platelet isolation, mRNA amplification, and sequencing. (D) Correlation plot of mRNAs detected in healthy donor (HD) platelets and cancer patients’ TEPs, including highlighted increased (red) and decreased (blue) TEP mRNAs. (E) Heatmap of unsupervised clustering of platelet mRNA profiles of healthy donors (red) and patients with cancer (gray).<em>Adapted from Best et al. (Best 2015), Figure 1</em>.
</div>

<p>Since Best <em>et al.</em> were interested in the discriminatory capacity of transcriptomes, they initially filtered RNA species for those that were intron-spanning and had sufficiently high expression counts (&gt;5) to reduce the amount of noise.</p>

<p>A reduced set of 5 003 protein and non-coding RNAs (excluding Y chromosome and mitochondrial) were used in a pair-wise comparison of expression between HD and pan-cancer samples. Across all cancers the authors identified 1 453 and 793 RNAs with increased and decreased representation, respectively (Figure 4D). These differentially expressed genes were sufficient to discriminate HD and cancer-derived platelets (Figure 4E).</p>

<h4 id="breast-cancer-diagnostics">Breast cancer diagnostics</h4>

<p>Is the information in platelets sufficiently informative to discriminate between healthy donors and those with breast cancer? To determine this, the authors first performed a clustering analysis to extract a subset of RNA species (n = 192) with discriminatory power then fed these genes into a machine learning algorithm trained to assign the correct category for each sample. In this case, the authors reported a 100% test accuracy in discriminating platelets derived from normal versus breast cancer patients (Figure 5).</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/process_data/figure_processdata_best_figure_s2c.png" alt="image" class="img-responsive" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 5. Sample classification accuracy and sensitivity.</strong> Cross table of Support Vector Machine algorithm diagnostics with healthy donor subjects and BrCa. Unique tumor-specific gene lists were determined and used to train the different algorithms. For the BrCa-classifying algorithm, only female healthy donors were included. Indicated are sample numbers and detection rates in percentages. Accuracy performance for each algorithm is indicated below the cross table. <em>Adapted from Best et al. (Best 2015), Figure S2C</em>.
</div>

<h3 id="zooming-out-pathway-analysis">Zooming out: Pathway analysis</h3>

<p>That such a plentiful and accessible material could provide the basis of an astonishingly accurate classification scheme is an exciting achievement and lends support for blood-based cancer diagnostics. We can repurpose the same RNA measurements to probe deeper into the biology of the platelets themselves.</p>

<p>Let us revisit two points raised earlier:</p>

<ol>
  <li>Platelets play an active and important role in the physiology of cancer</li>
  <li>Platelets are functional cells that modify gene expression and behaviour in response to environmental stimuli</li>
</ol>

<p>While a list of genes expression differences with exquisite discriminatory power for breast cancer diagnosis is useful <em>per se</em>, we can go a step further: What biological processes distinguish platelets from and healthy and diseased patients? In other words, we wish to better understand how those differences in RNA species might underlie various <em>pathways</em> inside a cell. What does the transcriptomic data really <em>mean</em>?</p>

<h2 id="a-hrefworkflowstep-nameworkflowstepiii-workflow-stepa"><a href="#workflow_step" name="workflow_step">III. Workflow Step</a></h2>

<div class="box">
  <div class="box-title">Box 2. Workflow Terminology</div>

  <dl class="dl-horizontal box-terms">

    <dt>Workflow</dt>
    <dd>
      A set of <em>steps</em> that transform data into a useful form.
    </dd>

    <dt>Input</dt>
    <dd>
      Each step takes in data, files and configuration settings that represent dependencies.
    </dd>

    <dt>Analysis</dt>
    <dd>
      Step dependencies are transformed into their useful form.
    </dd>

    <dt>Output</dt>
    <dd>
      The results of the analysis of dependencies.
    </dd>
  </dl>

</div>

<p><br /></p>

<p>In this workflow step (Box 2), we will walk through the transformation of RNA sequencing data counts generated by Best <em>et al.</em> for platelets from 16 HD and 16 BrCa patients into three output files that are dependencies for later steps:</p>

<ul>
  <li>
    <h4 id="rank-file">Rank file</h4>
    <ul>
      <li>This file will be required for next workflow step ‘Identify Pathways’ where pathways are distilled from changes in gene expression (Figure 1; Step 2)</li>
      <li>This file is a  <a href="http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#RNK:_Ranked_list_file_format_.28.2A.rnk.29" target="_blank">RNK</a> table where each row contains a gene name followed by its ‘rank’ which is inversely correlated with the value of the differential expression testing p-value.</li>
    </ul>
  </li>
  <li>
    <h4 id="expression-file">Expression file</h4>
    <ul>
      <li>This file will be used in the final workflow step ‘Visualize’ so that we can view the gene expression heatmap associated with any pathway</li>
      <li>This file is a <a href="http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#TXT:_Text_file_format_for_expression_dataset_.28.2A.txt.29" target="_blank">TXT</a> table where columns are normalized RNA counts for different biological samples.</li>
    </ul>
  </li>
  <li>
    <h4 id="phenotype-file">Phenotype file</h4>
    <ul>
      <li>This file will be used in the final workflow step ‘Visualize’ so that we can differentiate (i.e. colour) those pathways ‘up-regulated’ in HD versus BrCa samples (Figure 1; Step 3)</li>
      <li>The <a href="http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#CLS:_Categorical_.28e.g_tumor_vs_normal.29_class_file_format_.28.2A.cls.29" target="_blank">CLS</a> format contains information about the sample classes (aka ‘condition’, ‘phenotype’) and assigns each sample to one class.</li>
    </ul>
  </li>
</ul>

<h3 id="input">Input</h3>

<h4 id="rna-seq-files-and-metadata">RNA-seq files and metadata</h4>

<p>In this workflow, we will be focusing on a subset of 16 BrCa and 16 HD samples. Table 2 shows an excerpt of RNA-seq output for one biological sample: The first column indicates the gene symbol (Ensembl) and the second column indicates the mapped sequence read count for that RNA species.</p>

<p><strong>Table 2. Sample high-throughput RNA sequencing counts</strong></p>

<table class="table table-hover table-condensed table-responsive">
  <tbody>
    <tr>
      <td>ENSG00000000003</td>
      <td>0</td>
    </tr>
    <tr>
      <td>ENSG00000000005</td>
      <td>0</td>
    </tr>
    <tr>
      <td>ENSG00000000419</td>
      <td>100</td>
    </tr>
    <tr>
      <td>…</td>
      <td>…</td>
    </tr>
    <tr>
      <td>ENSG00000273493</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

<p><a href="/guide/media/workflows/pathway_enrichment_ii/process_data/htseq_hd_brca_best_cancell_2015.zip" type="button" class="btn btn-info btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Raw RNA-Seq (.zip)</a></p>

<p>We also provide a tab-delimited metadata file (tep_phenotypes.txt) that contains the name (id) of each RNA-seq file and its corresponding class (Table 3).</p>

<blockquote>
  <p>This metadata file is something that would not be provided by a sequencing facility but would be simple to create in a text editor or Excel.</p>
</blockquote>

<p><strong>Table 3. Contents of metadata file</strong></p>

<table class="table table-hover table-condensed table-responsive">
  <thead>
    <tr>
      <th>id</th>
      <th>class</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>MGH-BrCa-H92-TR472_htsqct.txt</td>
      <td>BrCa</td>
    </tr>
    <tr>
      <td>MGH-BrCa-H89-TR471_htsqct.txt</td>
      <td>BrCa</td>
    </tr>
    <tr>
      <td>MGH-BrCa-H84-TR551_htsqct.txt</td>
      <td>BrCa</td>
    </tr>
    <tr>
      <td>…</td>
      <td>…</td>
    </tr>
    <tr>
      <td>HD-18-2_htsqct.txt</td>
      <td>HD</td>
    </tr>
  </tbody>
</table>

<p><a href="/guide/media/workflows/pathway_enrichment_ii/process_data/tep_phenotypes.zip" type="button" class="btn btn-info btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Metadata (.zip)</a></p>

<h3 id="analysis">Analysis</h3>

<p>With our RNA-seq count files and metadata in-hand, the true work involved at this stage is assigning a rank to each RNA species that based on some measure of differential RNA expression. Rather than provide a detailed discussion of the concerns surrounding differential expression testing, we provide a thumbnail sketch of the tasks involved in achieving this goal.</p>

<blockquote>
  <p>We refer the reader to our primer on <a href="/guide/primers/functional_analysis/rna_sequencing_analysis/" target="_blank">RNA sequencing analysis</a> for a detailed description of the theory underlying the processing steps described here.</p>
</blockquote>

<h4 id="data-wrangling">1. Data wrangling</h4>

<p>Getting the data into the format that is useful for downstream analysis is an important but often underappreciated aspect of computational biology research. In this case, there are three tasks that we must accomplish with our data and metadata.</p>

<p>First, we must integrate or ‘merge’ the 32 RNA-seq files together into a single table. The reason for this is that data in table format is a form that is more easily loaded into RNA-seq analysis software packages. Typically, sequencing facilities often provide individual files for each RNA-seq sample similar to that in Table 2, but mileage may vary depending on your particular facility.</p>

<p>Second, we must perform gene ‘ID mapping’. This entails translating the names of genes/RNA species provided within the raw RNA-seq files into a desired namespace. This is necessary because the ‘enrichment’ software that distills pathways from gene expression must be able to match RNA counts for a gene with the genes that constitute candidate pathways.</p>

<p>Third, our metadata file is sufficient to generate our phenotype file (Table 4), which declares the number of samples and classes (row 1), names the classes (row 2) and then declares the class to which each sample belongs (row 3). Since our metadata contains the name (id) of the sample and the class, this is a simple task.</p>

<p><strong>Table 4. Phenotype output (.cls file)</strong></p>

<table class="table table-hover table-condensed table-responsive">
  <tbody>
    <tr>
      <td>30</td>
      <td>2</td>
      <td>1</td>
      <td> </td>
      <td>…</td>
      <td> </td>
    </tr>
    <tr>
      <td>#</td>
      <td>HD</td>
      <td>BrCa</td>
      <td> </td>
      <td>…</td>
      <td> </td>
    </tr>
    <tr>
      <td>HD</td>
      <td>HD</td>
      <td>HD</td>
      <td>…</td>
      <td>BrCa</td>
      <td>BrCa</td>
    </tr>
  </tbody>
</table>

<h4 id="filtering">2. Filtering</h4>

<p>Biological processes are inherently noisy (Raser 2005) and the same goes for gene expression. Some of this gene expression noise arises from the stochastic nature of biochemical reactions which are rather pronounced when dealing with small numbers of molecules.</p>

<p>In practical terms, RNA species with very low mapped read counts in a small number of samples can be highly variable. Consequently, we choose to ignore these in the search for differential expression. Best <em>et al.</em> use the rule of thumb that ‘genes with less than five (non-normalized) read counts in all samples were excluded from analyses’.</p>

<h4 id="normalization">3. Normalization</h4>

<p>RNA for a sample can be sequenced to varying ‘depths’. This means that the total number of sequence reads mapped to a gene for an individual sequencing run is not necessarily constant. The reason for this lies in the nature of next-generation sequencing technologies. Nevertheless, what most concerns us is not the absolute counts of an individual RNA species coming out of a sequencing run but rather the proportion. In practical terms, we desire a fair-comparison of RNA counts between samples that takes into account variation in depth.</p>

<p>Over the years, several approaches have been proposed to account for varying depth in RNA-seq outputs (Oshlack 2010). Our recommendation is to use a normalization technique called Trimmed mean of M-values (TMM; Robinson &amp; Oshlack 2010) that effectively standardizes counts between distinct sequencing runs by assuming that most genes are not expected to alter their expression.</p>

<p>At this stage, we can generate an expression file of normalized RNA counts where row names are gene symbols and column names are sample IDs (Table 5).</p>

<p><strong>Table 5. Expression output (counts per million mapped reads)</strong></p>

<table class="table table-hover table-condensed table-responsive">
  <thead>
    <tr>
      <th style="text-align: center">NAME</th>
      <th style="text-align: center">DESCRIPTION</th>
      <th style="text-align: center">MGH-BrCa-H92-TR472_htsqct</th>
      <th style="text-align: center">…</th>
      <th style="text-align: center">Breast-60_htsqct</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center">PPP1R2</td>
      <td style="text-align: center">PPP1R2</td>
      <td style="text-align: center">20.0960199625179</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">21.2351177118648</td>
    </tr>
    <tr>
      <td style="text-align: center">STARD7</td>
      <td style="text-align: center">STARD7</td>
      <td style="text-align: center">9.80293656708189</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">0</td>
    </tr>
    <tr>
      <td style="text-align: center">TDP2</td>
      <td style="text-align: center">TDP2</td>
      <td style="text-align: center">35.2905716414948</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">89.9369691326038</td>
    </tr>
    <tr>
      <td style="text-align: center">…</td>
      <td style="text-align: center"> </td>
      <td style="text-align: center"> </td>
      <td style="text-align: center"> </td>
      <td style="text-align: center"> </td>
    </tr>
    <tr>
      <td style="text-align: center">ANAPC13</td>
      <td style="text-align: center">ANAPC13</td>
      <td style="text-align: center">1.47044048506228</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">1.24912457128616</td>
    </tr>
  </tbody>
</table>

<h4 id="differential-expression-testing">4. Differential expression testing</h4>

<p>In this stage we perform a pair-wise comparison of RNA species counts in BrCa samples relative HD samples. The framework used to determine differential RNA expression is a ‘hypothesis-testing’ technique that entails the following:</p>

<ol>
  <li>Declare the null hypothesis of no difference in RNA counts for each gene</li>
  <li>Define a null distribution that describes how RNA counts vary under circumstances where there is <em>no association between RNA counts and class</em></li>
  <li>Calculate the probability (p-value) of observing a difference in RNA species counts at least as extreme as the one observed <em>assuming the null hypothesis/distribution</em></li>
</ol>

<p>At this stage, we can generate a rank file where row names are gene symbols and a single column indicates the rank calculated as a function of their p-value. The larger the magnitude of the positive or negative rank, the rarer such an observation would be under the assumption of no association between class and RNA count (Table 6).</p>

<p><strong>Table 6. Rank output</strong></p>

<table class="table table-hover table-condensed table-responsive">
  <thead>
    <tr>
      <th>gene</th>
      <th style="text-align: center">rank</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>ARHGAP45</td>
      <td style="text-align: center">58.1795999084575</td>
    </tr>
    <tr>
      <td>NCK2</td>
      <td style="text-align: center">50.6627080327017</td>
    </tr>
    <tr>
      <td>TRIM58</td>
      <td style="text-align: center">50.6246427495811</td>
    </tr>
    <tr>
      <td>…</td>
      <td style="text-align: center">…</td>
    </tr>
    <tr>
      <td>MDM4</td>
      <td style="text-align: center">-27.685318540466</td>
    </tr>
  </tbody>
</table>

<h3 id="a-hrefoutput-nameoutputoutputa"><a href="#output" name="output">Output</a></h3>

<p>Listed below are the outputs of this step that will be required as input dependencies for the next steps of the workflow.</p>

<h4 id="dependencies-for-gsea-workflow-step-2-identify-pathways">Dependencies for GSEA (Workflow Step 2: Identify Pathways)</h4>
<ol>
  <li>Ranks
    <ul>
      <li>Genes ranked as a function of p-value from differential expression.
  <a href="/guide/media/workflows/pathway_enrichment_ii/process_data/brca_hd_tep_ranks.rnk" type="button" class="btn btn-success btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Ranks (.rnk)</a></li>
    </ul>
  </li>
</ol>

<h4 id="enrichment-map-dependencies-workflow-step-3-visualize">Enrichment Map dependencies (Workflow Step 3: Visualize)</h4>

<ol>
  <li>Expression
    <ul>
      <li>Normalized counts for RNA species in for each sample.
  <a href="/guide/media/workflows/pathway_enrichment_ii/process_data/brca_hd_tep_expression.txt" type="button" class="btn btn-success btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Expression (.txt)</a></li>
    </ul>
  </li>
  <li>Phenotype
    <ul>
      <li>Declaration of classes and sample assignment. <a href="/guide/media/workflows/pathway_enrichment_ii/process_data/brca_hd_tep_phenotype.cls" type="button" class="btn btn-success btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Phenotype (.cls)</a></li>
    </ul>
  </li>
</ol>

        </div>
        
  <hr/>
  <h2>References</h2>
  
  <div class="reference_group" data-reflist="[26525104 16096058 21258396 26555171 21832279 21176179 21436837 24202395 16179466 20196867]"></div>


        
  <br/>
  <hr/>
  <div id="disqus_thread"></div>
  <script>
    /**
    *  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
    *  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
    var disqus_config = function () {
    this.page.url = "http://pathwaycommons.github.io/guide/workflows/pathway_enrichment_ii/process_data";  // Replace PAGE_URL with your page's canonical URL variable
    this.page.identifier = "/workflows/pathway_enrichment_ii/process_data"; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
    };

    (function() { // DON'T EDIT BELOW THIS LINE
    var d = document, s = d.createElement('script');
    s.src = '//pathway-commons.disqus.com/embed.js';
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
    })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>


      </div>
    </div>

    <script id="dsq-count-scr" src="//pathway-commons.disqus.com/count.js" async></script>
    <script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-43341809-8', 'auto');
  ga('send', 'pageview');

  $(document).ready(function() {
    // Link
    $('a').click(function() {
      var id = $(this).attr("href");

      var label = "link";
      var action = id;
      ga('send', 'event', label, action);
      // console.log(label + " " + action);
    });
  });
</script>

  </body>
</html>
