<!DOCTYPE html>

<html lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Process Data &middot; Guide
    
  </title>

  <!-- Icons -->
  <link rel="shortcut icon" href="/guide/media/favicon.ico">

  <!-- CSS -->
  
  <!-- Third-party CSS -->
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/bootstrap/dist/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/progress-tracker/app/styles/progress-tracker.css" />
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/font-awesome/css/font-awesome.min.css" />

  <!-- Custom CSS -->
  <link rel="stylesheet" href="/guide/public/css/main.css">

  <!-- Javascript -->
  <!-- Third-party javascript -->
  <script type="text/javascript" src="/guide/public/bower_components/jquery/dist/jquery.min.js"></script>
  <script type="text/javascript" src="/guide/public/bower_components/bootstrap/dist/js/bootstrap.min.js"></script>
  <script type="text/javascript" src="/guide/public/js/deps.js"></script>
  <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
    MathJax.Hub.Config({ TeX: { equationNumbers: { autoNumber: "AMS" } } });
  </script>
  <!-- Custom javascript -->
  <script type="text/javascript" src="/guide/public/js/babel-compiled.js" defer></script>
  
</head>


  <body class="theme-base-01">
    <div id="embedded">
    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
           content to avoid any CSS collisions with our real content. -->
      <div class="wrap">
        <div class="container content embedded">
          <div id='ajax-spinner'><img src="/guide/media/spinner.gif"/></div>

            <h1 class="page-title">Process Data</h1>
            <div class="subtitle">Derive a list of differentially expressed genes from RNA sequencing data</div>
            <hr/>
            <div class="linkout clearfix">
              <ul class="list-inline pull-right">
              <li>
  <a href="http://www.facebook.com/sharer/sharer.php?u=/workflows/pathway_enrichment_ii/process_data/&title=Process Data"
  data-toggle="tooltip"
  title="Share this page on Facebook"
  target="_blank" >
    <i class="fa fa-facebook-official fa-2x"></i>
  </a>
</li>
<li>
  <a href="http://twitter.com/intent/tweet?text=Derive a list of differentially expressed genes from RNA sequencing data+/workflows/pathway_enrichment_ii/process_data/&via=pathwaycommons"
  data-toggle="tooltip"
  title="Tweet this page"
  target="_blank" >
    <i class="fa fa-twitter-square fa-2x"></i>
  </a>
</li>
<li>
  <a href="https://plus.google.com/share?url=/workflows/pathway_enrichment_ii/process_data/"
  data-toggle="tooltip"
  title="Share on Google+"
  target="_blank" >
    <i class="fa fa-google-plus fa-2x"></i>
  </a>
</li>

              </ul>
            </div>
            <hr/>
            <ul>
  <li class="list-unstyled">Table of Contents
    <ul>
      <li class="list-unstyled"><a href="#goals">I. Goals</a></li>
      <li class="list-unstyled"><a href="#background">II. Background</a></li>
      <li class="list-unstyled"><a href="#workflow_step">III. Workflow Step</a></li>
      <li class="list-unstyled"><a href="#dependencies">IV. Dependencies</a></li>
      <li class="list-unstyled"><a href="#references">V. References</a></li>
    </ul>
  </li>
</ul>

<hr />

<div class="alert alert-warning text-justify" role="alert">
  This workflow transforms RNA sequencing data into an Enrichment Map. If you simply wish to obtain the file requirements for the next workflow steps, proceed directly to <a href="#dependencies">IV. Dependencies</a>.
</div>

<h2 id="a-hrefgoals-namegoalsi-goalsa"><a href="#goals" name="goals">I. Goals</a></h2>

<p>The overarching goal of this workflow is to generate an Enrichment Map - a visual description of the pathways altered between two states. We will infer these alterations via global gene expression measurements. To make these concepts concrete, we use data from a study by Best <em>et al.</em> (Best 2015) comparing expression profiles of blood platelets from healthy donors to those diagnosed with breast cancer.</p>

<p>By then end of this discussion you should (Figure 1):</p>

<ol>
  <li>Be familiar with a study examining gene expression signatures in ‘blood-based liquid biopsies’</li>
  <li>Be familiar with the RNA sequencing pipeline and differential expression analysis used in this study</li>
  <li>Obtain a set of processed data files required for the remainder of the enrichment workflow</li>
</ol>

<p><br />
<img src="/guide/media/workflows/pathway_enrichment_ii/process_data/figure_processdata_overview.jpg" alt="image" class="img-responsive short" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 1. Goals.</strong> For illustrative purposes, we describe a study be Best et al. (Best 2015) who compare RNA-seq profiles of platelets from healthy donors (HD) and those with a variety of malignancies including breast cancer (BrCa) with goal of defining blood-based signatures. We will provide a brief overview of the study and describe the methods used to generate RNA-seq profiles from platelets. In this workflow step, we use the RNA-seq data and sample class metadata as inputs and perform a differential expression analysis for each gene. The output of this step are various files required for the remainder of the workflow: Gene-level expression differences are mapped to pathway-level changes Gene Set Enrichment Analysis (GSEA) and displayed visually with an Enrichment Map.
</div>

<h2 id="a-hrefbackground-namebackgroundii-backgrounda"><a href="#background" name="background">II. Background</a></h2>

<h3 id="blood-based-cancer-biopsies">Blood-based cancer biopsies</h3>

<p>Cancer surveillance in the clinic would be greatly aided by practical and low-cost alternatives that would provide valuable support for early-detection, diagnosis, stratification and treatment. To accomplish this, such tests would ideally be non-invasive yet retain the sensitivity and accuracy neccessary to aid in the discrimination of different cancers types or the state of their progression.</p>

<p>Blood-based biomarkers have been agressively as a means to provide valuable information about a patient’s malignancy. In particular, the components of blood that may be informative include the cell-free molecules (e.g. DNA, RNA, proteins) along with immune cells (monocytes, platelets) that have had intimate contact with primary and metastatic cancer cells (Figure 2).</p>

<p><img src="/guide/media/workflows/pathway_enrichment_ii/process_data/figure_processdata_joose_cancell_2015_figure_1.png" alt="image" class="img-responsive slim" /></p>
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 2. Blood-based cancer biopsies.</strong> Blood can be separated into different fractions in order to enrich for tumor-associated biomarkers. From the mononuclear cell fraction, circulating tumor cells (CTCs) may provide genomic, transcriptomic, and proteomic information on the tumor. From plasma or serum, cell-free nucleotides and exosomes can be further used to interrogate cancer-secreted bioparticles. Tumor educated platelets (TEPs) carry additional information on the location of the tumor in their mRNA. <em>Adapted from Joose et al (Joose 2015)</em>.
</div>

<h3 id="platelets--malignancy">Platelets &amp; Malignancy</h3>

<p>The primary physiolocal role of platelets are to sense and accumulate at the sites of damaged endotheial tissue and initiate a blod clot to mitigate and vessel leak (Semple 2011). Within the marrow, platelets originate as cytoplasmic fragments of megakaryocytes which ‘bud-off’ into the circulation via shear forces generated by circulating blood. Approximately 1 trillion platelets circulate an adult human at any one time and live for an average of ten days before being sequestered by the spleen.</p>

<p>Disruption of the integrity of the endothelium exposes extracellular molecules that signal adhesion of platelets to formt the initial hemostatic plug. Upon activation, platelets secrete a host of molecules and proteins that recuit additional platelets and further activation.</p>

<p>Although anuclear, platelets are for all intents and purposes active cells in that they play</p>

<p>Can sense  (Nilsson 2011).
Can promote  (Kuznetsov 2011).</p>

<h2 id="a-hrefworkflowstep-nameworkflowstepiii-workflow-stepa"><a href="#workflow_step" name="workflow_step">III. Workflow Step</a></h2>

<h3 id="input">Input</h3>

<h3 id="analysis">Analysis</h3>

<p>We refer the reader to our primer on <a href="/guide/primers/functional_analysis/rna_sequencing_analysis/" target="_blank">RNA sequencing analysis</a> for a detailed description of the theory underlying the processing steps described here.</p>

<h3 id="output">Output</h3>

<h2 id="a-hrefdependencies-namedependenciesiv-dependenciesa"><a href="#dependencies" name="dependencies">IV. Dependencies</a></h2>

<h3 id="download">Download</h3>

<ul>
  <li>GSEA depedencies
    <ol>
      <li>Genes ranked as a function of p-value from differential expression testing. <a href="/guide/media/workflows/pathway_enrichment_ii/process_data/brca_hd_tep_ranks.rnk" type="button" class="btn btn-success btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Ranks (.rnk)</a></li>
    </ol>
  </li>
  <li>Enrichment Map depedencies
    <ol>
      <li>Normalized expression counts for each gene in each sample. <a href="/guide/media/workflows/pathway_enrichment_ii/process_data/brca_hd_tep_expression.txt" type="button" class="btn btn-success btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Expression (.txt)</a></li>
      <li>Definition of sample classes (i.e. BrCa or HD). <a href="/guide/media/workflows/pathway_enrichment_ii/process_data/brca_hd_tep_phenotype.cls" type="button" class="btn btn-success btn-lg btn-block" download=""><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Phenotype (.cls)</a></li>
    </ol>
  </li>
</ul>

<h3 id="original-data-from-geo">Original Data from GEO</h3>

<p>The original data has been deposited in the NCBI <a href="https://www.ncbi.nlm.nih.gov/geo/" target="_blank">Gene Expression Omnibus</a> under the accession number <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE68086" target="_blank">GSE68086</a>.</p>

<hr />

<h2 id="a-hrefreferences-namereferencesv-referencesa"><a href="#references" name="references">V. References</a></h2>
<!-- <div class="panel_group" data-inline="26525104,21436837"></div> -->

        </div>        
        

      </div>
    </div>

  <script id="dsq-count-scr" src="//pathway-commons.disqus.com/count.js" async></script>
  </body>
</html>
