---
title: Get Data
subtitle: Use the Genomic Data Commons (GDC) data portal to retrieve TCGA ovarian cancer project (TCGA-OV) RNA sequencing data
pdf: nihms-313090.pdf
date: 2014-02-27
layout: embedded
category: pathway_enrichment
order: 1
gists:
  id: 293acd56bfc4181727f3832daed795b1
  file_1: get_data.R
data:
  subtype: Verhaak_JCI_2013_tableS1.txt
  tcgaov_data: TCGAOV_data.rda
figures:
  figure_1: figure_getdata_overview.jpg
  figure_2: gdc_data_model.jpg
  figure_3: figure_rnaSeq_gdc.jpg
  figure_4: hgsovca_bowtell_2011.jpg
  figure_5: tcga_barcode.png
  figure_6: figure_getdata_tcgabiolinks_overview.jpg
  figure_7: figure_getdata_summarizedExperiment.jpg
---

<!-- Global options -->


- {:.list-unstyled} Table of Contents
  - {:.list-unstyled} [I. Goals](#goals)
  - {:.list-unstyled} [II. Background](#background)
  - {:.list-unstyled} [III. Practical](#practical)
  - {:.list-unstyled} [IV. Data](#data)
  - {:.list-unstyled} [V. References](#references)

<hr/>

<div class="alert alert-warning text-justify" role="alert">
  To directly get the TCGA-OV project RNA sequencing data, see <a href="#data">IV. Data</a>.
</div>

## <a href="#goals" name="goals">I. Goals</a>
Efforts to characterize cancers from the clinical to molecular level are underway. An important aspect is to make this data accessible to the research community. [The Cancer Genome Atlas (TCGA)](http://cancergenome.nih.gov/abouttcga/overview){:target="_blank"} is a cooperative aimed at the comprehensive characterization of common cancers. This data is being made available via the [Genomic Data Commons](https://gdc.nci.nih.gov/){:target="_blank"} (GDC) repository of the National Cancer Institute (NCI) (Figure 1).

This section provides detailed instructions on sourcing raw gene expression *data* and *metadata* (Box 1) suitable for downstream differential expression analysis (Figure 1). Though the focus here is on obtaining gene expression data, the methods are applicable to other data types.

By then end of this discussion you should:

1. Be familiar with the GDC data repository
2. Be able to source raw RNA sequencing (meta)data from a TCGA study
3. Be able to prepare the data for downstream differential gene expression analysis

<br/>
![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_1 }}){: .img-responsive.slim }
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 1. Summary & goals.</strong> The Genomic Data Commons (GDC) is a repository that stores data from  The Cancer Genome Atlas (TCGA) effort to characterize a wide variety of cancers. TCGA aims to generate data concerning mutations, copy number variants, post-transcriptional modifications and gene expression, including RNA sequencing. Detailed clinical and biospecimen metadata are also made available for each sample. Our goal is to show how to obtain RNA-seq data in a format suitable for downstream analysis of differential gene expression. This involves querying the GDC for data availability (cancer, data type), downloading the data and preparing it for downstream analysis.
</div>

## <a href="#background" name="background">II. Background</a>

### The Genomic Data Commons (GDC)

TCGA data is now housed under the GDC in an attempt to centralize and harmonize access to large-scale biological data generation efforts. The GDC website provides extensive documentation on the growing body of [programs](https://gdc.nci.nih.gov/about-gdc/contributed-genomic-data-cancer-research){:target="_blank"} who contribute genomic data. They also make available [documentation](https://gdc.nci.nih.gov/access-data){:target="_blank"} on how to search the portal and access the data.

> *The GDC Data Portal provides access to the subset of TCGA data that has been harmonized by the GDC using its data generation and harmonization pipelines. TCGA data in the GDC Data Portal includes BAM files aligned to the latest human genome build, VCF files containing variants called by the GDC, and RNA-seq expression data harmonized by the GDC.*
> <footer class="text-right"><a href="https://gdc.nci.nih.gov/gdc-tcga-data-access-matrix-users">GDC for TCGA Data Access Matrix Users</a></footer>

<ul class="aside terms">
  <div class="aside-title">Box 1. Terminology</div>

  <li class="aside terms">
    <strong>Raw data.</strong> Un-manipulated data exactly as it is generated by the measurement technology.
  </li>

  <li class="aside terms">
  <strong>Metadata.</strong> Descriptors of the biological data. This can include technical details (e.g. sequencing platform) and biological details (e.g. patient age).
  </li>

  <li class="aside terms">
  <strong>Case.</strong> The collection of all data related to a specific subject in the context of the specific project.
  </li>

  <li class="aside terms">
  <strong>Sample.</strong> A biological material sourced from a case.
  </li>

  <li class="aside terms">
  <strong>Program.</strong> Name for the large-scale genome research organization with a broad framework of goals to be achieved (e.g. The Cancer Genome Atlas).
  </li>

  <li class="aside terms">
  <strong>Project.</strong> A specifically defined piece of work that is undertaken or attempted to meet a single requirement.
  </li>
</ul>

#### GDC data model

The pipeline from study participant to data consists of a complex constellation of partners, participants and analysis protocols. For a full description we refer the reader to the [TCGA documentation on Data Flow](https://wiki.nci.nih.gov/display/TCGA/Introduction+to+TCGA#IntroductiontoTCGA-TCGADataFlow){:target="_blank"}. Figure 2 presents the GDC data model which is the central method of organization of all data artifacts ingested by the GDC. It provides a high-level overview of the elements involved in the data generation process and the nomenclature that will be referred to herein.

![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_2 }}){: .img-responsive.slim }
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 2. GDC data model.</strong> The data model is represented as a graph with nodes and edges, and this graph is the store of record for the GDC. It maintains the critical relationship between  projects, cases, clinical data and molecular data for a program and insures that this data is linked correctly to the actual data file objects themselves, by means of unique identifiers. The GDC maintains a <a href="https://gdc-docs.nci.nih.gov/Data_Dictionary/viewer/">Data Dictionary</a> that describes the data model elements.
</div>


#### Raw RNA sequencing data

The GDC mRNA-seq alignment workflow follows the [International Cancer Genome Consortium (ICGC)](https://icgc.org/){:target="_blank"} and [Spliced Transcripts Alignment to a Reference (STAR)](https://github.com/alexdobin/STAR){:target="_blank"} alignment standard operating procedures (Dobin 2013) (Figure 3A).

![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_3 }}){: .img-responsive }
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 3. Overview of RNA-sequencing data generation.</strong> <strong>A. </strong> RNA-seq alignment. FastQC and RNA-SeQC are used to collect alignment metrics. <strong>B. </strong> Expression quantification. For the purposes of our downstream data analyses, we will require an unnormalized 'Gene Count' data format. This figure was adapted from the GDC document on <a href="https://gdc.nci.nih.gov/about-data/data-harmonization-and-generation/genomic-data-harmonization/high-level-data-generation/rna-seq-quantification">Data Harmonization and Generation</a>.
</div>

The GDC portal makes available files of RNA sequencing count data ('Gene Count' also referred to as 'HT-Seq Counts') along with normalized gene level quantification in Fragments Per Kilobase of transcript per Million mapped reads (FPKM). To facilitate cross-sample comparison and differential expression analysis, the GDC also provides Upper Quartile normalized FPKM (FPKM-UQ) values and raw mapping count (Figure 3B).

When describing the method for obtaining RNA-seq data from GDC, we use as a concrete example the TCGA effort to characterize high-grade serous ovarian carcinomas (HGS-OvCa). A brief description of the study follows.

### The TCGA-OV project

Nearly 70% of deaths from ovarian cancer are attributed to HGS-OvCa (Vaughan 2011). Therapy and survival rates for ovarian cancers in general have not appreciably changed over the last 40 years:  HGS-OvCa are treated with aggressive surgery and taxane-platinum therapy. Approximately a quarter of naive patients will relapse within a year with 80-90% showing resistance to therapy. The five-year survival rate remains poor at only 31%.

It is now appreciated that ovarian cancers arise in varying anatomical locations and possess modest similarity to one another with regards to epidemiology and molecular alterations. In particular, a substantial proportion of HGS-OvCa are believed to arise in the distal fallopian tubes (Figure 4).

![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_4 }}){: .img-responsive }
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 4. Clinical and molecular features of HGS-OvCa.</strong>  Different stages of HGS-OvCa development in the human fallopian tube marked by p53 staining and cellular morphology. A substantial proportion of HGS-OvCa arises from the fallopian tube. p53 staining marks clonal expansion of cells (signatures) in the absence of morphological transformation of the fallopian tube epithelium. Piling up of cells and loss of epithelial architecture occurs in early lesions (tubal intraepithelial carcinoma (TIC)), finally leading to invasive cancer. <em>Adapted from Bowtell et al. (Bowtell 2011)</em>.
</div>

Genomic characterization of HGS-OvCa tumor samples has found a near universal inactivation of the p53 tumour suppressor pathway and half of these cancers are defective for the homologous recombination DNA repair pathway (Ahmed 2010). However, the high prevalence of such alterations is the exception, as large-scale expression, copy number analyses and mutational screens have failed to identify recurrent 'druggable' targets. This partly explains the finding that single-agent, molecularly targeted therapies have yielded but incremental benefits for HGS-OvCa in the clinic.

It is clear that one of the most pressing challenges for the management and treatment of HGS-OvCa is a better understanding of the nature of cellular and molecular deregulation that underlie the genesis, progression and resistance of this cancer to therapy. This motivates the TCGA effort to characterize HGS-OvCa.

#### Tissue samples

The TCGA study collected stage II-IV, clinically annotated HGS-OvCa and matched normal samples (489 total). Patients were selected to match the demographic and epidemiological attributes of the typical population diagnosed with disease. Patients had all been treated with platinum and most with taxane prior to surgery.

#### Gene expression subtypes

Previous work by Tothill *et al.* (Tothill 2008) was aimed at performing molecular subtype analysis of 285 well-annotated invasive ovarian, fallopian tube and peritoneal cancers. An unsupervised clustering analysis of gene expression revealed the presence of six robust molecular subtypes (C1 to C6) with discernible differences in malignant potential and grade.

The TCGA analysis of HGS-OvCa described four expression subtypes based on unsupervised clustering (Table 1). These subtypes were named according to the overall theme of genes within each cluster and from incorporating the previous descriptions by Tothill *et al*.

**Table 1. HGS-OvCa gene expression subtypes**

{:.table .table-hover .table-condensed .table-responsive}
| Expression Subtype | Gene Content |
|:-------|:------|:------|:-----|
| Immunoreactive | T-cell chomokine ligands *CXCL11* and *CXCL10* and the receptor CXCR3 |
| Mesenchymal |  High expression of *HOX* genes and increased stromal components such as *FAP*, *ANGPTL2* and *ANGPTL1* |
| Proliferative | High expression of *HMGA2* and *SOX11* and low expression of proliferation markers *MCM2* and *PCNA* |
| Differentiated | High levels of *MUC16* and *MUC1* along with the fallopian tube marker *SLP1* |

A follow-up study by Verhaak *et al.* (Verhaak 2013) used the four expression subtypes as a basis to associate mutation and patient survival data, suggesting that these expression categories have clinical and pathological relevance.

## <a href="#practical" name="practical">III. Practical</a>

A pair-wise differential gene expression analysis requires at least two pieces of information. The 'category information' (Table 2) assigns each sample to a group. This assignment is typically made *a priori* and depends upon the biological question of interest.

**Table 2. Category information**

{:.table .table-hover .table-condensed .table-responsive}
| Sample ID  |  Category  |
|:----------:|:----------:|
| Sample 1   |    A       |
| Sample 2   |    B       |
| Sample 3   |    B       |
| ...        |    ...     |
| Sample m   |    A       |

The second piece we need is the 'assay information' or experimental observations. In this case, this is the transcript counts for each gene and each sample (Table 3).

**Table 3. Assay information**

{:.table .table-hover .table-condensed .table-responsive}
|  Gene    | Sample 1 | Sample 2 |  ...     | Sample m  |
|:--------:|:--------:|:--------:|:--------:|:---------:|
| gene 1   |    0     |    200   |   ...    |    0     |
| gene 2   |    15    |    10    |   ...    |    2     |
|   ...    |   ...    |   ...    |   ...    |    ...   |
| gene n   |    8     |    0     |   ...    |    41    |

Below we describe how to obtain each of these pieces of information for the TCGA-OV project.

### Category information: Gene expression subtypes

Verhaak *et al.* used the TCGA HGS-OvCa data to generate a prognostic gene expression signature. In doing so they made available a [Supplementary Excel file 1](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3533304/bin/JCI65833sd1.xls) which contains Supplemental Table 1 that assigns each case a subtype (Table 1). The supplemental table includes dataset samples from other studies. We have filtered Supplemental Table 1 for the TCGA discovery cohort and provide it in <a href="{{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.data.subtype }}" download>`Verhaak_JCI_2013_tableS1.txt`</a>.

<a type="button" href="{{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.data.subtype }}" class="btn btn-success btn-lg btn-block" download><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> Subtypes (.txt)</a>

For the purposes of this guide, we will assume you have downloaded this file to the location `/home/TCGA/`. The general layout of the TCGA-OV category information can be seen in Table 3 for 489 cases.

**Table 3. TCGA subtype information**

{:.table .table-hover .table-condensed .table-responsive}
|        | ID       |  DATASET | SUBTYPE  |      AGE |      ...  |
|:------:|:--------:|:--------:|:--------:|:--------:|:---------:|
|   1    | TCGA-59-2348 | TCGA-discovery | Differentiated | 59 | ...|
|   2    | TCGA-24-1604 | TCGA-discovery | Differentiated | 67 | ... |
|   ...  | ... | ... | ... | ... | ... |
|   488  | TCGA-24-1560 | TCGA-discovery | Proliferative | 51 | ... |
|   489  | TCGA-04-1350 | TCGA-discovery | Proliferative | 46 | ... |

You will still see a lot of the [TCGA barcode](https://wiki.nci.nih.gov/display/TCGA/TCGA+barcode){:target="_blank"} (Figure 5) which are IDs assigned by Biospecimen Core Resource (BCR) centers in charge of handling patient samples. These human-readable TCGA barcodes are being phased out in favour of  [UUIDs](https://wiki.nci.nih.gov/display/TCGA/Universally+Unique+Identifier#transition){:target="_blank"} in the near future.

![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_5 }}){: .img-responsive }
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 5. The TCGA barcode.</strong>
</div>

### Assay information: RNA-seq data

Below we provide step-by-step instructions to obtain RNA-seq data. The GDC currently provides programmatic access to the repository via a [web service API](https://gdc-docs.nci.nih.gov/API/Users_Guide/Getting_Started/){:target="_blank"} however, this can be inefficient when attempting to download large numbers of files or data. A more robust method is to use the GDC [Data Transfer Tool](https://gdc-docs.nci.nih.gov/Data_Transfer_Tool/Users_Guide/Getting_Started/){:target="_blank"} which can be used via the command line terminal. The Data Transfer Tool provides many capabilities including resuming interrupted or failed downloads.

The GDC web service API and Data Transfer Tool offer precise and flexible control over what data is downloaded. Nevertheless, deriving useful data using these low-level tools requires data transformation and merging on the part of the user which can be laborious and error prone. Instead, we leverage the rich bioinformatics software ecosystem available in the  [R/Bioconductor](http://bioconductor.org/){:target="_blank"} suite that makes available powerful solutions for fetching, storing and transforming biological data in robust and interchangeable formats. In particular, we detail the use of the [TCGAbiolinks](https://www.bioconductor.org/packages/release/bioc/html/TCGAbiolinks.html){:target="_blank"} Bioconductor package.

#### Software requirements

**Run inside Docker** (*Recommended*). To ease the burden of loading the correct software and dependencies, we have generated a [Github repository](https://github.com/jvwong/docker_enrichment_workflow_gdc/tree/01442e4397c036c65da74563aa633f45d9a8117d){:target="_blank"} containing a the neccessary code to run a [Docker](https://www.docker.com/){:target="_blank"} version of [RStudio](https://www.rstudio.com/){:target="_blank"} linked to the necessary workflow files.

**Run code on your computer**. If you choose to run the R code on your own computer, you will need to take special note of [R/Bioconductor](https://bioconductor.org){: target="_blank"} versions. In particular, [TCGAbiolinks](https://bioconductor.org/packages/release/bioc/html/TCGAbiolinks.html) has recently undergone important changes to accommodate the move of TCGA data to the GDC web service API.

  - [R](https://www.r-project.org/){:target="_blank"}: >version 3.3.1
    - [Bioconductor](https://bioconductor.org){:target="_blank"}: >version 3.3
      - [TCGAbiolinks](https://www.bioconductor.org/packages/release/bioc/html/TCGAbiolinks.html){:target="_blank"}
      - [SummarizedExperiment](https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html){:target="_blank"}
      - [DEFormats](https://bioconductor.org/packages/release/bioc/html/DEFormats.html){:target="_blank"}

#### Data retrieval

The TCGAbiolinks package is effectively a thin wrapper around the GDC webservice API and Data Transfer Tool but provides additional algorithms and data structures that ease the association of data with metadata and compatibility with other Bioconductor analysis packages. Please view the vignette on [Working with TCGAbiolinks](https://www.bioconductor.org/packages/release/bioc/vignettes/TCGAbiolinks/inst/doc/tcgaBiolinks.html){:target="_blank"} for a nice description of the package capabilities.

Figure 6 summarizes the main steps that we will follow in to use the TCGAbiolinks package to download the TCGA-OV data from GDC.

![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_6 }}){: .img-responsive.short }
<div class="figure-legend well well-lg text-justify">
  <strong>Figure 6. GDC data retrieval.</strong> Steps required to fetch data from the GDC. Text to the right of arrows are functions provided as part of the R/Bioconductor packages. In step 1, the GDC is queried for the data of interest and this query (data frame) is used to instruct the file downloads in step 2. Note that data files are temporarily downloaded to the client computer. In step 3, the raw data are prepared as a SummarizedExperiment container (Huber 2015) then combined with the category information in the subtypes file into a DGEList from the edgeR package.
</div>

Below we detail the R commands used to retrieve the data. We will assume that you have downloaded the subtype file `Verhaak_JCI_2013_tableS1.txt`. Alternatively, you can find the subtypes file and all of the following code in our [github gist](https://gist.github.com/jvwong/293acd56bfc4181727f3832daed795b1). We will assume a directory structure on your computer that looks something like the following


```shell
home
|
|--- TCGA
|   |
|   |--- scripts
|   |    |
|   |    |--- get_data.R
|   |    ...
|   |
|   |--- data
|   |    |
|   |    |--- Verhaak_JCI_2013_tableS1.txt
|   |
|   |--- output
...
```


**Step 0: Load**

Load R/Bioconductor packages.

  {% highlight r %}
    {% github_sample jvwong/docker_enrichment_workflow_gdc/blob/e4735bc9d94b44259c432f11985cf0131629d53a/src/scripts/get_data.R 1 18 %}
  {% endhighlight %}

**Step 1: Query**

The `GDCquery(...)` function allows us to search the GDC for data based on the filter restrictions we pass. These restrictions mirror those provided by the GDC data portal and web service API. In this case we are looking for unnormalized RNA-seq counts ('HTSeq - Counts') for the TCGA-OV project. We store the search results in the `query` data frame.

{% highlight r %}
  {% github_sample jvwong/docker_enrichment_workflow_gdc/blob/e4735bc9d94b44259c432f11985cf0131629d53a/src/scripts/get_data.R 19 26 %}
{% endhighlight %}

**Step 2: Download**

The `GDCdownload(...)` function allows us to actually retrieve files based on the search results in step 1. We set `method=client` so that it uses the GDC Data Transfer Tool under the hood and declare a local directory (`TCGAOV_RNASEQ_DIR`) where the files should be saved.

  {% highlight r %}
    {% github_sample jvwong/docker_enrichment_workflow_gdc/blob/e4735bc9d94b44259c432f11985cf0131629d53a/src/scripts/get_data.R 28 32 %}
  {% endhighlight %}

> Note: These commands will trigger a download of the GDC Data Transfer Tool (gdc-client) and a manifest (gdc_manifest.txt) to your home directory.


>Note: There can be a large number of files downloaded. Downloading 379 files above took on the order of 15 minutes but your mileage may vary depending on your connection speed.

**Step 3: Prepare**

A bunch of data files isn't going to be much help to us. The goal of this step is to prepare the data by combining the data and metadata. `GDCprepare(...)` combines the downloaded file data into a single container, the [`SummarizedExperiment`](https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html){:target="_blank"} (Figure 7) (Huber 2015). We will also save this container to an RData file (.rda) and keep the downloaded files.

  {% highlight r %}
    {% github_sample jvwong/docker_enrichment_workflow_gdc/blob/e4735bc9d94b44259c432f11985cf0131629d53a/src/scripts/get_data.R 34 41 %}
  {% endhighlight %}

![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_7 }}){: .img-responsive }
<div class="figure-legend well well-lg text-justify">
<strong>Figure 7. SummarizedExperiment container.</strong> <strong>A. </strong> The SummarizedExperiment container of the TCGAbiolinks package is used to store rectangular matrices of experimental results. Each object stores features (<strong>B</strong>) and samples (<strong>C</strong>) corresponding to experimental data (<strong>D</strong>). A key aspect of the SummarizedExperiment class is the coordination of the meta-data and assays when subsetting. Shown are the functions used to access the data.
</div>

**Step 4: Integrate**

The `SummarizedExperiment` object containing the RNA-seq assay data and associated metadata for the TCGA-OV samples is now available. The differential gene expression analysis prescribed by [edgeR](http://bioconductor.org/packages/release/bioc/html/edgeR.html){:target="_blank"} uses a `DGEList` container to house both the assay and category information.

To this end, we use the `DEFormats` package function `DEGList(...)` to convert our SummarizedExperiment and integrate the subtypes. Note that we need to find the samples with an assigned subtype and subset accordingly.

  {% highlight r %}
    {% github_sample jvwong/docker_enrichment_workflow_gdc/blob/e4735bc9d94b44259c432f11985cf0131629d53a/src/scripts/get_data.R 42 64 %}
  {% endhighlight %}

The resulting `DGEList` called `TCGAOV_data` is saved as an RData file `TCGAOV_data.rda` for use in downstream differential expression analysis.

  {% highlight r %}
    {% github_sample jvwong/docker_enrichment_workflow_gdc/blob/e4735bc9d94b44259c432f11985cf0131629d53a/src/scripts/get_data.R 64 66 %}
  {% endhighlight %}

<hr/>

You should have a directory structure similar to the following

```shell
home
|
|--- TCGA
|   |
|   |--- scripts
|   |    |
|   |    |--- get_data.R
|   |    ...
|   |
|   |--- data
|   |    |
|   |    |--- Verhaak_JCI_2013_tableS1.txt
|   |
|   |--- output
|   |    |
|   |    |--- TCGAOV_data.rda
|   |    |--- TCGAOV_se_datahtseq_counts.rda
...
```

<hr/>

The preceding R code is presented in its entirety and available as a Github repo
<a href="https://github.com/jvwong/docker_enrichment_workflow_gdc/blob/master/src/scripts/get_data.R"
  target="_blank">
  <i class="fa fa-github fa-2x"></i>
</a>

{% highlight r %}
  {% github_sample jvwong/docker_enrichment_workflow_gdc/blob/e4735bc9d94b44259c432f11985cf0131629d53a/src/scripts/get_data.R %}
{% endhighlight %}

## <a href="#data" name="data">IV. Data</a>

File description: The TCGA-OV RNA-seq expression data (counts) and subtypes (group) were inserted into an edgeR DGEList variable `TCGAOV_data` and saved to an RData file named `TCGAOV_data.rda`.

<a href="{{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.data.tcgaov_data }}" type="button" class="btn btn-success btn-lg btn-block" download><span class="glyphicon glyphicon-download-alt" aria-hidden="true"></span> TCGAOV_data.rda</a>

The R code and subtype designation file are available in Github <a href="https://github.com/jvwong/docker_enrichment_workflow_gdc/blob/master/src/scripts/get_data.R"
  target="_blank">
  <i class="fa fa-github fa-2x"></i>
</a>

## <a href="#references" name="references">V. References</a>
<div class="panel_group" data-inline="25633503,21720365,20229506,26493647,23104886,20022975,18698038,21941283,23257362,20802226,21436879,12529460"></div>
<!-- - Huber W et al. Orchestrating high-throughput genomic analysis with Bioconductor. Nat. Methods vol. 12 (2015)
- Cancer Genome Atlas Research Network. Integrated genomic analyses of ovarian carcinoma. Nature vol. 474 (2011)
- Ahmed A et al. Driver mutations in TP53 are ubiquitous in high grade serous carcinoma of the ovary. J. Pathol. vol. 221 (2010)
- Bowtell D et al. Rethinking ovarian cancer II: reducing mortality from high-grade serous ovarian cancer. Nat. Rev. Cancer vol. 15 (2015)
- Dobin A et al. STAR: ultrafast universal RNA-seq aligner. Bioinformatics vol. 29 (2013)
- Li B et al. RNA-Seq gene expression estimation with read mapping uncertainty. Bioinformatics vol. 26 (2010)
- Tothill R et al. Novel molecular subtypes of serous and endometrioid ovarian cancer linked to clinical outcome. Clin. Cancer Res. vol. 14 (2008)
- Vaughan S et al. Rethinking ovarian cancer: recommendations for improving outcomes. Nat. Rev. Cancer vol. 11 (2011)
AnimalsAntineoplastic Agents
- Verhaak R et al. Prognostically relevant gene signatures of high-grade serous ovarian carcinoma. J. Clin. Invest. vol. 123 (2013)
- Wang K et al. MapSplice: accurate mapping of RNA-seq reads for splice junction discovery. Nucleic Acids Res. vol. 38 (2010)
- Wang X et al. Unifying gene expression measures from multiple platforms using factor analysis. PLoS ONE vol. 6 (2011)
- Zhang L et al. Intratumoral T cells, recurrence, and survival in epithelial ovarian cancer. N. Engl. J. Med. vol. 348 (2003) -->
