<!DOCTYPE html>

<html lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      RNA-Seq to Enrichment Map - R Notebook &middot; Guide
    
  </title>

  <!-- Icons -->
  <link rel="shortcut icon" href="/guide/media/favicons/favicon.ico">
  <link rel="apple-touch-icon" sizes="180x180" href="/guide/media/favicons/apple-touch-icon.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/guide/media/favicons/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/guide/media/favicons/favicon-16x16.png">
  <link rel="manifest" href="/guide/media/favicons/manifest.json">
  <meta name="theme-color" content="#ffffff">

  <!-- CSS -->
  
  <!-- Third-party CSS -->
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/bootstrap/dist/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/progress-tracker/app/styles/progress-tracker.css" />
  <link rel="stylesheet" type="text/css" href="/guide/public/bower_components/font-awesome/css/font-awesome.min.css" />

  <!-- Custom CSS -->
  <link rel="stylesheet" href="/guide/public/css/main.css">

  <!-- Javascript -->
  <!-- Third-party javascript -->
  <script type="text/javascript" src="/guide/public/bower_components/jquery/dist/jquery.min.js"></script>
  <script type="text/javascript" src="/guide/public/bower_components/bootstrap/dist/js/bootstrap.min.js"></script>
  <script type="text/javascript" src="/guide/public/bower_components/cytoscape/dist/cytoscape.min.js"></script>
  <script type="text/javascript" src="/guide/public/js/deps.js"></script>
  <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
    MathJax.Hub.Config({ TeX: { equationNumbers: { autoNumber: "AMS" } } });
  </script>
  <!-- Custom javascript -->
  <script type="text/javascript" src="/guide/public/js/babel-compiled.js" defer></script>
  
</head>


  <body class="theme-base-01">
    <a name="top"></a>

    <!-- Target for toggling the sidebar `.sidebar-checkbox` is for regular
     styles, `#sidebar-checkbox` for behavior. -->
<input type="checkbox" class="sidebar-checkbox" id="sidebar-checkbox">

<!-- Toggleable sidebar -->
<div class="sidebar" id="sidebar">
  <nav class="sidebar-nav">
    <div class="sidebar-item">
      Guide: v0.1.0
    </div>
    <a class="sidebar-nav-item" href="/guide/">Home</a>

    

    
    
      <!-- List items with a valid title and page layout -->
      
        

          
          <a class="sidebar-nav-item"
          href="/guide/case_studies/archive/">Case studies
          </a>

        
      
    
      <!-- List items with a valid title and page layout -->
      
    
      <!-- List items with a valid title and page layout -->
      
        

          
          <a class="sidebar-nav-item"
          href="/guide/presentations/archive/">Presentations
          </a>

        
      
    
      <!-- List items with a valid title and page layout -->
      
        

          
          <a class="sidebar-nav-item"
          href="/guide/primers/archive/">Primers
          </a>

        
      
    
      <!-- List items with a valid title and page layout -->
      
        

          
          <a class="sidebar-nav-item active"
          href="/guide/workflows/archive/">Workflows
          </a>

        
      
    
  </nav>
</div>


    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
         content to avoid any CSS collisions with our real content. -->
    <div class="wrap">
    <div class="masthead">
  <div class="container">
    <h3 class="masthead-title">
      <a href="/guide/" title="Home">Guide</a>
      <small>
        <a href="http://www.pathwaycommons.org/">&middot; Pathway Commons</a>
        
      </small>
      <span class="pull-right">
        <small class="contact">
          <a href="https://groups.google.com/forum/#!forum/pathway-commons-help" target="_blank">Contact Us</a>
        </small>
      </span>
    </h3>
  </div>
</div>


      <div class="container content">
        <div id='ajax-spinner'><img src="/guide/media/spinner.gif"/></div>
        <div class="page">
    
  
    

    
    
    <ol class="breadcrumb">
    
      

      <!-- we can only handle 1 nesting -->
      
        <li><a href="/guide/">Guide</a></li>
      
    
      

      <!-- we can only handle 1 nesting -->
      
        <li><a href="/guide/workflows/archive/">Workflows</a></li>
      
    
      

      <!-- we can only handle 1 nesting -->
      
    
      

      <!-- we can only handle 1 nesting -->
      
        
          <li class="active">RNA-Seq to Enrichment Map - R Notebook</li>
        
      
    
    </ol>
  


  

  <h1 class="page-title">RNA-Seq to Enrichment Map - R Notebook</h1>
  <div class="collection">
  <div class="document">
    <div class="subtitle">Process platelet RNA-Seq data, identify altered pathways then visualize using Enrichment Map</div>
    <hr/>
    <div class="linkout clearfix">
      <ul class="list-inline pull-right">
      <li>
  <a href="http://twitter.com/intent/tweet?text=Process platelet RNA-Seq data, identify altered pathways then visualize ...+http://pathwaycommons.github.io/guide/workflows/rna_seq_to_enrichment_map_r_notebook/index.html&via=pathwaycommons"
  data-toggle="tooltip"
  title="Tweet this page"
  target="_blank" >
    <i class="fa fa-twitter-square fa-2x"></i>
  </a>
</li>
<li>
  <a href="https://plus.google.com/share?url=http://pathwaycommons.github.io/guide/workflows/rna_seq_to_enrichment_map_r_notebook/index.html"
  data-toggle="tooltip"
  title="Share on Google+"
  target="_blank" >
    <i class="fa fa-google-plus fa-2x"></i>
  </a>
</li>

      <li></li>
      
        <li>
  <a data-toggle="tooltip"
    title="Download"
    href="/guide/media/workflows/rna_seq_to_enrichment_map_r_notebook/index/rna_seq_to_enrichment_map_r_notebook.zip">
    <i class="fa fa-download fa-2x" aria-hidden="true"></i>
  </a>
</li>

      
      </ul>
    </div>
    <hr/>
    <div class="alert alert-info">
  This is an <a href="http://rmarkdown.rstudio.com/r_notebooks.html" target="_blank">R Notebook</a> companion to the pathway enrichment analysis workflow 'RNA-Seq to Enrichment Map'. This notebook and file dependencies are available for downloaded via the <i class="fa fa-download" aria-hidden="true"></i> link (top, right).
</div>

<h2 id="table-of-contents">Table of Contents</h2>

<ul>
  <li><a href="#overview">A. Overview</a></li>
  <li><a href="#sampleStudy">B. Sample Study</a></li>
  <li><a href="#dataPreProcessing">C. Data Pre-Processing</a></li>
  <li><a href="#differentialExpressionTesting">D. Differential Expression Testing</a></li>
  <li><a href="#geneSetEnrichmentAnalysis">E. Gene Set Enrichment Analysis</a></li>
  <li><a href="#enrichmentMap">F. Enrichment Map</a></li>
</ul>

<h2 id="a-overview"><a href="#overview" name="overview">A. Overview</a></h2>

<div class="alert alert-warning">
  <p>Section TODOs</p>
  <ul><li>Install software and package dependencies</li></ul>
</div>

<p>This R Notebook documents a comparison mRNA levels between two conditions and uses this information to identify and then visualize pathway-level differences. In particular, you will use convert RNA-Seq count data into a single ranked list where genes are ordered according to their differential expression. Enriched pathways from this list are distilled using <a href="http://software.broadinstitute.org/gsea/index.jsp">Gene Set Enrichment Analysis (GSEA)</a> then visualized as a <a href="http://www.cytoscape.org/">Cytoscape</a> <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0013984">Enrichment Map</a>.</p>

<h3 id="software-requirements">Software requirements</h3>

<p>We will be using the following software and packages along the way.</p>

<ul>
  <li><a href="https://www.r-project.org/">R</a> (&gt;= 3.2.5)
    <ul>
      <li><a href="https://bioconductor.org/">BiocInstaller</a> (&gt;=1.24.0)</li>
      <li><a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html">SummarizedExperiment</a> (&gt;= 1.2.3)</li>
      <li><a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> (&gt;= 3.14.0)</li>
      <li><a href="https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html">GenomicRanges</a> (&gt;= 1.26.4)</li>
      <li><a href="https://bioconductor.org/packages/release/bioc/html/IRanges.html">IRanges</a>	(&gt;= 2.8.2)</li>
      <li><a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html">biomaRt</a> (&gt;= 2.30.0)</li>
      <li><a href="https://cran.r-project.org/web/packages/devtools/README.html">devtools</a> (&gt;=1.13.3)</li>
      <li><a href="https://github.com/cytoscape/cytoscape-automation/tree/78c068f6293630693682a40d371646013343ae37/for-scripters/R/r2cytoscape">r2cytoscape</a> (Commit 78c068f629)</li>
    </ul>
  </li>
  <li><a href="https://www.java.com/en/">Java</a> (v8)
    <ul>
      <li><a href="http://software.broadinstitute.org/gsea/index.jsp">Gene Set Enrichment Analysis (GSEA)</a> - Java Jar file (3.0)</li>
      <li><a href="http://www.cytoscape.org/">Cytoscape</a> (3.5.1)</li>
    </ul>
  </li>
</ul>

<h4 id="gsea">GSEA</h4>

<p>After <a href="http://software.broadinstitute.org/gsea/register.jsp">registering</a>, <a href="http://software.broadinstitute.org/gsea/login.jsp">login</a> then download the GSEA Java Jar file (e.g. <code class="highlighter-rouge">gsea-3.0.jar</code>). For the purposes of this notebook, we will assume the GSEA path is <code class="highlighter-rouge">/Users/username/bin/gsea-3.0.jar</code>.</p>

<h4 id="cytoscape">Cytoscape</h4>

<p>After installing Cytoscape you’ll need to load in the Enrichment Map app. Do this through the <code class="highlighter-rouge">Apps -&gt; App Manager</code> in the toolbar.</p>

<ul>
  <li><a href="http://apps.cytoscape.org/apps/enrichmentmap">EnrichmentMap</a> (3.0.0)</li>
</ul>

<p>For subsequent clustering and automated labelling of gene set themes, you should consider installing the following apps:</p>

<ul>
  <li><a href="http://apps.cytoscape.org/apps/clustermaker2">Clustermaker2</a> (1.1.0)</li>
  <li><a href="http://apps.cytoscape.org/apps/wordcloud">WordCloud</a> (3.1.0)</li>
  <li><a href="http://apps.cytoscape.org/apps/autoannotate">AutoAnnotate</a> (1.1.0)</li>
</ul>

<h2 id="b-sample-study"><a href="#sampleStudy" name="sampleStudy">B. Sample Study</a></h2>

<div class="alert alert-warning">
  <p>Section To Dos</p>
  <ul><li>Familiarize yourself with study data and related files</li></ul>
</div>

<p>In this workflow we will use expression data collected in a study by Myron G. Best and colleagues <sup id="fnref:1"><a href="#fn:1" class="footnote">1</a></sup> whose aim was to differentiate blood platelets from healthy donors (HD) to those diagnosed with a breast cancer (BrCa) towards a proof-of-principle for blood-based cancer diagnosis.</p>

<h3 id="rationale">Rationale</h3>

<p>The primary physiological role of platelets is to sense and accumulate at the sites of damaged endothelial tissue and initiate a blood clot to mitigate and vessel leakage<sup id="fnref:2"><a href="#fn:2" class="footnote">2</a></sup>. Previous observations are consistent with the notion that platelets have close contact with cancer cells - a process termed ‘education’ - and may play a role in their metastatic potential<sup id="fnref:3"><a href="#fn:3" class="footnote">3</a></sup>. Diversity in tumour-educated platelets (TEP) could be clinically relevant if they enable discrimination between different stages of a malignancy.</p>

<h3 id="rna-seq-data">RNA-Seq Data</h3>

<p>Best <em>et al.</em> collected blood platelets from 55 HD and from 39 individuals with BrCa and subjected these samples to RNA-sequencing. Herein, we will be restricting our attention to a subset of 5 patient samples from each class. The metadata for the RNA-Seq files is contained in a file named <code class="highlighter-rouge">tep_rnaseq_metadata.txt</code> which lists RNA-Seq data filenames (‘id’) and the ‘class’ of each patient sample (Table I).</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="c1">### Declare general file directory paths</span><span class="w">
  </span><span class="n">base_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">getwd</span><span class="p">()</span><span class="w">
  </span><span class="n">data_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">base_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"data"</span><span class="p">)</span><span class="w">
  </span><span class="n">output_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">base_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"output"</span><span class="p">)</span><span class="w">
  </span><span class="n">dir.create</span><span class="p">(</span><span class="n">output_dir</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">

  </span><span class="c1">### Declare paths to RNA-Seq (meta)data files</span><span class="w">
  </span><span class="n">tep_rnaseq_metadata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"tep_rnaseq_metadata.txt"</span><span class="p">)</span><span class="w">
  </span><span class="n">tep_rnaseq_filelist</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">file.path</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"MGH-BrCa-H-74_htsqct.txt"</span><span class="p">),</span><span class="w">
                           </span><span class="n">file.path</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"MGH-BrCa-H-68_htsqct.txt"</span><span class="p">),</span><span class="w">
                           </span><span class="n">file.path</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"MGH-BrCa-H-66_htsqct.txt"</span><span class="p">),</span><span class="w">
                           </span><span class="n">file.path</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"MGH-BrCa-H-59_htsqct.txt"</span><span class="p">),</span><span class="w">
                           </span><span class="n">file.path</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"MGH-BrCa-H-11_htsqct.txt"</span><span class="p">),</span><span class="w">
                           </span><span class="n">file.path</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"HD-5_htsqct.txt"</span><span class="p">),</span><span class="w">
                           </span><span class="n">file.path</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"HD-4_htsqct.txt"</span><span class="p">),</span><span class="w">
                           </span><span class="n">file.path</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"HD-3-1_htsqct.txt"</span><span class="p">),</span><span class="w">
                           </span><span class="n">file.path</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"HD-2-1_htsqct.txt"</span><span class="p">),</span><span class="w">
                           </span><span class="n">file.path</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"HD-1_htsqct.txt"</span><span class="p">))</span></code></pre></figure>

<p><strong>Table I. Patient TEP RNA-Seq metadata (<code class="highlighter-rouge">tep_rnaseq_metadata.txt</code>)</strong></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">tep_rnaseq_metadata_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="n">tep_rnaseq_metadata</span><span class="p">,</span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">tep_rnaseq_metadata_df</span><span class="p">)</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left">id</th>
      <th style="text-align: left">class</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-74_htsqct.txt</td>
      <td style="text-align: left">BrCa</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-68_htsqct.txt</td>
      <td style="text-align: left">BrCa</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-66_htsqct.txt</td>
      <td style="text-align: left">BrCa</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-59_htsqct.txt</td>
      <td style="text-align: left">BrCa</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-11_htsqct.txt</td>
      <td style="text-align: left">BrCa</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-5_htsqct.txt</td>
      <td style="text-align: left">HD</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-4_htsqct.txt</td>
      <td style="text-align: left">HD</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-3-1_htsqct.txt</td>
      <td style="text-align: left">HD</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-2-1_htsqct.txt</td>
      <td style="text-align: left">HD</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-1_htsqct.txt</td>
      <td style="text-align: left">HD</td>
    </tr>
  </tbody>
</table>

<p>Each RNA-Seq measurement is represented as a tabular text file that, in this instance, contains 57 736 rows: The first column holds an <a href="http://www.ensembl.org/info/genome/genebuild/genome_annotation.html">Ensembl ID</a> representing a region of the human genome and the second column a read count for transcripts mapped to that Ensembl ID. All of this is typically taken care of by your sequencing facility. An excerpt of raw data for sample <code class="highlighter-rouge">HD-1_htsqct.txt</code> is shown below in Table II.</p>

<p><strong>Table II. Excerpt of RNA-Seq file HD-1_htsqct.txt</strong></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">index_HD_1_htsqct</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">grepl</span><span class="p">(</span><span class="s2">"HD-1_htsqct.txt"</span><span class="p">,</span><span class="w"> </span><span class="n">tep_rnaseq_filelist</span><span class="p">)</span><span class="w">
  </span><span class="n">rnaseq_HD_1_htsqct</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="n">tep_rnaseq_filelist</span><span class="p">[</span><span class="n">index_HD_1_htsqct</span><span class="p">],</span><span class="w">
    </span><span class="n">check.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">head</span><span class="p">(</span><span class="n">rnaseq_HD_1_htsqct</span><span class="p">),</span><span class="w"> </span><span class="n">col.names</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s2">""</span><span class="p">,</span><span class="s2">""</span><span class="p">))</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left"> </th>
      <th style="text-align: right"> </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">ENSG00000000003</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">ENSG00000000005</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">ENSG00000000419</td>
      <td style="text-align: right">100</td>
    </tr>
    <tr>
      <td style="text-align: left">ENSG00000000457</td>
      <td style="text-align: right">6</td>
    </tr>
    <tr>
      <td style="text-align: left">ENSG00000000460</td>
      <td style="text-align: right">11</td>
    </tr>
    <tr>
      <td style="text-align: left">ENSG00000000938</td>
      <td style="text-align: right">159</td>
    </tr>
  </tbody>
</table>

<h2 id="c-data-pre-processing"><a href="#dataPreProcessing" name="dataPreProcessing">C. Data Pre-Processing</a></h2>

<div class="alert alert-warning">
  <p>Section To Dos</p>
  <ul><li>Perform RNA-Seq data gene ID mapping</li>
  <li>Merge RNA-Seq data into a single R object</li></ul>
</div>

<p>At the end of this section we wish to have the RNA-Seq data into an R data strcuture that we can more easily pipe into our differential gene expression testing procedures. Because of the numerous formats and states that raw RNA-Seq data can present itself, you may require a customized approach.</p>

<h3 id="gene-identifier-mapping">Gene identifier mapping</h3>

<p>Our pathway enrichment analysis in GSEA identifies gene sets that are enriched in gene expression data. We will be using candidate gene sets named with the <a href="http://www.genenames.org/">HGNC</a> approved symbol. Since our RNA-Seq data use Ensembl identifiers (Table II) we will need to ‘map’ these to their corresponding HGNC symbol. If it is not possible to perform this mapping, we simply omit the gene. Also, beware of duplicates.</p>

<h3 id="rna-seq-data-merging">RNA-Seq data merging</h3>

<p>We will take the individual RNA-Seq data files and merge them and any associated metadata into a single object, the <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html">SummarizedExperiment</a><sup id="fnref:4"><a href="#fn:4" class="footnote">4</a></sup>.</p>

<blockquote>
  <p>A SummarizedExperiment stores experimental assays as a rectangular array whose rows correspond to the (genomic) ranges and whose columns correspond to the different samples. The SummarizedExperiment class stores metadata on the rows and columns. Metadata on the samples usually include experimental or observational covariates … Row metadata comprise the start and end coordinates of each feature and the identifier of the containing polymer, for example, the chromosome name.</p>
</blockquote>

<hr />

<p>We offload the above work onto an R function <code class="highlighter-rouge">merge_data</code> and its helpers that perform the required ID-mapping and data merging and returns a single <code class="highlighter-rouge">SummarizedExperiment</code> instance.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">library</span><span class="p">(</span><span class="s2">"SummarizedExperiment"</span><span class="p">)</span><span class="w">
  
  </span><span class="cd">#' Merge a set of files representing RNA sequencing gene counts</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' 1. Default parameter is do nothing (already matching gene set ids)</span><span class="w">
  </span><span class="cd">#' 2. Optional parameter is to map from input to target namespace</span><span class="w">
  </span><span class="cd">#' Data files must be in tab-delimited format with two columns and no header. The classes are defined by a meta data file which is a tab-delimited text file with headers for the sample read 'id' and 'class'. Each row entry is a corresponding filename and class assignment. Only accepts pair-wise comparison so there must be exactly 2 classes.</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' @param metadata_file A metadata file</span><span class="w">
  </span><span class="cd">#' @param species A character array indicating the species with which to fetch gene models from bioMart, If NULL this mapping will not be performed</span><span class="w">
  </span><span class="cd">#' @param source_name attribute (gene namespace) input</span><span class="w">
  </span><span class="cd">#' @param target_name attribute (gene namespace) desired</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' @return A \code{\link[SummarizedExperiment]{SummarizedExperiment}}</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' @export</span><span class="w">
  </span><span class="n">merge_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">metadata_file</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="p">,</span><span class="w"> </span><span class="n">source_name</span><span class="p">,</span><span class="w"> </span><span class="n">target_name</span><span class="p">,</span><span class="w"> </span><span class="n">...</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">

    </span><span class="k">if</span><span class="p">(</span><span class="o">!</span><span class="nf">is.character</span><span class="p">(</span><span class="n">species</span><span class="p">))</span><span class="w"> </span><span class="n">stop</span><span class="p">(</span><span class="s1">'species must be of class character'</span><span class="p">)</span><span class="w">

    </span><span class="n">i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0</span><span class="w">
    </span><span class="n">class_order</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">()</span><span class="w">
    </span><span class="n">meta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">create_meta</span><span class="p">(</span><span class="n">metadata_file</span><span class="p">)</span><span class="w">

    </span><span class="n">filelist</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">...</span><span class="p">)</span><span class="w">
    </span><span class="k">if</span><span class="p">(</span><span class="nf">is.list</span><span class="p">(</span><span class="n">filelist</span><span class="p">)){</span><span class="w">
      </span><span class="n">filelist</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">unlist</span><span class="p">(</span><span class="n">filelist</span><span class="p">)</span><span class="w">
    </span><span class="p">}</span><span class="w">

    </span><span class="k">if</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">meta</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">filelist</span><span class="p">))</span><span class="w"> </span><span class="n">stop</span><span class="p">(</span><span class="s1">'Mismatch in files declared in metadata'</span><span class="p">)</span><span class="w">

    </span><span class="k">for</span><span class="p">(</span><span class="n">file</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="n">filelist</span><span class="p">){</span><span class="w">

      </span><span class="k">if</span><span class="p">(</span><span class="o">!</span><span class="n">file.exists</span><span class="p">(</span><span class="n">file</span><span class="p">))</span><span class="w"> </span><span class="n">stop</span><span class="p">(</span><span class="s1">'invalid file/directory'</span><span class="p">)</span><span class="w">

      </span><span class="n">fname</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">basename</span><span class="p">(</span><span class="n">file</span><span class="p">)</span><span class="w">
      </span><span class="n">findex</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">which</span><span class="p">(</span><span class="n">meta</span><span class="o">$</span><span class="n">id</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">fname</span><span class="p">)</span><span class="w">

      </span><span class="k">if</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">findex</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="n">stop</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s1">'Could not find match in metadata for '</span><span class="p">,</span><span class="w"> </span><span class="n">fname</span><span class="p">))</span><span class="w">

      </span><span class="n">class_order</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">append</span><span class="p">(</span><span class="n">class_order</span><span class="p">,</span><span class="w"> </span><span class="n">which</span><span class="p">(</span><span class="n">meta</span><span class="o">$</span><span class="n">id</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">fname</span><span class="p">))</span><span class="w">

      </span><span class="n">input_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="n">file</span><span class="p">,</span><span class="w">
        </span><span class="n">check.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
        </span><span class="n">stringsAsFactors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
        </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w">
        </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">,</span><span class="w">
        </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
      </span><span class="n">colnames</span><span class="p">(</span><span class="n">input_df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">tools</span><span class="o">::</span><span class="n">file_path_sans_ext</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span><span class="w">

      </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="p">){</span><span class="w">
        </span><span class="n">data_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">input_df</span><span class="w">
        </span><span class="n">i</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">i</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">
        </span><span class="k">next</span><span class="p">()</span><span class="w">
      </span><span class="p">}</span><span class="w">

      </span><span class="n">data_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merge</span><span class="p">(</span><span class="n">data_df</span><span class="p">,</span><span class="w">
        </span><span class="n">input_df</span><span class="p">,</span><span class="w">
        </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"row.names"</span><span class="p">,</span><span class="w">
        </span><span class="n">all</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
      </span><span class="n">rownames</span><span class="p">(</span><span class="n">data_df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data_df</span><span class="o">$</span><span class="n">Row.names</span><span class="w">
      </span><span class="n">data_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">subset</span><span class="p">(</span><span class="n">data_df</span><span class="p">,</span><span class="w"> </span><span class="n">select</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">Row.names</span><span class="p">)</span><span class="w">
      </span><span class="n">i</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">i</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">
    </span><span class="p">}</span><span class="w">

    </span><span class="n">gene_model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">get_gene_model</span><span class="p">(</span><span class="w"> </span><span class="n">data_df</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="p">,</span><span class="w"> </span><span class="n">source_name</span><span class="p">,</span><span class="w"> </span><span class="n">target_name</span><span class="w"> </span><span class="p">)</span><span class="w">
    </span><span class="k">if</span><span class="p">(</span><span class="nf">is.null</span><span class="p">(</span><span class="n">gene_model</span><span class="p">))</span><span class="w"> </span><span class="n">stop</span><span class="p">(</span><span class="w"> </span><span class="s1">'Could not reliably map input gene ids'</span><span class="w"> </span><span class="p">)</span><span class="w">

    </span><span class="c1"># Sync up data rows (name, order) with gene_model returned</span><span class="w">
    </span><span class="n">data_df_mapped</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">map_names</span><span class="p">(</span><span class="n">data_df</span><span class="p">,</span><span class="w"> </span><span class="n">gene_model</span><span class="p">,</span><span class="w"> </span><span class="n">source_name</span><span class="p">)</span><span class="w">

    </span><span class="c1"># Create the SummarizedExperiment</span><span class="w">
    </span><span class="n">colData</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">class</span><span class="o">=</span><span class="n">meta</span><span class="p">[</span><span class="n">class_order</span><span class="p">,]</span><span class="o">$</span><span class="n">class</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="n">colnames</span><span class="p">(</span><span class="n">data_df_mapped</span><span class="p">))</span><span class="w">

    </span><span class="n">data_se</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">SummarizedExperiment</span><span class="o">::</span><span class="n">SummarizedExperiment</span><span class="p">(</span><span class="w">
      </span><span class="n">assays</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data.matrix</span><span class="p">(</span><span class="n">data_df_mapped</span><span class="p">)),</span><span class="w">
      </span><span class="n">rowRanges</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gene_model</span><span class="p">,</span><span class="w">
      </span><span class="n">colData</span><span class="o">=</span><span class="n">colData</span><span class="p">)</span><span class="w">

    </span><span class="nf">return</span><span class="p">(</span><span class="n">data_se</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">

  </span><span class="cd">#' Create a data frame housing the RNA-seq metadata</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' This function requires a tab-delimited text file with headers for</span><span class="w">
  </span><span class="cd">#' the sample read 'id' and 'class'. Each row entry is a corresponding filename</span><span class="w">
  </span><span class="cd">#' and class assignment. Only accepts pair-wise comparison so there must be</span><span class="w">
  </span><span class="cd">#' exactly 2 classes.</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' @param metadata_file a path to a tab-delimited metadata file</span><span class="w">
  </span><span class="cd">#' @return A data frame</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' @export</span><span class="w">
  </span><span class="n">create_meta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">metadata_file</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">

    </span><span class="k">if</span><span class="p">(</span><span class="o">!</span><span class="n">file.exists</span><span class="p">(</span><span class="n">metadata_file</span><span class="p">))</span><span class="w"> </span><span class="n">stop</span><span class="p">(</span><span class="s1">'file does not exist'</span><span class="p">)</span><span class="w">
    </span><span class="k">if</span><span class="p">(</span><span class="o">!</span><span class="n">grepl</span><span class="p">(</span><span class="s2">".txt$"</span><span class="p">,</span><span class="w"> </span><span class="n">metadata_file</span><span class="p">)){</span><span class="w">
      </span><span class="n">stop</span><span class="p">(</span><span class="s2">"Uploaded file must be a tab-delimited .txt file!"</span><span class="p">)</span><span class="w">
    </span><span class="p">}</span><span class="w">

    </span><span class="n">meta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="n">metadata_file</span><span class="p">,</span><span class="w">
      </span><span class="n">check.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
      </span><span class="n">colClasses</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"character"</span><span class="p">,</span><span class="w"> </span><span class="s2">"factor"</span><span class="p">),</span><span class="w">
      </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">

    </span><span class="k">if</span><span class="p">(</span><span class="o">!</span><span class="n">all.equal</span><span class="p">(</span><span class="n">colnames</span><span class="p">(</span><span class="n">meta</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"id"</span><span class="p">,</span><span class="w"> </span><span class="s2">"class"</span><span class="p">)))</span><span class="w"> </span><span class="n">stop</span><span class="p">(</span><span class="s1">'check column headers'</span><span class="p">)</span><span class="w">    

    </span><span class="nf">return</span><span class="p">(</span><span class="n">meta</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">

  </span><span class="cd">#' Normalize the data.frame nameswith those on a \code{\link[GenomicRanges]{GRanges}} object</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' !!!!!!!!!!!!Alert alert alert ---- mapping between namespaces is not bijective.!!!!!!!!!!!!!</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' Assumes that the gene_model has a meta-data column with source_name</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' @param data_df the data frame to synchronize</span><span class="w">
  </span><span class="cd">#' @param gene_model the \code{\link[GenomicRanges]{GRanges}} object to match rows with</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' @return A \code{\link[base]{data.frame}} with the same rows and names as the input \code{\link[GenomicRanges]{GRanges}}</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' @export</span><span class="w">
  </span><span class="n">map_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">data_df</span><span class="p">,</span><span class="w"> </span><span class="n">gene_model</span><span class="p">,</span><span class="w"> </span><span class="n">source_name</span><span class="p">){</span><span class="w">

    </span><span class="c1"># Filter data_df for source_name</span><span class="w">
    </span><span class="n">indices_data_df_source</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="w"> </span><span class="n">data_df</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">GenomicRanges</span><span class="o">::</span><span class="n">mcols</span><span class="p">(</span><span class="w"> </span><span class="n">gene_model</span><span class="w"> </span><span class="p">)[[</span><span class="n">source_name</span><span class="p">]]</span><span class="w">
    </span><span class="n">subset_data_df_source</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data_df</span><span class="p">[</span><span class="w"> </span><span class="n">indices_data_df_source</span><span class="p">,</span><span class="w"> </span><span class="p">]</span><span class="w">

    </span><span class="c1"># Recreate the data frame</span><span class="w">
    </span><span class="n">merged_data_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merge</span><span class="p">(</span><span class="w"> </span><span class="n">subset_data_df_source</span><span class="p">,</span><span class="w">
      </span><span class="n">data.frame</span><span class="p">(</span><span class="w"> </span><span class="n">GenomicRanges</span><span class="o">::</span><span class="n">mcols</span><span class="p">(</span><span class="w"> </span><span class="n">gene_model</span><span class="w"> </span><span class="p">),</span><span class="w"> </span><span class="n">target_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">gene_model</span><span class="p">)</span><span class="w"> </span><span class="p">),</span><span class="w">
      </span><span class="n">by.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"row.names"</span><span class="p">,</span><span class="w"> </span><span class="n">by.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">source_name</span><span class="p">)</span><span class="w">

    </span><span class="c1"># merged data could still have duplicates!</span><span class="w">
    </span><span class="c1"># indices_merged_data_df_unique &lt;- !duplicated(merged_data_df[[ "Row.names" ]])</span><span class="w">
    </span><span class="c1"># merged_data_df_unique &lt;- merged_data_df[indices_merged_data_df_unique,]</span><span class="w">

    </span><span class="c1"># Set row names to target_name</span><span class="w">
    </span><span class="n">row.names</span><span class="p">(</span><span class="w"> </span><span class="n">merged_data_df</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merged_data_df</span><span class="o">$</span><span class="n">target_name</span><span class="w">
    </span><span class="c1"># Drop name columns</span><span class="w">
    </span><span class="n">merged_data_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merged_data_df</span><span class="p">[,</span><span class="w"> </span><span class="o">-</span><span class="n">which</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">merged_data_df</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Row.names"</span><span class="p">,</span><span class="w"> </span><span class="s2">"target_name"</span><span class="p">))</span><span class="w"> </span><span class="p">]</span><span class="w">

    </span><span class="c1"># Reorder the rows to match</span><span class="w">
    </span><span class="n">merged_data_df_reordered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merged_data_df</span><span class="p">[</span><span class="w"> </span><span class="n">match</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">gene_model</span><span class="p">),</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">merged_data_df</span><span class="p">)),</span><span class="w"> </span><span class="p">]</span><span class="w">

    </span><span class="nf">return</span><span class="p">(</span><span class="n">merged_data_df_reordered</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">


  </span><span class="cd">#' Retrieve the gene info</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#'  !!!!!!!!!!!!Alert alert alert ---- mapping between namespaces is not bijective.!!!!!!!!!!!!!</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' @param data_df the data frame of genes (rownames) and samples (colnames)</span><span class="w">
  </span><span class="cd">#' @param species the species (mouse, human)</span><span class="w">
  </span><span class="cd">#' @param source_name attribute (gene namespace) input</span><span class="w">
  </span><span class="cd">#' @param target_name attribute (gene namespace) desired</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' @return A \code{\link[GenomicRanges]{GRanges}} having unique and valid target_name entries</span><span class="w">
  </span><span class="cd">#'</span><span class="w">
  </span><span class="cd">#' @export</span><span class="w">
  </span><span class="n">get_gene_model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="w"> </span><span class="n">data_df</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="p">,</span><span class="w"> </span><span class="n">source_name</span><span class="p">,</span><span class="w"> </span><span class="n">target_name</span><span class="w"> </span><span class="p">){</span><span class="w">

    </span><span class="k">if</span><span class="p">(</span><span class="nf">missing</span><span class="p">(</span><span class="n">species</span><span class="p">)</span><span class="w"> </span><span class="o">||</span><span class="w">
        </span><span class="o">!</span><span class="n">grepl</span><span class="p">(</span><span class="s2">"mouse"</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="p">,</span><span class="w"> </span><span class="n">ignore.case</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w">
        </span><span class="o">!</span><span class="n">grepl</span><span class="p">(</span><span class="s2">"human"</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="p">,</span><span class="w"> </span><span class="n">ignore.case</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="n">stop</span><span class="p">(</span><span class="s2">"Species must be human or mouse"</span><span class="p">)</span><span class="w">

    </span><span class="n">dataset</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">switch</span><span class="p">(</span><span class="n">species</span><span class="p">,</span><span class="w">
      </span><span class="n">mouse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"mmusculus_gene_ensembl"</span><span class="p">,</span><span class="w">
      </span><span class="n">human</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"hsapiens_gene_ensembl"</span><span class="p">)</span><span class="w">
    </span><span class="n">mart_used</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">biomaRt</span><span class="o">::</span><span class="n">useMart</span><span class="p">(</span><span class="s2">"ENSEMBL_MART_ENSEMBL"</span><span class="p">)</span><span class="w">
    </span><span class="n">ensembl</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">biomaRt</span><span class="o">::</span><span class="n">useDataset</span><span class="p">(</span><span class="n">dataset</span><span class="p">,</span><span class="w"> </span><span class="n">mart</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mart_used</span><span class="p">)</span><span class="w">
    </span><span class="n">attributes_available</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">biomaRt</span><span class="o">::</span><span class="n">listAttributes</span><span class="p">(</span><span class="n">ensembl</span><span class="p">)</span><span class="w">

    </span><span class="k">if</span><span class="p">(</span><span class="w"> </span><span class="o">!</span><span class="n">source_name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">attributes_available</span><span class="o">$</span><span class="n">name</span><span class="w"> </span><span class="o">||</span><span class="w">
        </span><span class="o">!</span><span class="n">target_name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">attributes_available</span><span class="o">$</span><span class="n">name</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="n">stop</span><span class="p">(</span><span class="s2">"Invalid source/target name"</span><span class="p">)</span><span class="w">

    </span><span class="n">bm_info</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">biomaRt</span><span class="o">::</span><span class="n">getBM</span><span class="p">(</span><span class="n">attributes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"chromosome_name"</span><span class="p">,</span><span class="w">
      </span><span class="s2">"start_position"</span><span class="p">,</span><span class="w"> </span><span class="s2">"end_position"</span><span class="p">,</span><span class="w"> </span><span class="s2">"strand"</span><span class="p">,</span><span class="w">
      </span><span class="n">source_name</span><span class="p">,</span><span class="w"> </span><span class="n">target_name</span><span class="p">),</span><span class="w">
      </span><span class="n">filters</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">source_name</span><span class="p">,</span><span class="w">
      </span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">data_df</span><span class="p">),</span><span class="w">
      </span><span class="n">mart</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ensembl</span><span class="p">)</span><span class="w">

    </span><span class="n">rowRanges</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">GenomicRanges</span><span class="o">::</span><span class="n">GRanges</span><span class="p">(</span><span class="n">seqnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"chr"</span><span class="p">,</span><span class="w"> </span><span class="n">bm_info</span><span class="o">$</span><span class="n">chromosome_name</span><span class="p">),</span><span class="w">
      </span><span class="n">ranges</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">IRanges</span><span class="o">::</span><span class="n">IRanges</span><span class="p">(</span><span class="n">start</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">bm_info</span><span class="o">$</span><span class="n">start_position</span><span class="p">,</span><span class="w"> </span><span class="n">end</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">bm_info</span><span class="o">$</span><span class="n">end_position</span><span class="p">),</span><span class="w">
      </span><span class="n">strand</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">bm_info</span><span class="o">$</span><span class="n">strand</span><span class="p">)</span><span class="w">

    </span><span class="n">meta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">bm_info</span><span class="p">[[</span><span class="n">source_name</span><span class="p">]])</span><span class="w">
    </span><span class="n">colnames</span><span class="p">(</span><span class="n">meta</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">source_name</span><span class="p">)</span><span class="w">
    </span><span class="n">GenomicRanges</span><span class="o">::</span><span class="n">mcols</span><span class="p">(</span><span class="n">rowRanges</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">meta</span><span class="w">

    </span><span class="nf">names</span><span class="p">(</span><span class="n">rowRanges</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">bm_info</span><span class="p">[[</span><span class="n">target_name</span><span class="p">]]</span><span class="w">

    </span><span class="c1"># Filter gene_model for valid target_name</span><span class="w">
    </span><span class="c1"># GRanges objects act like vectors for subsetting</span><span class="w">
    </span><span class="n">gene_model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rowRanges</span><span class="p">[</span><span class="w"> </span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">rowRanges</span><span class="p">)</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">""</span><span class="p">)</span><span class="w"> </span><span class="p">]</span><span class="w">
    </span><span class="n">gene_model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_model</span><span class="p">[</span><span class="w"> </span><span class="o">!</span><span class="n">duplicated</span><span class="p">(</span><span class="n">GenomicRanges</span><span class="o">::</span><span class="n">mcols</span><span class="p">(</span><span class="n">gene_model</span><span class="p">)[[</span><span class="n">source_name</span><span class="p">]])</span><span class="w"> </span><span class="p">]</span><span class="w">
    </span><span class="n">gene_model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_model</span><span class="p">[</span><span class="w"> </span><span class="o">!</span><span class="n">duplicated</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">gene_model</span><span class="p">))</span><span class="w"> </span><span class="p">]</span><span class="w">

    </span><span class="nf">return</span><span class="p">(</span><span class="n">gene_model</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span></code></pre></figure>

<p>Let’s perform the ID-mapping and data merging.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">source_name</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"ensembl_gene_id"</span><span class="w">
  </span><span class="n">target_name</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"hgnc_symbol"</span><span class="w">
  </span><span class="n">species</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"human"</span><span class="w">

  </span><span class="n">brca_hd_tep_se</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merge_data</span><span class="p">(</span><span class="w">
    </span><span class="n">tep_rnaseq_metadata</span><span class="p">,</span><span class="w">
    </span><span class="n">species</span><span class="p">,</span><span class="w">
    </span><span class="n">source_name</span><span class="p">,</span><span class="w">
    </span><span class="n">target_name</span><span class="p">,</span><span class="w">
    </span><span class="n">tep_rnaseq_filelist</span><span class="p">)</span></code></pre></figure>

<p>The result of the merge is a <code class="highlighter-rouge">SummarizedExperiment</code> object named <code class="highlighter-rouge">brca_hd_tep_se</code>.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">brca_hd_tep_se</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## class: RangedSummarizedExperiment 
## dim: 34996 10 
## metadata(0):
## assays(1): counts
## rownames(34996): RNU6-871P MIR626 ... RNA5SP302 RNU6-179P
## rowData names(1): ensembl_gene_id
## colnames(10): MGH-BrCa-H-74_htsqct MGH-BrCa-H-68_htsqct ...
##   HD-2-1_htsqct HD-1_htsqct
## colData names(1): class</code></pre></figure>

<p>The <code class="highlighter-rouge">SummarizedExperiment</code> package provides a number of helpful accessor functions to examine the data.</p>

<p>For instance, we can take a peek at the column (sample) metadata using the <code class="highlighter-rouge">SummarizedExperiment::colData</code> function.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">SummarizedExperiment</span><span class="o">::</span><span class="n">colData</span><span class="p">(</span><span class="n">brca_hd_tep_se</span><span class="p">))</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left"> </th>
      <th style="text-align: left">class</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-74_htsqct</td>
      <td style="text-align: left">BrCa</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-68_htsqct</td>
      <td style="text-align: left">BrCa</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-66_htsqct</td>
      <td style="text-align: left">BrCa</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-59_htsqct</td>
      <td style="text-align: left">BrCa</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-11_htsqct</td>
      <td style="text-align: left">BrCa</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-5_htsqct</td>
      <td style="text-align: left">HD</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-4_htsqct</td>
      <td style="text-align: left">HD</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-3-1_htsqct</td>
      <td style="text-align: left">HD</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-2-1_htsqct</td>
      <td style="text-align: left">HD</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-1_htsqct</td>
      <td style="text-align: left">HD</td>
    </tr>
  </tbody>
</table>

<p>Similarly, we can peek at an excerpt of the row metadata (i.e. <a href="https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html">GenomicRanges</a>) using the <code class="highlighter-rouge">SummarizedExperiment::rowRanges</code> function.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">brca_hd_tep_rowRanged_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">SummarizedExperiment</span><span class="o">::</span><span class="n">rowRanges</span><span class="p">(</span><span class="n">brca_hd_tep_se</span><span class="p">))</span><span class="w">
  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">head</span><span class="p">(</span><span class="n">brca_hd_tep_rowRanged_df</span><span class="p">))</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left"> </th>
      <th style="text-align: left">seqnames</th>
      <th style="text-align: right">start</th>
      <th style="text-align: right">end</th>
      <th style="text-align: right">width</th>
      <th style="text-align: left">strand</th>
      <th style="text-align: left">ensembl_gene_id</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">RNU6-871P</td>
      <td style="text-align: left">chr12</td>
      <td style="text-align: right">59450673</td>
      <td style="text-align: right">59450772</td>
      <td style="text-align: right">100</td>
      <td style="text-align: left">-</td>
      <td style="text-align: left">ENSG00000251931</td>
    </tr>
    <tr>
      <td style="text-align: left">MIR626</td>
      <td style="text-align: left">chr15</td>
      <td style="text-align: right">41691585</td>
      <td style="text-align: right">41691678</td>
      <td style="text-align: right">94</td>
      <td style="text-align: left">+</td>
      <td style="text-align: left">ENSG00000207766</td>
    </tr>
    <tr>
      <td style="text-align: left">RNU6-35P</td>
      <td style="text-align: left">chr4</td>
      <td style="text-align: right">109992325</td>
      <td style="text-align: right">109992431</td>
      <td style="text-align: right">107</td>
      <td style="text-align: left">+</td>
      <td style="text-align: left">ENSG00000207260</td>
    </tr>
    <tr>
      <td style="text-align: left">MIR5694</td>
      <td style="text-align: left">chr14</td>
      <td style="text-align: right">67441855</td>
      <td style="text-align: right">67441930</td>
      <td style="text-align: right">76</td>
      <td style="text-align: left">-</td>
      <td style="text-align: left">ENSG00000265993</td>
    </tr>
    <tr>
      <td style="text-align: left">RNU6-1157P</td>
      <td style="text-align: left">chr11</td>
      <td style="text-align: right">118593988</td>
      <td style="text-align: right">118594093</td>
      <td style="text-align: right">106</td>
      <td style="text-align: left">+</td>
      <td style="text-align: left">ENSG00000207185</td>
    </tr>
    <tr>
      <td style="text-align: left">RNU4-85P</td>
      <td style="text-align: left">chr3</td>
      <td style="text-align: right">19996803</td>
      <td style="text-align: right">19996925</td>
      <td style="text-align: right">123</td>
      <td style="text-align: left">+</td>
      <td style="text-align: left">ENSG00000201545</td>
    </tr>
  </tbody>
</table>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="nf">dim</span><span class="p">(</span><span class="n">brca_hd_tep_rowRanged_df</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## [1] 34996     6</code></pre></figure>

<p>Note that we were only able to successfully map 34 996 unique HGNC symbols from the original set of Ensembl IDs.</p>

<p>Finally, we can peek at the assay data using the <code class="highlighter-rouge">SummarizedExperiment::assay</code> function.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">head</span><span class="p">(</span><span class="n">SummarizedExperiment</span><span class="o">::</span><span class="n">assays</span><span class="p">(</span><span class="n">brca_hd_tep_se</span><span class="p">)[[</span><span class="s2">"counts"</span><span class="p">]])[,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">6</span><span class="p">,</span><span class="m">7</span><span class="p">)])</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left"> </th>
      <th style="text-align: right">MGH-BrCa-H-74_htsqct</th>
      <th style="text-align: right">MGH-BrCa-H-68_htsqct</th>
      <th style="text-align: right">HD-5_htsqct</th>
      <th style="text-align: right">HD-4_htsqct</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">RNU6-871P</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">MIR626</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">RNU6-35P</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">MIR5694</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">RNU6-1157P</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">RNU4-85P</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
  </tbody>
</table>

<h2 id="d-differential-expression-testing"><a href="#differentialExpressionTesting" name="differentialExpressionTesting">D. Differential Expression Testing</a></h2>

<div class="alert alert-warning">
  <p>Section To Dos</p>
  <ul>
    <li>GSEA input file dependencies</li>
    <ul>
      <li>
        <a href="http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#RNK:_Ranked_list_file_format_.28.2A.rnk.29" target="_blank">Ranked list file (RNK)</a>: Single list of genes ordered according to the p-value derived form differential expression testing.
      </li>
    </ul>
    <li>Enrichment Map input file dependencies</li>
    <ul>
      <li>
        <a href="http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#TXT:_Text_file_format_for_expression_dataset_.28.2A.txt.29" target="_blank">Text file for expression dataset (TXT)</a>: A file containing normalized gene expression values.
      </li>
      <li>
        <a href="http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#CLS:_Categorical_.28e.g_tumor_vs_normal.29_class_file_format_.28.2A.cls.29" target="_blank">Categorical class file (CLS)</a>: A description of each sample's class.
      </li>
    </ul>
  </ul>
</div>

<p>It will be convenient to generate the ranked list, expression and class files alongside the RNA-Seq data transformations. We will be using the <code class="highlighter-rouge">edgeR</code> Bioconductor package.</p>

<h3 id="filtering">Filtering</h3>

<p>RNA species with very low mapped read counts in a small number of samples can be highly variable. Consequently, we choose to ignore these in the search for differential expression. Best <em>et al.</em> use the rule of thumb that ‘genes with less than five (non-normalized) read counts in all samples were excluded from analyses’.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="c1">### Minimum number of mapped read counts per sample</span><span class="w">
  </span><span class="n">min_counts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">5</span><span class="w">

  </span><span class="c1">### Declare baseline (i.e. control) and test classes</span><span class="w">
  </span><span class="n">baseline_class</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"HD"</span><span class="w">
  </span><span class="n">test_class</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"BrCa"</span><span class="w">
  </span><span class="n">comparison</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">baseline_class</span><span class="p">,</span><span class="w"> </span><span class="n">test_class</span><span class="p">)</span><span class="w">

  </span><span class="n">brca_hd_tep_se_counts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">SummarizedExperiment</span><span class="o">::</span><span class="n">assays</span><span class="p">(</span><span class="n">brca_hd_tep_se</span><span class="p">)[[</span><span class="s2">"counts"</span><span class="p">]]</span><span class="w">
  </span><span class="n">brca_hd_tep_se_group</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">SummarizedExperiment</span><span class="o">::</span><span class="n">colData</span><span class="p">(</span><span class="n">brca_hd_tep_se</span><span class="p">)[[</span><span class="s2">"class"</span><span class="p">]]</span><span class="w">

  </span><span class="c1">### Find genes (rows) with a minimum number of counts</span><span class="w">
  </span><span class="n">index_test_class</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">brca_hd_tep_se_group</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">comparison</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w">
  </span><span class="n">index_baseline_class</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">brca_hd_tep_se_group</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">comparison</span><span class="p">[</span><span class="m">2</span><span class="p">]</span><span class="w">
  </span><span class="n">row_with_mincount</span><span class="w"> </span><span class="o">&lt;-</span><span class="w">
    </span><span class="n">rowSums</span><span class="p">(</span><span class="n">edgeR</span><span class="o">::</span><span class="n">cpm</span><span class="p">(</span><span class="n">brca_hd_tep_se_counts</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">min_counts</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;=</span><span class="w">
      </span><span class="nf">min</span><span class="p">(</span><span class="nf">sum</span><span class="p">(</span><span class="n">index_baseline_class</span><span class="p">),</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">index_test_class</span><span class="p">))</span><span class="w">

  </span><span class="c1">### Subset the original data accordingly</span><span class="w">
  </span><span class="n">brca_hd_tep_dge_counts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">brca_hd_tep_se_counts</span><span class="p">[</span><span class="n">row_with_mincount</span><span class="p">,]</span><span class="w">

  </span><span class="c1">### Push the data into the edgeR::DGEList</span><span class="w">
  </span><span class="n">brca_hd_tep_filtered_dge</span><span class="w"> </span><span class="o">&lt;-</span><span class="w">
    </span><span class="n">edgeR</span><span class="o">::</span><span class="n">DGEList</span><span class="p">(</span><span class="n">counts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">brca_hd_tep_dge_counts</span><span class="p">,</span><span class="w"> </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">brca_hd_tep_se_group</span><span class="p">)</span></code></pre></figure>

<p>Note that our end product is a very convenient class, the <code class="highlighter-rouge">edgeR::DGEList</code>.</p>

<blockquote>
  <p>The main components of an DGEList object are a matrix counts containing the integer counts, a data.frame samples containing information about the samples or libraries, and a optional data.frame genes containing annotation for the genes or genomic features. The data.frame samples contains a column lib.size for the library size or sequencing depth for each sample. If not specified by the user, the library sizes will be computed from the column sums of the counts. For classic edgeR the data.frame samples must also contain a column group, identifying the group membership of each sample.</p>
</blockquote>

<p>With this in mind, let’s take a look at the ‘counts’ component.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">head</span><span class="p">(</span><span class="n">brca_hd_tep_filtered_dge</span><span class="p">[[</span><span class="s2">"counts"</span><span class="p">]])[,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">5</span><span class="p">,</span><span class="m">6</span><span class="p">)])</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left"> </th>
      <th style="text-align: right">MGH-BrCa-H-74_htsqct</th>
      <th style="text-align: right">MGH-BrCa-H-68_htsqct</th>
      <th style="text-align: right">MGH-BrCa-H-11_htsqct</th>
      <th style="text-align: right">HD-5_htsqct</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">MAP4K5</td>
      <td style="text-align: right">255</td>
      <td style="text-align: right">520</td>
      <td style="text-align: right">854</td>
      <td style="text-align: right">238</td>
    </tr>
    <tr>
      <td style="text-align: left">SAV1</td>
      <td style="text-align: right">167</td>
      <td style="text-align: right">189</td>
      <td style="text-align: right">323</td>
      <td style="text-align: right">34</td>
    </tr>
    <tr>
      <td style="text-align: left">IRF3</td>
      <td style="text-align: right">5</td>
      <td style="text-align: right">5</td>
      <td style="text-align: right">23</td>
      <td style="text-align: right">22</td>
    </tr>
    <tr>
      <td style="text-align: left">ZC3H11A</td>
      <td style="text-align: right">80</td>
      <td style="text-align: right">145</td>
      <td style="text-align: right">181</td>
      <td style="text-align: right">16</td>
    </tr>
    <tr>
      <td style="text-align: left">NUP88</td>
      <td style="text-align: right">24</td>
      <td style="text-align: right">13</td>
      <td style="text-align: right">49</td>
      <td style="text-align: right">45</td>
    </tr>
    <tr>
      <td style="text-align: left">STXBP5-AS1</td>
      <td style="text-align: right">67</td>
      <td style="text-align: right">159</td>
      <td style="text-align: right">90</td>
      <td style="text-align: right">46</td>
    </tr>
  </tbody>
</table>

<p>Similarly, we can peek inside the ‘samples’ component.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">brca_hd_tep_filtered_dge</span><span class="p">[[</span><span class="s2">"samples"</span><span class="p">]])</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left"> </th>
      <th style="text-align: left">group</th>
      <th style="text-align: right">lib.size</th>
      <th style="text-align: right">norm.factors</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-74_htsqct</td>
      <td style="text-align: left">BrCa</td>
      <td style="text-align: right">1010942</td>
      <td style="text-align: right">1</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-68_htsqct</td>
      <td style="text-align: left">BrCa</td>
      <td style="text-align: right">1458995</td>
      <td style="text-align: right">1</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-66_htsqct</td>
      <td style="text-align: left">BrCa</td>
      <td style="text-align: right">1104454</td>
      <td style="text-align: right">1</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-59_htsqct</td>
      <td style="text-align: left">BrCa</td>
      <td style="text-align: right">375353</td>
      <td style="text-align: right">1</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-11_htsqct</td>
      <td style="text-align: left">BrCa</td>
      <td style="text-align: right">1567897</td>
      <td style="text-align: right">1</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-5_htsqct</td>
      <td style="text-align: left">HD</td>
      <td style="text-align: right">978816</td>
      <td style="text-align: right">1</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-4_htsqct</td>
      <td style="text-align: left">HD</td>
      <td style="text-align: right">942170</td>
      <td style="text-align: right">1</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-3-1_htsqct</td>
      <td style="text-align: left">HD</td>
      <td style="text-align: right">1006665</td>
      <td style="text-align: right">1</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-2-1_htsqct</td>
      <td style="text-align: left">HD</td>
      <td style="text-align: right">771351</td>
      <td style="text-align: right">1</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-1_htsqct</td>
      <td style="text-align: left">HD</td>
      <td style="text-align: right">1295886</td>
      <td style="text-align: right">1</td>
    </tr>
  </tbody>
</table>

<h3 id="normalization">Normalization</h3>

<p>RNA for a sample can be sequenced to varying ‘depths’ in that the total number of sequence reads mapped to a gene for an individual sequencing run is not necessarily constant. What most concerns us is not the absolute counts of an individual RNA species coming out of a sequencing run but rather the proportion. In practical terms, we desire a fair-comparison of RNA counts between samples that takes into account variation in depth. Our recommendation is to use a normalization technique called Trimmed mean of M-values (TMM)<sup id="fnref:5"><a href="#fn:5" class="footnote">5</a></sup> that effectively standardizes counts between distinct sequencing runs by assuming that most genes are not expected to alter their expression.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">brca_hd_tep_tmm_normalized_dge</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">edgeR</span><span class="o">::</span><span class="n">calcNormFactors</span><span class="p">(</span><span class="n">brca_hd_tep_filtered_dge</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"TMM"</span><span class="p">)</span><span class="w">
  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">brca_hd_tep_tmm_normalized_dge</span><span class="p">[[</span><span class="s2">"samples"</span><span class="p">]])</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left"> </th>
      <th style="text-align: left">group</th>
      <th style="text-align: right">lib.size</th>
      <th style="text-align: right">norm.factors</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-74_htsqct</td>
      <td style="text-align: left">BrCa</td>
      <td style="text-align: right">1010942</td>
      <td style="text-align: right">0.8467126</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-68_htsqct</td>
      <td style="text-align: left">BrCa</td>
      <td style="text-align: right">1458995</td>
      <td style="text-align: right">0.9171140</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-66_htsqct</td>
      <td style="text-align: left">BrCa</td>
      <td style="text-align: right">1104454</td>
      <td style="text-align: right">1.0598800</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-59_htsqct</td>
      <td style="text-align: left">BrCa</td>
      <td style="text-align: right">375353</td>
      <td style="text-align: right">0.9411707</td>
    </tr>
    <tr>
      <td style="text-align: left">MGH-BrCa-H-11_htsqct</td>
      <td style="text-align: left">BrCa</td>
      <td style="text-align: right">1567897</td>
      <td style="text-align: right">1.1394811</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-5_htsqct</td>
      <td style="text-align: left">HD</td>
      <td style="text-align: right">978816</td>
      <td style="text-align: right">0.8057214</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-4_htsqct</td>
      <td style="text-align: left">HD</td>
      <td style="text-align: right">942170</td>
      <td style="text-align: right">1.0404548</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-3-1_htsqct</td>
      <td style="text-align: left">HD</td>
      <td style="text-align: right">1006665</td>
      <td style="text-align: right">0.9094940</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-2-1_htsqct</td>
      <td style="text-align: left">HD</td>
      <td style="text-align: right">771351</td>
      <td style="text-align: right">1.1216675</td>
    </tr>
    <tr>
      <td style="text-align: left">HD-1_htsqct</td>
      <td style="text-align: left">HD</td>
      <td style="text-align: right">1295886</td>
      <td style="text-align: right">1.3247572</td>
    </tr>
  </tbody>
</table>

<p>Note that the TMM-normalized <code class="highlighter-rouge">DGEList</code> called <code class="highlighter-rouge">brca_hd_tep_tmm_normalized_dge</code> has updated column <code class="highlighter-rouge">norm.factors</code> in component ‘samples’. This value will be used to determine an ‘effective library size’, leaving the raw counts untouched.</p>

<h4 id="text-file-for-expression-dataset-txt">Text file for expression dataset (TXT)</h4>

<p>At this point we can generate an expression file of normalized RNA counts where row names are gene symbols and column names are sample IDs. We will use this later to be able to display an expression heatmap for any enriched pathway in our Enrichment Map.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="c1">### Combine two data frames - gene metadata and sample counts</span><span class="w">
  </span><span class="n">brca_hd_tep_tmm_normalized_mat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">edgeR</span><span class="o">::</span><span class="n">cpm</span><span class="p">(</span><span class="n">brca_hd_tep_tmm_normalized_dge</span><span class="p">,</span><span class="w"> </span><span class="n">normalized.lib.size</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">

  </span><span class="n">meta_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="w">
    </span><span class="n">NAME</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">brca_hd_tep_tmm_normalized_mat</span><span class="p">),</span><span class="w">
    </span><span class="n">DESCRIPTION</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">brca_hd_tep_tmm_normalized_mat</span><span class="p">),</span><span class="w">
    </span><span class="n">check.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">

  </span><span class="n">rownames</span><span class="p">(</span><span class="n">brca_hd_tep_tmm_normalized_mat</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span><span class="w">
  </span><span class="n">brca_hd_tep_tmm_normalized_expression_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">meta_df</span><span class="p">,</span><span class="w"> </span><span class="n">brca_hd_tep_tmm_normalized_mat</span><span class="p">,</span><span class="w">  </span><span class="n">check.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">

  </span><span class="c1">### Write out</span><span class="w">
  </span><span class="n">expression_dataset_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">output_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"brca_hd_tep_tmm_normalized_expression.txt"</span><span class="p">)</span><span class="w">
  </span><span class="n">write.table</span><span class="p">(</span><span class="n">brca_hd_tep_tmm_normalized_expression_df</span><span class="p">,</span><span class="w">
              </span><span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
              </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">,</span><span class="w">
              </span><span class="n">file</span><span class="o">=</span><span class="n">expression_dataset_path</span><span class="p">,</span><span class="w">
              </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span></code></pre></figure>

<p>We should have a file <code class="highlighter-rouge">brca_hd_tep_tmm_normalized_expression.txt</code> containing the following tab-delimited content:</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">brca_hd_tep_tmm_normalized_expression_file_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="n">expression_dataset_path</span><span class="p">,</span><span class="w">
                                                              </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
                                                             </span><span class="n">check.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">head</span><span class="p">(</span><span class="n">brca_hd_tep_tmm_normalized_expression_file_df</span><span class="p">)[,</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,</span><span class="m">8</span><span class="p">,</span><span class="m">9</span><span class="p">)])</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left">NAME</th>
      <th style="text-align: left">DESCRIPTION</th>
      <th style="text-align: right">MGH-BrCa-H-74_htsqct</th>
      <th style="text-align: right">MGH-BrCa-H-68_htsqct</th>
      <th style="text-align: right">HD-5_htsqct</th>
      <th style="text-align: right">HD-4_htsqct</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">MAP4K5</td>
      <td style="text-align: left">MAP4K5</td>
      <td style="text-align: right">297.905074</td>
      <td style="text-align: right">388.620956</td>
      <td style="text-align: right">301.78036</td>
      <td style="text-align: right">181.57979</td>
    </tr>
    <tr>
      <td style="text-align: left">SAV1</td>
      <td style="text-align: left">SAV1</td>
      <td style="text-align: right">195.098617</td>
      <td style="text-align: right">141.248771</td>
      <td style="text-align: right">43.11148</td>
      <td style="text-align: right">20.40222</td>
    </tr>
    <tr>
      <td style="text-align: left">IRF3</td>
      <td style="text-align: left">IRF3</td>
      <td style="text-align: right">5.841276</td>
      <td style="text-align: right">3.736740</td>
      <td style="text-align: right">27.89566</td>
      <td style="text-align: right">32.64356</td>
    </tr>
    <tr>
      <td style="text-align: left">ZC3H11A</td>
      <td style="text-align: left">ZC3H11A</td>
      <td style="text-align: right">93.460415</td>
      <td style="text-align: right">108.365459</td>
      <td style="text-align: right">20.28776</td>
      <td style="text-align: right">73.44800</td>
    </tr>
    <tr>
      <td style="text-align: left">NUP88</td>
      <td style="text-align: left">NUP88</td>
      <td style="text-align: right">28.038125</td>
      <td style="text-align: right">9.715524</td>
      <td style="text-align: right">57.05931</td>
      <td style="text-align: right">71.40778</td>
    </tr>
    <tr>
      <td style="text-align: left">STXBP5-AS1</td>
      <td style="text-align: left">STXBP5-AS1</td>
      <td style="text-align: right">78.273098</td>
      <td style="text-align: right">118.828331</td>
      <td style="text-align: right">58.32730</td>
      <td style="text-align: right">67.32734</td>
    </tr>
  </tbody>
</table>

<h3 id="differential-expression-testing">Differential expression testing</h3>

<p>In this step we perform a pair-wise comparison of RNA species counts in BrCa samples relative HD samples. The framework used to determine differential RNA expression is a ‘hypothesis-testing’ technique that entails:</p>

<ul>
  <li>Declare the null hypothesis of no difference in RNA counts for each gene</li>
  <li>Define a null distribution that describes how RNA counts vary under circumstances where there is no association between RNA counts and class</li>
  <li>Calculate the probability (p-value) of observing a difference in RNA species counts at least as extreme as the one observed assuming the null hypothesis/distribution</li>
  <li>Perform multiple-testing correction</li>
</ul>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="c1">### Calculate variability (dispersions) in data</span><span class="w">
  </span><span class="n">brca_hd_tep_fitted_commondisp_dge</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">edgeR</span><span class="o">::</span><span class="n">estimateCommonDisp</span><span class="p">(</span><span class="n">brca_hd_tep_tmm_normalized_dge</span><span class="p">)</span><span class="w">
  </span><span class="n">brca_hd_tep_fitted_tagwise_dge</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">edgeR</span><span class="o">::</span><span class="n">estimateTagwiseDisp</span><span class="p">(</span><span class="n">brca_hd_tep_fitted_commondisp_dge</span><span class="p">)</span><span class="w">

  </span><span class="c1">### Perform differential expression testing (comparison is 'BrCa' vs 'HD')</span><span class="w">
  </span><span class="n">brca_hd_tep_de_tested_dge</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">edgeR</span><span class="o">::</span><span class="n">exactTest</span><span class="p">(</span><span class="n">brca_hd_tep_fitted_tagwise_dge</span><span class="p">,</span><span class="w"> </span><span class="n">pair</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">comparison</span><span class="p">)</span><span class="w">

  </span><span class="c1">### Perform multiple-testing correction using Benjamini-Hockberg procedure</span><span class="w">
  </span><span class="n">brca_hd_tep_de_tested_tt</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">edgeR</span><span class="o">::</span><span class="n">topTags</span><span class="p">(</span><span class="n">brca_hd_tep_de_tested_dge</span><span class="p">,</span><span class="w">
    </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">nrow</span><span class="p">(</span><span class="n">brca_hd_tep_tmm_normalized_dge</span><span class="p">),</span><span class="w">
    </span><span class="n">adjust.method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"BH"</span><span class="p">,</span><span class="w">
    </span><span class="n">sort.by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"PValue"</span><span class="p">)</span></code></pre></figure>

<p>The result of these transformations is a <code class="highlighter-rouge">topTags</code> object named <code class="highlighter-rouge">brca_hd_tep_de_tested_tt</code>. We can peek inside our result to view the list of genes ranked by p-value from differential expression testing.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">head</span><span class="p">(</span><span class="n">brca_hd_tep_de_tested_tt</span><span class="p">[[</span><span class="s2">"table"</span><span class="p">]]))</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left"> </th>
      <th style="text-align: right">logFC</th>
      <th style="text-align: right">logCPM</th>
      <th style="text-align: right">PValue</th>
      <th style="text-align: right">FDR</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">TRIM58</td>
      <td style="text-align: right">6.577953</td>
      <td style="text-align: right">7.818481</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">ARHGAP45</td>
      <td style="text-align: right">6.612944</td>
      <td style="text-align: right">9.047699</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">NCK2</td>
      <td style="text-align: right">5.679970</td>
      <td style="text-align: right">7.160183</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">GAS2L1</td>
      <td style="text-align: right">5.946095</td>
      <td style="text-align: right">7.488178</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">SPTB</td>
      <td style="text-align: right">5.334870</td>
      <td style="text-align: right">7.041291</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
    <tr>
      <td style="text-align: left">ANKRD9</td>
      <td style="text-align: right">6.913215</td>
      <td style="text-align: right">6.214325</td>
      <td style="text-align: right">0</td>
      <td style="text-align: right">0</td>
    </tr>
  </tbody>
</table>

<h4 id="rank-list-file-rnk">Rank list file (RNK)</h4>

<p>At this stage, we can generate a rank list file where row names are gene symbols and a single column indicates the rank calculated as a function of their p-value. The larger the magnitude of the positive or negative rank, the rarer such an observation would be under the assumption of no association between class and RNA count.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="c1">### Rank by inverse of p-value taking into account 'sign' of change in BrCa (i.e. increase/decrease) relative to HD</span><span class="w">
  </span><span class="n">brca_hd_tep_rank_values</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sign</span><span class="p">(</span><span class="n">brca_hd_tep_de_tested_tt</span><span class="p">[[</span><span class="s2">"table"</span><span class="p">]][[</span><span class="s2">"logFC"</span><span class="p">]])</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="p">(</span><span class="m">-1</span><span class="p">)</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">log10</span><span class="p">(</span><span class="n">brca_hd_tep_de_tested_tt</span><span class="p">[[</span><span class="s2">"table"</span><span class="p">]][[</span><span class="s2">"PValue"</span><span class="p">]])</span><span class="w">

  </span><span class="c1">### Take into account log10(0) = -Inf</span><span class="w">
  </span><span class="n">brca_hd_tep_rank_values_max</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">brca_hd_tep_rank_values</span><span class="p">[</span><span class="w"> </span><span class="n">brca_hd_tep_rank_values</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="kc">Inf</span><span class="w"> </span><span class="p">])</span><span class="w">
  </span><span class="n">brca_hd_tep_rank_values_unique</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sapply</span><span class="p">(</span><span class="w"> </span><span class="n">brca_hd_tep_rank_values</span><span class="p">,</span><span class="w">
                                            </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="n">replace</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="nf">is.infinite</span><span class="p">(</span><span class="n">x</span><span class="p">),</span><span class="w">
                                            </span><span class="nf">sign</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="p">(</span><span class="n">brca_hd_tep_rank_values_max</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">runif</span><span class="p">(</span><span class="m">1</span><span class="p">)))</span><span class="w"> </span><span class="p">)</span><span class="w">


  </span><span class="c1">### Construct the data frame we wish place into a tabular file</span><span class="w">
  </span><span class="n">genenames</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">brca_hd_tep_de_tested_tt</span><span class="p">[[</span><span class="s2">"table"</span><span class="p">]]))</span><span class="w">
  </span><span class="n">brca_hd_tep_ranks_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">gene</span><span class="o">=</span><span class="n">genenames</span><span class="p">,</span><span class="w">
                                     </span><span class="n">rank</span><span class="o">=</span><span class="n">brca_hd_tep_rank_values_unique</span><span class="p">,</span><span class="w">
                                     </span><span class="n">stringsAsFactors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
  </span><span class="n">brca_hd_tep_ordered_ranks_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">brca_hd_tep_ranks_df</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">brca_hd_tep_ranks_df</span><span class="p">[,</span><span class="m">2</span><span class="p">],</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="w">

  </span><span class="c1">## Write out to file</span><span class="w">
  </span><span class="n">rank_list_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">output_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"brca_hd_tep.rnk"</span><span class="p">)</span><span class="w">
  </span><span class="n">write.table</span><span class="p">(</span><span class="n">brca_hd_tep_ordered_ranks_df</span><span class="p">,</span><span class="w">
              </span><span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
              </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">,</span><span class="w">
              </span><span class="n">file</span><span class="o">=</span><span class="n">rank_list_path</span><span class="p">,</span><span class="w">
              </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span></code></pre></figure>

<p>Let’s peek at the top and bottom of our rank list saved in <code class="highlighter-rouge">brca_hd_tep.rnk</code>.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">brca_hd_tep_ordered_ranks_file_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="n">rank_list_path</span><span class="p">,</span><span class="w">
                                                  </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
                                                  </span><span class="n">check.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">

  </span><span class="n">ranks_head</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">brca_hd_tep_ordered_ranks_file_df</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="o">=</span><span class="m">5</span><span class="p">)</span><span class="w">
  </span><span class="n">rownames</span><span class="p">(</span><span class="n">ranks_head</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span><span class="w">
  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">ranks_head</span><span class="p">)</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left">gene</th>
      <th style="text-align: right">rank</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">TRIM58</td>
      <td style="text-align: right">34.46062</td>
    </tr>
    <tr>
      <td style="text-align: left">ARHGAP45</td>
      <td style="text-align: right">32.99957</td>
    </tr>
    <tr>
      <td style="text-align: left">NCK2</td>
      <td style="text-align: right">27.51070</td>
    </tr>
    <tr>
      <td style="text-align: left">GAS2L1</td>
      <td style="text-align: right">23.96204</td>
    </tr>
    <tr>
      <td style="text-align: left">SPTB</td>
      <td style="text-align: right">23.90996</td>
    </tr>
  </tbody>
</table>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">ranks_tail</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">tail</span><span class="p">(</span><span class="n">brca_hd_tep_ordered_ranks_file_df</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="o">=</span><span class="m">5</span><span class="p">)</span><span class="w">
  </span><span class="n">rownames</span><span class="p">(</span><span class="n">ranks_tail</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span><span class="w">
  </span><span class="n">knitr</span><span class="o">::</span><span class="n">kable</span><span class="p">(</span><span class="n">ranks_tail</span><span class="p">)</span></code></pre></figure>

<table>
  <thead>
    <tr>
      <th style="text-align: left">gene</th>
      <th style="text-align: right">rank</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">MS4A1</td>
      <td style="text-align: right">-11.50463</td>
    </tr>
    <tr>
      <td style="text-align: left">CD79A</td>
      <td style="text-align: right">-11.53814</td>
    </tr>
    <tr>
      <td style="text-align: left">MDM4</td>
      <td style="text-align: right">-12.93184</td>
    </tr>
    <tr>
      <td style="text-align: left">LYZ</td>
      <td style="text-align: right">-15.42705</td>
    </tr>
    <tr>
      <td style="text-align: left">CD74</td>
      <td style="text-align: right">-16.58382</td>
    </tr>
  </tbody>
</table>

<h4 id="categorical-class-file-cls">Categorical class file (CLS)</h4>

<p>This file will be used in the Enrichment Map so that we can differentiate (i.e. colour) those pathways ‘up-regulated’ in HD versus BrCa samples. The CLS format contains information about the sample classes (aka ‘condition’, ‘phenotype’) and assigns each sample to one class.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">n_samples</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dim</span><span class="p">(</span><span class="n">brca_hd_tep_filtered_dge</span><span class="p">)[</span><span class="m">2</span><span class="p">]</span><span class="w">
  </span><span class="n">n_classes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">2</span><span class="w">

  </span><span class="n">l</span><span class="m">1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="n">n_samples</span><span class="p">,</span><span class="w"> </span><span class="n">n_classes</span><span class="p">,</span><span class="w"> </span><span class="s2">"1"</span><span class="p">)</span><span class="w">
  </span><span class="n">l</span><span class="m">2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="s2">"#"</span><span class="p">,</span><span class="w"> </span><span class="n">brca_hd_tep_de_tested_tt</span><span class="p">[[</span><span class="s2">"comparison"</span><span class="p">]][</span><span class="m">1</span><span class="p">],</span><span class="w"> </span><span class="n">brca_hd_tep_de_tested_tt</span><span class="p">[[</span><span class="s2">"comparison"</span><span class="p">]][</span><span class="m">2</span><span class="p">])</span><span class="w">
  </span><span class="n">l</span><span class="m">3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="n">brca_hd_tep_filtered_dge</span><span class="p">[[</span><span class="s2">"samples"</span><span class="p">]][[</span><span class="s2">"group"</span><span class="p">]],</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">" "</span><span class="p">)</span><span class="w">
  </span><span class="n">brca_hd_tep_cls</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rbind</span><span class="p">(</span><span class="n">l</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">l</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">l</span><span class="m">3</span><span class="p">)</span><span class="w">

  </span><span class="c1">### Write out to file</span><span class="w">
  </span><span class="n">categorical_class_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">output_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"brca_hd_tep.cls"</span><span class="p">)</span><span class="w">
  </span><span class="n">write</span><span class="p">(</span><span class="n">brca_hd_tep_cls</span><span class="p">,</span><span class="w">
        </span><span class="n">file</span><span class="o">=</span><span class="n">categorical_class_path</span><span class="p">,</span><span class="w">
        </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">)</span></code></pre></figure>

<p>The matrix <code class="highlighter-rouge">brca_hd_tep_cls</code> has the following format, assuming N samples:</p>

<table>
  <thead>
    <tr>
      <th> </th>
      <th> </th>
      <th> </th>
      <th> </th>
      <th> </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Total samples</td>
      <td>Total classes</td>
      <td>1</td>
      <td> </td>
      <td> </td>
    </tr>
    <tr>
      <td>#</td>
      <td>Class Name</td>
      <td>Class Name</td>
      <td> </td>
      <td> </td>
    </tr>
    <tr>
      <td>Sample 1 class</td>
      <td>Sample 2 class</td>
      <td>Sample 3 class</td>
      <td>…</td>
      <td>Sample N class</td>
    </tr>
  </tbody>
</table>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">rownames</span><span class="p">(</span><span class="n">brca_hd_tep_cls</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span><span class="w">
  </span><span class="n">brca_hd_tep_cls</span></code></pre></figure>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code> [,1]                                      [1,] "10 2 1"                                  [2,] "# HD BrCa"                               [3,] "BrCa BrCa BrCa BrCa BrCa HD HD HD HD HD"
</code></pre></div></div>

<h2 id="e-gene-set-enrichment-analysis"><a href="#geneSetEnrichmentAnalysis" name="geneSetEnrichmentAnalysis">E. Gene Set Enrichment Analysis</a></h2>

<div class="alert alert-warning">
  <p>Section To Dos</p>
  <ul>
    <li>Obtain gene set database of candidates</li>
    <li>Run GSEA and obtain enriched gene sets</li>
  </ul>
</div>

<h3 id="gene-set-database">Gene set database</h3>

<p>A <a href="http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#Gene_Set_Database_Formats">gene set database</a> is a text file that describes names of gene sets and one of more gene IDs that are in that gene set. We will be using the <a href="http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29">Gene Matric Transposed (GMT)</a> format.</p>

<p>For your convenience, gene set database (GMT) files for <a href="http://download.baderlab.org/EM_Genesets/">human, mouse and rat are curated by the Bader lab</a> on a regular basis from various sources. In this instance we provide <a href="http://download.baderlab.org/EM_Genesets/August_01_2017/Human/symbol/Human_GOBP_AllPathways_no_GO_iea_August_01_2017_symbol.gmt"><code class="highlighter-rouge">Human_GOBP_AllPathways_no_GO_iea_August_01_2017_symbol.gmt</code></a> which is a database of human gene sets that does not include terms inferred from electronic annotations (iea).</p>

<h3 id="run-gsea">Run GSEA</h3>

<p>GSEA comes as stand-alone Java program.  Recall that we asked you to register, login and download the Java jar file to a convenient location (e.g. <code class="highlighter-rouge">/Users/username/bin/gsea-3.0.jar</code>). We will run GSEA via R with a set of options. Any of the supplied options can be <a href="http://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm#_Running_GSEA_from">customized</a>.</p>

<p>In our GSEA run, the following relevant options have been specified:</p>

<ul>
  <li><em>rpt_label</em> - name of output folder for this run</li>
  <li><em>out</em> - directory for results</li>
  <li><em>gmx</em> - path to the gene set definition (gmt) file</li>
  <li><em>rnk</em> - path to rank list file (RNK)</li>
  <li><em>nperm</em> - number of permutations to generate null distribution</li>
  <li><em>set_min (set_max)</em> - limits on number of genes for candidate gene sets</li>
  <li><em>scoring_scheme</em> - how to calculate contributions of genes to a gene set’s score</li>
  <li><em>permute</em> - for GSEA preranked you can only permute via gene_sets</li>
  <li><em>num</em> - number of results to plot output file for</li>
  <li><em>rnd_seed</em> - random seed to use</li>
  <li><em>zip_report</em> - zip to output directory</li>
  <li><em>gui</em> - when running from the commandline this needs to be false</li>
</ul>

<div class="alert alert-danger">
  GSEA can take many minutes to complete. Mileage may vary depending on the settings. Our analysis took 5 minutes on a MacBook Air (Early 2015), 2.2 GHz Intel Core i7 (8GB 16600 MHz DDR3).
</div>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">doEnrichment</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">FALSE</span><span class="w">
  </span><span class="c1">### Declare user-defined settings</span><span class="w">
  </span><span class="n">gsea_jar_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="s2">"/Users/jeffreywong/bin/gsea-3.0.jar"</span><span class="p">)</span><span class="w">
  </span><span class="n">gsea_rpt_label</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"tep_BrCa_HD_analysis"</span><span class="w">
  </span><span class="n">gsea_analysis_name</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"tep_BrCa_HD"</span><span class="w">
  </span><span class="n">gsea_out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">getwd</span><span class="p">(),</span><span class="w"> </span><span class="s2">"gsea_output"</span><span class="p">)</span><span class="w">
  </span><span class="n">gsea_gmx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">getwd</span><span class="p">(),</span><span class="w">
                        </span><span class="s2">"data"</span><span class="p">,</span><span class="w">
                        </span><span class="s2">"PathwayCommons9.All.hgnc.names.gmt"</span><span class="p">)</span><span class="w">
  </span><span class="n">gsea_rank_list_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rank_list_path</span><span class="w">
  </span><span class="n">gsea_num_permutations</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">1000</span><span class="w">
  </span><span class="n">gsea_min_gs_size</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">15</span><span class="w">
  </span><span class="n">gsea_max_gs_size</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">200</span><span class="w">

  </span><span class="c1">## Execute GSEA</span><span class="w">
  </span><span class="n">command</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="s2">"java -cp"</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_jar_path</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-Xmx1G xtools.gsea.GseaPreranked"</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-rpt_label"</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_analysis_name</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-out"</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_out</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-gmx"</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_gmx</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-rnk"</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_rank_list_path</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-nperm"</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_num_permutations</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-set_min"</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_min_gs_size</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-set_max"</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_max_gs_size</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-collapse false"</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-scoring_scheme weighted"</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-permute gene_set"</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-num 100"</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-plot_top_x 20"</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-rnd_seed 12345"</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-zip_report false"</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"-gui false"</span><span class="p">,</span><span class="w">
                   </span><span class="s2">"&gt;"</span><span class="p">,</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="s2">"gsea_output_"</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_rpt_label</span><span class="p">,</span><span class="w"> </span><span class="s2">".txt"</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s2">""</span><span class="p">),</span><span class="w">
            </span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span><span class="w">

  </span><span class="k">if</span><span class="p">(</span><span class="w"> </span><span class="n">doEnrichment</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">TRUE</span><span class="w"> </span><span class="p">){</span><span class="w">
    </span><span class="n">system</span><span class="p">(</span><span class="n">command</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span></code></pre></figure>

<p>In this case, we will have a directory <code class="highlighter-rouge">tep_BrCa_HD.Preranked.XXXXXXXXXXXXX</code> inside of the directory declared by in <code class="highlighter-rouge">gsea_out</code> that contains reports and figures generated during GSEA.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">gsea_tep_BrCa_HD_analysis_directory</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">list.files</span><span class="p">(</span><span class="n">path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gsea_out</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\\.GseaPreranked"</span><span class="p">)</span></code></pre></figure>

<p>Look for the following files:</p>

<ul>
  <li><code class="highlighter-rouge">index.html</code> - This is a <a href="http://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm#_GSEA_Report">GSEA report</a> that summarizes the analysis results</li>
  <li><code class="highlighter-rouge">gsea_report_for_na_pos_XXXXXXXXXXXXX.xls</code> - The collection of gene sets enriched in the BrCa class. The ‘pos’ label originates from the way in which we compared the RNA-Seq classes, that is, BrCa relative to HD</li>
  <li><code class="highlighter-rouge">gsea_report_for_na_neg_XXXXXXXXXXXXX.xls</code> - The collection of gene sets enriched in the HD class</li>
</ul>

<p>Please refer to the GSEA documentation on <a href="http://software.broadinstitute.org/gsea/doc/GSEAUserGuideTEXT.htm#_Interpreting_GSEA_Results">‘Interpreting GSEA Results’</a> for full details.</p>

<h2 id="f-enrichment-map"><a href="#enrichmentMap" name="enrichmentMap">F. Enrichment Map</a></h2>

<div class="alert alert-warning">
  <p>Section To Dos</p>
  <ul>
    <li>Obtain an Enrichment Map of the enriched pathways</li>
  </ul>
</div>

<p>Enrichment Map<sup id="fnref:6"><a href="#fn:6" class="footnote">6</a></sup> is a visualization analysis tool that organizes gene sets into an information-rich similarity network. The true power of Enrichment Map is that it is a visual display method that reduces complexity by grouping similar gene sets as defined by the number of overlapping genes.</p>

<p>Gene sets are represented as nodes whose radius is proportional to the number of genes. Edges indicate nodes with shared genes, where the thickness of the line is proportional to the degree of overlap. Finally, EM can use node color to represent other dimensions of the data, for example, gene sets enriched in different classes.</p>

<h3 id="cytoscape-1">Cytoscape</h3>

<p>Recall that we asked you to download and install Cytoscape (3.5.1) along with various apps. Open Cytoscape on your computer as this notebook will run commands to it through a custom package  <a href="https://github.com/cytoscape/cytoscape-automation"><code class="highlighter-rouge">cytoscape-automation/r2cytoscape</code></a> (0.0.3).</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">devtools</span><span class="o">::</span><span class="n">install_github</span><span class="p">(</span><span class="s2">"cytoscape/cytoscape-automation/for-scripters/R/r2cytoscape"</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Basic connection settings</span><span class="w">
  </span><span class="n">port.number</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1234</span><span class="w">
  </span><span class="n">base.url</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="s2">"http://localhost:"</span><span class="p">,</span><span class="w"> </span><span class="n">toString</span><span class="p">(</span><span class="n">port.number</span><span class="p">),</span><span class="w"> </span><span class="s2">"/v1"</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s2">""</span><span class="p">)</span></code></pre></figure>

<h3 id="create-enrichment-map">Create Enrichment Map</h3>

<p>We’re ready to declare our options for the Enrichment Map Cytoscape app.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="c1">### Construct path to GSEA results - 'edb' folder</span><span class="w">
  </span><span class="c1">### Ouptut from GSEA - update below to match your directory name</span><span class="w">
  </span><span class="n">gsea_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">gsea_out</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_tep_BrCa_HD_analysis_directory</span><span class="p">)</span><span class="w">
  </span><span class="n">gsea_results_filename</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">gsea_results</span><span class="p">,</span><span class="w"> </span><span class="s2">"edb"</span><span class="p">,</span><span class="w"> </span><span class="s2">"results.edb"</span><span class="p">)</span><span class="w">

  </span><span class="c1">### Define thresholds for GSEA enrichments</span><span class="w">
  </span><span class="n">em_pvalue_gsea_threshold</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"0.01"</span><span class="w">
  </span><span class="n">em_qvalue_gsea_threshold</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"0.01"</span><span class="w">

  </span><span class="c1">### Define thresholds for gene set similarity</span><span class="w">
  </span><span class="n">em_similarity_threshold</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"0.375"</span><span class="w">
  </span><span class="n">em_similarity_metric</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"COMBINED"</span><span class="w">

  </span><span class="c1">#######################################</span><span class="w">
  </span><span class="c1">#create EM pvalue &lt; 0.01 and qvalue &lt; 0.01</span><span class="w">
  </span><span class="c1">#######################################</span><span class="w">
  </span><span class="n">em_network_name</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="n">gsea_analysis_name</span><span class="p">,</span><span class="w"> </span><span class="n">em_pvalue_gsea_threshold</span><span class="p">,</span><span class="w"> </span><span class="n">em_qvalue_gsea_threshold</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s2">"_"</span><span class="p">)</span><span class="w">
  </span><span class="n">em_command</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="s2">"enrichmentmap build analysisType=gsea"</span><span class="p">,</span><span class="w">
                    </span><span class="s2">"gmtFile="</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_gmx</span><span class="p">,</span><span class="w">
                    </span><span class="s2">"pvalue="</span><span class="p">,</span><span class="w"> </span><span class="n">em_pvalue_gsea_threshold</span><span class="p">,</span><span class="w">
                    </span><span class="s2">"qvalue="</span><span class="p">,</span><span class="w"> </span><span class="n">em_qvalue_gsea_threshold</span><span class="p">,</span><span class="w">
                    </span><span class="s2">"similaritycutoff="</span><span class="p">,</span><span class="w"> </span><span class="n">em_similarity_threshold</span><span class="p">,</span><span class="w">
                    </span><span class="s2">"coefficients="</span><span class="p">,</span><span class="w"> </span><span class="n">em_similarity_metric</span><span class="p">,</span><span class="w">
                    </span><span class="s2">"ranksDataset1="</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_rank_list_path</span><span class="p">,</span><span class="w">
                    </span><span class="s2">"enrichmentsDataset1="</span><span class="p">,</span><span class="w"> </span><span class="n">gsea_results_filename</span><span class="p">,</span><span class="w">
                    </span><span class="s2">"expressionDataset1="</span><span class="p">,</span><span class="w"> </span><span class="n">expression_dataset_path</span><span class="p">,</span><span class="w">
                    </span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span><span class="w">
  </span><span class="n">current_network_suid</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="w"> </span><span class="n">doEnrichment</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">TRUE</span><span class="w"> </span><span class="p">){</span><span class="w">
    </span><span class="n">current_network_suid</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">r</span><span class="m">2</span><span class="n">cytoscape</span><span class="o">::</span><span class="n">commandRun</span><span class="p">(</span><span class="n">em_command</span><span class="p">)</span><span class="w">    
    </span><span class="n">response</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">r</span><span class="m">2</span><span class="n">cytoscape</span><span class="o">::</span><span class="n">renameNetwork</span><span class="p">(</span><span class="n">em_network_name</span><span class="p">,</span><span class="w"> </span><span class="n">network</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">current_network_suid</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">  </span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## Error in curl::curl_fetch_memory(url, handle = handle): Failed to connect to localhost port 1234: Connection refused</code></pre></figure>

<p>Let’s take a peek at the Enrichment Map.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">em_output</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">output_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"em_output.png"</span><span class="p">)</span><span class="w">
  </span><span class="n">url_png</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="n">base.url</span><span class="p">,</span><span class="w"> </span><span class="s2">"networks"</span><span class="p">,</span><span class="w"> </span><span class="n">current_network_suid</span><span class="p">,</span><span class="w"> </span><span class="s2">"views/first.png"</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s2">"/"</span><span class="p">)</span><span class="w">
  
  </span><span class="c1">### Pause for Cytoscape to render</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="w"> </span><span class="n">doEnrichment</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">TRUE</span><span class="w"> </span><span class="p">){</span><span class="w">
    </span><span class="n">response</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">httr</span><span class="o">::</span><span class="n">GET</span><span class="p">(</span><span class="n">url</span><span class="o">=</span><span class="n">url_png</span><span class="p">)</span><span class="w">
    </span><span class="n">writeBin</span><span class="p">(</span><span class="n">response</span><span class="o">$</span><span class="n">content</span><span class="p">,</span><span class="w"> </span><span class="n">em_output</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span></code></pre></figure>

<p><img src="./em_output.png" alt="Enrichment Map" /></p>

<p>Often times, the complexity of an Enrichment Map can be reduced even further: Clusters of gene sets can be collapsed and annotated with a representative label gleaned from the characteristics of the individual gene sets.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="c1">### Auto-Annotate the Enrichment Map</span><span class="w">
  </span><span class="n">aa_command</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="s2">"autoannotate annotate-clusterBoosted clusterAlgorithm=MCL maxWords=3 network="</span><span class="p">,</span><span class="w"> </span><span class="n">em_network_name</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span><span class="w">
  
  </span><span class="c1">### Enrichment Map command will return the suid of newly created network.</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="w"> </span><span class="n">doEnrichment</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">TRUE</span><span class="w"> </span><span class="p">){</span><span class="w">
    </span><span class="n">response</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">r</span><span class="m">2</span><span class="n">cytoscape</span><span class="o">::</span><span class="n">commandRun</span><span class="p">(</span><span class="n">aa_command</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span><span class="w">  </span></code></pre></figure>

<p>Finally, let’s get a view of our annotated Enrichment Map.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="w">  </span><span class="n">em_output_aa</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file.path</span><span class="p">(</span><span class="n">output_dir</span><span class="p">,</span><span class="w"> </span><span class="s2">"em_output_aa.png"</span><span class="p">)</span><span class="w">
  </span><span class="n">url_png</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="n">base.url</span><span class="p">,</span><span class="w"> </span><span class="s2">"networks"</span><span class="p">,</span><span class="w"> </span><span class="n">current_network_suid</span><span class="p">,</span><span class="w"> </span><span class="s2">"views/first.png"</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s2">"/"</span><span class="p">)</span><span class="w">
  
  </span><span class="c1">### Pause for Cytoscape to render</span><span class="w">
  </span><span class="k">if</span><span class="p">(</span><span class="w"> </span><span class="n">doEnrichment</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">TRUE</span><span class="w"> </span><span class="p">){</span><span class="w">
    </span><span class="n">Sys.sleep</span><span class="p">(</span><span class="m">30</span><span class="p">)</span><span class="w">
    </span><span class="n">response</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">httr</span><span class="o">::</span><span class="n">GET</span><span class="p">(</span><span class="n">url</span><span class="o">=</span><span class="n">url_png</span><span class="p">)</span><span class="w">
    </span><span class="n">writeBin</span><span class="p">(</span><span class="n">response</span><span class="o">$</span><span class="n">content</span><span class="p">,</span><span class="w"> </span><span class="n">em_output_aa</span><span class="p">)</span><span class="w">
  </span><span class="p">}</span></code></pre></figure>

<p><img src="./em_output_aa.png" alt="Annotated Enrichment Map" /></p>

<p>Please refer to the full ‘RNA-Seq to Enrichment Map’ workflow for details.</p>

<h2 id="references">References</h2>
<div class="footnotes">
  <ol>
    <li id="fn:1">
      <p><a href="http://linkinghub.elsevier.com/retrieve/pii/S1535610815003499">Best MG <em>et al.</em> RNA-Seq of Tumor-Educated Platelets Enables Blood-Based Pan-Cancer, Multiclass, and Molecular Pathway Cancer Diagnostics. Cancer Cell. 2015 Nov 9; 28(5): 666–676</a> <a href="#fnref:1" class="reversefootnote">&#8617;</a></p>
    </li>
    <li id="fn:2">
      <p><a href="https://www.ncbi.nlm.nih.gov/pubmed/21436837">Semple JW. <em>et al</em>. Platelets and the immune continuum. Nat Rev Immunol. 2011 Apr;11(4):264-74</a> <a href="#fnref:2" class="reversefootnote">&#8617;</a></p>
    </li>
    <li id="fn:3">
      <p><a href="https://www.ncbi.nlm.nih.gov/pubmed/21258396">Gay LJ and Felding-Habermann B. Contribution of platelets to tumour metastasis. Nat Rev Cancer. 2011 Feb;11(2):123-34</a> <a href="#fnref:3" class="reversefootnote">&#8617;</a></p>
    </li>
    <li id="fn:4">
      <p><a href="http://www.nature.com/doifinder/10.1038/nmeth.3252">Huber <em>et al.</em> Orchestrating high-throughput genomic analysis with Bioconductor. Nature Methods. 2015 Feb, 12(2)</a> <a href="#fnref:4" class="reversefootnote">&#8617;</a></p>
    </li>
    <li id="fn:5">
      <p><a href="http://genomebiology.com/2010/11/3/R25">Robinson MD and Oshlack 1. A scaling normalization method for differential expression analysis of RNA-seq data. Genome Biology 2010, 11:R25</a> <a href="#fnref:5" class="reversefootnote">&#8617;</a></p>
    </li>
    <li id="fn:6">
      <p><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0013984">Daniele Merico <em>et al.</em> Enrichment Map: A Network-Based Method for Gene-Set Enrichment Visualization and Interpretation. PLoS One. 2010; 5(11): e13984</a> <a href="#fnref:6" class="reversefootnote">&#8617;</a></p>
    </li>
  </ol>
</div>

  </div>
</div>

  

  

</div>

      </div>
    </div>

    

    <label for="sidebar-checkbox" class="sidebar-toggle" data-toggle="tooltip" data-placement="right" title="Toggle sidebar"></label>

    <script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-43341809-8', 'auto');
  ga('send', 'pageview');

  $(document).ready(function() {
    // Link
    $('a').click(function() {
      var id = $(this).attr("href");

      var label = "link";
      var action = id;
      ga('send', 'event', label, action);
      // console.log(label + " " + action);
    });
  });
</script>

  </body>
</html>
