---
title: Press Pause
subtitle: Enrichment analysis identifies transcriptional elongation factors as key vulnerabilities for glioblastoma growth <em>in vivo</em> 
reference: Miller <em>et al.</em> Nature 547 (2017)
figures: 
  figure_1: figure_northcott_nature_2017.png
  figure_2: figure_1e_miller_nature_2017.png
badge: pathway enrichment
cover: cover.jpg
pdf: 
date: 2017-07-01
layout: document
category: Pathway_Enrichment_Analysis
draft: FALSE
reflist:
  - 25250735
  - 24360279
  - 18772890
  - 22770212
  - 24153301
  - 25693130
  - 28886003
  - 28678782
  - 28678780
---

## Quick Summary
- Glioblastoma Multiforme (GBM) is an aggressive form of malignant brain tumour 
- A functional screen identified genes that GBM requires for a growth advantage *in vivo* 
- Enrichment analysis identified 'transcriptional pause release' as a shared theme 
- Knockout of transcriptional pause release genes in GBM xenografts improves survival   

## Author Profile

## Context
Genes encoding chromatin regulators are often altered in cancers and a growing body of experimental observations indicate that global remodelling of chromatin is common in oncogenesis (Dawson 2012). This has spurred intense interest in drugs targeting protein regulators of chromatin state and transcription. Indeed, the US Food and Drug Administration has approved several such small molecule inhibitors for routine clinical use against certain cancers (Helin 2013). Functional screens that determine how gene expression knockdown impact cell growth are an attractive way to discover new epigenetic-based targets (Letai 2017).

## Question
What molecular mechanisms support the growth advantage of brain tumour cells?     

## Goals

Gliomas account for about 30% of tumours arising in the central nervous system and nearly half of these are categorized as Glioblastoma Multiforme (GBM), the most common and aggressive form of malignant brain tumours (Bastien 2015). Treatment for GBM involves surgery followed by chemotherapy and/or radiation, however long-term survival remains poor. Few therapeutic advancements for GBM have been seen over the past two decades. 

Large-scale characterization of the mutational landscape of GBM tumours have helped flag well-known pathways (RB, p53 and PI3K/AKT) as well as those not previously appreciated to have an association with the disease (NF-1 and IDH1) (The Cancer Genome Atlas Research Network 2008). Therapeutics targeting PI3K, receptor tyrosine kinases or the p53 pathway have advanced to clinical evaluation (Bastien 2015). However, their performance in clinical trials has been disappointing, arguing that alternative therapeutic targets should be identified. 

Miller *et al.* (Miller 2017) couple functional screening with enrichment analysis to uncover epigenentic genes that are vulnerabilities for the growth advantage of GBM *in vivo*.

## Approach

Miller and colleagues set out to determine whether chromatin regulators are amongst the genes that provide GBM a growth advantage. To accomplish this, they developed a parallel screening strategy whereby patient-derived glioblastoma cells were stably transduced with trangenes for short hairpin RNAs (shRNA) targeting expression of 406 chromatin regulators such that, on average, each cell would express only a single sequence of shRNA (Figure 1). Next, the pooled transgenic population was either passaged in culture (*in vitro*) or injected intracranially (IC) into mice (*in vivo*). Following a three week growth period in each microenvironment, the number of cells expressing each type of shRNA was determined. Relatively low counts were interpreted as evidence that GBM depends on the endogenous gene to maintain its numbers in the population, that is, for a growth advantage.  

Using this approach, the authors observed a large disparity in the number of dependencies identified in each microenvironment: 60 for cells grown *in vivo* compared to only 14 for the *in vitro* population. The most surprising and unsettling finding was that, apart from control and essential genes (RNA polymerase and DNA methyltrasferase), there was no overlap in dependencies (Figure 1). This microenvironment-specificity should cause us to pause and reconsider the extent to which dependencies identified using *in vitro* systems can be extrapolated to more realistic situations.   

![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_1 }}){: height="100px".img-responsive }

<div class="figure-legend well well-lg text-justify">
  <strong>Figure 1. </strong> RNA knockdown screens for essential genes. Miller <em>et al.</em> took cells from patients with the cancer glioblastoma, and introduced different inhibitory RNA molecules into different cells (colour coded). Each RNA molecule inhibited expression of a gene involved in epigenetic modification — regulation of chemical modifications on DNA and associated proteins that alter gene activity without changing the underlying DNA sequence. The authors grew the cells in culture or in the brains of mice for two to three weeks, then analysed cell depletion, to see which genes must be properly expressed for normal glioblastoma-cell growth. The <em>in vivo</em> and <em>in vitro</em> results showed almost no overlap. Genes vital for growth <em>in vivo</em> screen related to a regulatory process in gene transcription called pause release. (Adapted Northcott 2017).
</div>

Many experimental approaches, including the screening strategy described above, can generate a list of 'genes of interest'. However, it is not immediately obvious how genes relate to one another.

> Many experimental approaches, such as the screening strategy described above, can generate a list of 'genes of interest'. However, it is not immediately obvious how genes relate to one another.

To provide insight into function relationships between dependencies, the authors employed Gene Ontology enrichment analysis. Intuitively, this approach determines if the overlap between genes annotated with the same Gene Ontology (GO) term and those identified as growth dependencies is larger than would reasonably be expected by chance alone. When this is indeed the case, we say that GBM growth dependencies in a microenvironment are 'enriched' for a particular GO term. 

Using this approach, Miller *et al.* found that *in vitro* dependencies were enriched for genes that promote cellular metabolism and macromolecular biogenesis, whereas *in vivo* there was an enrichment for genes controlling transcriptional elongation and pause release. The relationship between *in vivo* dependencies is underscored by the physical proximity and associated functions they have as part of the transcriptional machinery (Figure 2). Using a related technique called Gene Set Enrichment Analysis (GSEA) on global gene expression measurements, the authors were also able to demonstrate microenvironment-specific differences in pathway activation, further underscoring their disparity.             

![image]({{ site.baseurl }}/{{ site.media_root }}{{ page.id }}/{{ page.figures.figure_2 }}){: height="100px".img-responsive }

<div class="figure-legend well well-lg text-justify">
  <strong>Figure 2. Parallel in vivo and in vitro screen identifies environment-specific cancer dependencies and reveals transcriptional pause–release and elongation as an in vivo-specific target. </strong> Schematic of transcription elongation machinery, highlighting in vivo-specific hits. TFs, transcription factors; TSS, transcription start site. (Adapted Miller 2017)
</div>

Finally, Miller *et al.* prioritized the 12 elongation factors that were *in vivo* dependencies (Figure 2) based upon how their expression correlated with the some of the largest gene expression differences *in vivo*. The most positively correlated gene was the JmjC-domain-containing protein, JMJD6, which has been shown to promote transcriptional pause release by removing repressive histone methylation (Liu 2013). The preclinical value of JMJD6 was assessed by growing glioblastoma cells with knockout (CRISPR-Cas9) of JMJD6 in an orthotopic xenograft mouse model. Indeed, mice harbouring JMJD6-deficient cells demonstrated a notable survival advantage: Over 25% were tumour-free long after control mice had succumb to disease.   

## Summary

Here, a list of genes representing growth dependencies in GBM was derived from a functional screen. The intimate  physical and functional relationship of the genes in the list was illuminated using enrichment analysis. This  experimental and analysis pipeline represents an efficient discovery pipeline to identify clinically-relevant vulnerabilities of GBM.